<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="SpatialEcoTyper">
<title>Discovery of Spatial Ecotypes from Multiple Samples • SpatialEcoTyper</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Discovery of Spatial Ecotypes from Multiple Samples">
<meta property="og:description" content="SpatialEcoTyper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-default navbar-expand-lg bg-light" data-bs-theme="default"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpatialEcoTyper</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.0.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href="../articles/SingleSample.html">Discovery of Spatial Ecotypes from a Single Sample</a>
    <a class="dropdown-item" href="../articles/Integration.html">Discovery of Spatial Ecotypes from Multiple Samples</a>
    <a class="dropdown-item" href="../articles/TrainRecoveryModels.html">Development of NMF Models for Spatial Ecotype Recovery</a>
    <a class="dropdown-item" href="../articles/Recovery_Spatial.html">Recovery of Spatial Ecotypes from Spatial Transcriptomics Data</a>
    <a class="dropdown-item" href="../articles/Recovery_scRNA.html">Recovery of Spatial Ecotypes from Single-Cell Gene Expression Data</a>
    <a class="dropdown-item" href="../articles/Recovery_Bulk.html">Recovery of Spatial Ecotypes from Bulk Gene Expression Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">R functions</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Discovery of Spatial Ecotypes from Multiple Samples</h1>
            
      
      <small class="dont-index">Source: <a href="https://digitalcytometry.github/SpatialEcoTypervignettes/Integration.Rmd" class="external-link"><code>vignettes/Integration.Rmd</code></a></small>
      <div class="d-none name"><code>Integration.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>In this tutorial, we will illustrate how to identify conserved
spatial ecotypes (SEs) across multiple samples using the
<code>MultiSpatialEcoTyper</code> function. Each sample represents a
single-cell spatial transcriptomics data, including a gene expression
profile and associated single-cell metadata.</p>
<p><strong>First load required packages for this vignette</strong></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">parallel</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://googledrive.tidyverse.org" class="external-link">googledrive</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://renozao.github.io/NMF/" class="external-link">NMF</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://digitalcytometry.github/SpatialEcoTyper" class="external-link">SpatialEcoTyper</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<p>In this tutorial, we will analyze single-cell spatial transcriptomics
data from cancer samples provided by <a href="https://vizgen.com/data-release-program/" class="external-link">Vizgen’s MERSCOPE FFPE
Human Immuno-oncology</a>. Specifically, we will identify conserved SEs
from a melanoma sample and a colon cancer sample. For efficiency, we
have selected a subset region from each sample. You can download the
gene expression profiles and single-cell metadata <a href="https://drive.google.com/open?id=1CYncsGZ4aTaIRGkR4rVAmbOGskVyoTEQ&amp;usp=drive_fs" class="external-link"><code>here</code></a>.</p>
<details open><summary><strong>Text files as input</strong>
</summary><p>First, download the demo data from Google Drive</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_deauth.html" class="external-link">drive_deauth</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Disable Google sign-in requirement</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1CoQmU3u8MoVC8RbLUvTDQmOuJJ703HHB"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_counts.tsv.gz"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1CgUOQKrWY_TG61o5aw7J9LZzE20D6NuI"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_scmeta.tsv"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1ChwONUjr_yoURodnkDBj68ZUbjdHtmP6"</span><span class="op">)</span>, <span class="st">"HumanColonCancerPatient2_subset_counts.tsv.gz"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1CipRjjD7cqzqKO0Yf4LUdsEw1XDzP6JS"</span><span class="op">)</span>, <span class="st">"HumanColonCancerPatient2_subset_scmeta.tsv"</span><span class="op">)</span></span></code></pre></div>
<p>Then, load text files into R. You can use <code>read.table</code> for
small files or <code>fread</code> from the <code>data.table</code>
package for larger files.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load single-cell metadata</span></span>
<span><span class="co"># Required columns: X, Y, CellType</span></span>
<span><span class="co"># Recommend a Region column if pathologist region annotations are available</span></span>
<span><span class="va">scmeta1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"HumanMelanomaPatient1_subset_scmeta.tsv"</span>,</span>
<span>                      sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">scmeta2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"HumanColonCancerPatient2_subset_scmeta.tsv"</span>,</span>
<span>                      sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scmeta1</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"X"</span>, <span class="st">"Y"</span>, <span class="st">"CellType"</span>, <span class="st">"Region"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                         X         Y   CellType Region</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3655 1894.706 -6367.766 Fibroblast Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657 1942.480 -6369.602 Fibroblast Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658 1963.007 -6374.026 Fibroblast Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660 1981.600 -6372.266 Fibroblast Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661 1742.939 -6374.851 Fibroblast Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3663 1921.683 -6383.309 Fibroblast Stroma</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scmeta2</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"X"</span>, <span class="st">"Y"</span>, <span class="st">"CellType"</span>, <span class="st">"Region"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                            X         Y   CellType       Region</span></span>
<span><span class="co">## HumanColonCancerPatient2__cell_1085 5057.912 -1721.871 Macrophage Inner margin</span></span>
<span><span class="co">## HumanColonCancerPatient2__cell_1205 5035.339 -1783.352 Macrophage Inner margin</span></span>
<span><span class="co">## HumanColonCancerPatient2__cell_1312 5012.093 -1908.390 Macrophage Inner margin</span></span>
<span><span class="co">## HumanColonCancerPatient2__cell_1625 5073.743 -1861.740 Macrophage Inner margin</span></span>
<span><span class="co">## HumanColonCancerPatient2__cell_1629 5030.235 -1867.441 Macrophage Inner margin</span></span>
<span><span class="co">## HumanColonCancerPatient2__cell_1639 5020.448 -1882.751 Macrophage Inner margin</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load single-cell gene expression data</span></span>
<span><span class="va">scdata1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">"HumanMelanomaPatient1_subset_counts.tsv.gz"</span>,</span>
<span>                sep <span class="op">=</span> <span class="st">"\t"</span>,header <span class="op">=</span> <span class="cn">TRUE</span>, data.table <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scdata1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">scdata1</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">scdata1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">scdata1</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">scdata2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">"HumanColonCancerPatient2_subset_counts.tsv.gz"</span>,</span>
<span>                sep <span class="op">=</span> <span class="st">"\t"</span>,header <span class="op">=</span> <span class="cn">TRUE</span>, data.table <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scdata2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">scdata2</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">scdata2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">scdata2</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Match the cell ids in the meta data and gene expression data</span></span>
<span><span class="va">scdata1</span> <span class="op">&lt;-</span> <span class="va">scdata1</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scmeta1</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">scdata2</span> <span class="op">&lt;-</span> <span class="va">scdata2</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scmeta2</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
</details><details><summary><strong>Sparse matrix as input</strong>
</summary><p>Mtx files can be loaded into R using the <a href="https://satijalab.org/seurat/reference/readmtx" class="external-link"><code>ReadMtx</code>
function</a> from the Seurat package.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"13M3xhRxp0xK9gf5F4DE9idSBFqVQIXDT"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_counts.mtx.gz"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"136feRaFjMtNvduLTm5xqa3WhyyoG4Xzo"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_cells.tsv.gz"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"13QprWzJhzzUy_w3XSrjlt9pjf2n-G7HV"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_genes.tsv.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"17fH9BAAugYi1FMLrMuTxojtOoklFBB-K"</span><span class="op">)</span>, <span class="st">"HumanColonCancerPatient2_subset_counts.mtx.gz"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"17a1f1VjxJSje_uyPt6zA97zTps9ko6rk"</span><span class="op">)</span>, <span class="st">"HumanColonCancerPatient2_subset_cells.tsv.gz"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"17fH0jE5b2YqJ5fpQtYil27k-7YhzArD6"</span><span class="op">)</span>, <span class="st">"HumanColonCancerPatient2_subset_genes.tsv.gz"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scdata1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ReadMtx.html" class="external-link">ReadMtx</a></span><span class="op">(</span>mtx <span class="op">=</span> <span class="st">"HumanMelanomaPatient1_subset_counts.mtx.gz"</span>, cells <span class="op">=</span> <span class="st">"HumanMelanomaPatient1_subset_cells.tsv.gz"</span>, features <span class="op">=</span> <span class="st">"HumanMelanomaPatient1_subset_genes.tsv.gz"</span>, feature.column <span class="op">=</span> <span class="fl">1</span>, cell.column <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">scdata2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ReadMtx.html" class="external-link">ReadMtx</a></span><span class="op">(</span>mtx <span class="op">=</span> <span class="st">"HumanColonCancerPatient2_subset_counts.mtx.gz"</span>, cells <span class="op">=</span> <span class="st">"HumanColonCancerPatient2_subset_cells.tsv.gz"</span>, features <span class="op">=</span> <span class="st">"HumanColonCancerPatient2_subset_genes.tsv.gz"</span>, feature.column <span class="op">=</span> <span class="fl">1</span>, cell.column <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
</details>
</div>
<div class="section level2">
<h2 id="data-normalization">Data normalization<a class="anchor" aria-label="anchor" href="#data-normalization"></a>
</h2>
<p>Before proceeding with the SpatialEcoTyper analysis, the gene
expression data should be normalized. This can be done using either <a href="https://satijalab.org/seurat/reference/normalizedata" class="external-link">NormalizeData</a>
or <a href="https://satijalab.org/seurat/articles/seurat5_spatial_vignette_2" class="external-link">SCTransform</a>.
Here, we will use <a href="https://satijalab.org/seurat/articles/seurat5_spatial_vignette_2" class="external-link">SCTransform</a>
normalization. For faster computation, it is recommended to install the
<code>glmGamPoi</code> package.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"glmGamPoi"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"glmGamPoi"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">tmpobj1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span><span class="va">scdata1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://satijalab.org/seurat/reference/SCTransform.html" class="external-link">SCTransform</a></span><span class="op">(</span>clip.range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">10</span>, <span class="fl">10</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">tmpobj2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span><span class="va">scdata2</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://satijalab.org/seurat/reference/SCTransform.html" class="external-link">SCTransform</a></span><span class="op">(</span>clip.range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">10</span>, <span class="fl">10</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_version</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html" class="external-link">packageVersion</a></span><span class="op">(</span><span class="st">"SeuratObject"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="va">seurat_version</span><span class="op">&lt;</span><span class="fl">5</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">normdata1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">tmpobj1</span>, <span class="st">"data"</span><span class="op">)</span></span>
<span>  <span class="va">normdata2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">tmpobj2</span>, <span class="st">"data"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="kw">else</span><span class="op">{</span></span>
<span>  <span class="va">normdata1</span> <span class="op">&lt;-</span> <span class="va">tmpobj1</span><span class="op">[[</span><span class="st">"SCT"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">data</span></span>
<span>  <span class="va">normdata2</span> <span class="op">&lt;-</span> <span class="va">tmpobj2</span><span class="op">[[</span><span class="st">"SCT"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">data</span></span>
<span><span class="op">}</span></span></code></pre></div>
<summary>
To use <code>NormalizeData</code> for the normalization:
</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normdata1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">scdata1</span><span class="op">)</span></span>
<span><span class="va">normdata2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">scdata2</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preview-of-the-samples">Preview of the samples<a class="anchor" aria-label="anchor" href="#preview-of-the-samples"></a>
</h2>
<p>The melanoma sample includes spatial expression data for 500 genes
across 27,907 cells, while the colon cancer sample contains data for
38,080 cells. In both samples, cells were categorized into ten distinct
cell types: B cells, CD4+ T cells, CD8+ T cells, NK cells, plasma cells,
macrophages, dendritic cells (DCs), fibroblasts, endothelial cells, and
cancer cells of origin (CCOs). Due to their high heterogeneity, CCOs
were excluded from the analysis.</p>
<p>You can use the <code>SpatialView</code> function to visualize the
spatial distribution of single cells within the tissues:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize the cell type annotations in the tissue</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">scmeta1</span>, by <span class="op">=</span> <span class="st">"CellType"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"SKCM"</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/discrete.html" class="external-link">kelly</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">scmeta2</span>, by <span class="op">=</span> <span class="st">"CellType"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"CRC"</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/discrete.html" class="external-link">kelly</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span></span></code></pre></div>
<p><img src="Integration_files/figure-html/viewct-1.png" width="700"></p>
<p>All spatial regions were annotated into four groups: tumor, inner
margin, outer margin, and stroma. The tumor and stroma regions were
defined based on the density of cancer cells. The inner and outer
margins were delineated as regions extending 250 μm inside and outside
the tumor boundaries, respectively. The <code>SpatialView</code>
function can also be used to visualize these regions.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize the regions in the tissue</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">scmeta1</span>, by <span class="op">=</span> <span class="st">"Region"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"SKCM"</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span>, palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">scmeta2</span>, by <span class="op">=</span> <span class="st">"Region"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"CRC"</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span>, palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span></span></code></pre></div>
<p><img src="Integration_files/figure-html/viewregion-1.png" width="700"></p>
<p>You can also visualize continuous characteristics, such as the
minimum distance of each single cell to the tumor/stroma margin, or
specific gene expression levels, using the <code>SpatialView</code>
function:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize the distance to tumor margin</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">scmeta1</span>, by <span class="op">=</span> <span class="st">"Dist2Interface"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"SKCM"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_colour_gradient2</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">"#5e3c99"</span>, high <span class="op">=</span> <span class="st">"#e66101"</span>, mid <span class="op">=</span> <span class="st">"#d9d9d9"</span>, midpoint <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">scmeta2</span>, by <span class="op">=</span> <span class="st">"Dist2Interface"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"CRC"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_colour_gradient2</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">"#5e3c99"</span>, high <span class="op">=</span> <span class="st">"#e66101"</span>, mid <span class="op">=</span> <span class="st">"#d9d9d9"</span>, midpoint <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span></span></code></pre></div>
<p><img src="Integration_files/figure-html/viewdist-1.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Visualize gene expression in the tissue</span></span>
<span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">scmeta1</span></span>
<span><span class="va">gg</span><span class="op">$</span><span class="va">Expression</span> <span class="op">&lt;-</span> <span class="va">normdata1</span><span class="op">[</span><span class="st">"STAT1"</span>, <span class="op">]</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">gg</span>, by <span class="op">=</span> <span class="st">"Expression"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_color_viridis_c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">scmeta2</span></span>
<span><span class="va">gg</span><span class="op">$</span><span class="va">Expression</span> <span class="op">&lt;-</span> <span class="va">normdata2</span><span class="op">[</span><span class="st">"STAT1"</span>, <span class="op">]</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">gg</span>, by <span class="op">=</span> <span class="st">"Expression"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_color_viridis_c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span></span></code></pre></div>
<p><img src="Integration_files/figure-html/viewgene-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="spatialecotyper-analysis-across-multiple-samples">SpatialEcoTyper analysis across multiple samples<a class="anchor" aria-label="anchor" href="#spatialecotyper-analysis-across-multiple-samples"></a>
</h2>
<p><img src="images/MultiSpatialEcoTyper.png" width="100%"></p>
<p>The <code>MultiSpatialEcoTyper</code> function enables the
integration of single-cell spatial transcriptomics data across multiple
samples, facilitating the identification of SEs shared across samples or
cancer types. The analysis is performed in three stages:</p>
<ul>
<li>
<strong>Independent SpatialEcoTyper Analysis</strong>:
SpatialEcoTyper is run on each sample independently to identify spatial
clusters and construct cell-type-specific gene expression profiles for
all spatial clusters.</li>
<li>
<strong>Integration via Similarity Network Fusion</strong>: Spatial
clusters from different samples are integrated to learn a unified
embedding using Similarity Network Fusion (Wang et al., 2014).</li>
<li>
<strong>NMF Clustering</strong>: Non-negative Matrix Factorization
(NMF) is applied to the unified embedding to identify conserved SEs
across samples.</li>
</ul>
<details open><summary><strong>Key arguments for <code>MultiSpatialEcoTyper</code></strong>
</summary><ul>
<li>
<code>data_list</code> A named list of expression matrices where
each matrix represents gene expression data for a sample. The columns of
each matrix correspond to cells, and the rows correspond to genes. The
list names should match sample names; otherwise, samples will be named
as ‘Sample1’, ‘Sample2’, etc.</li>
<li>
<code>metadata_list</code> A named list of metadata data frames,
where each data frame contains metadata corresponding to the cells in
the expression matrices. Each metadata data frame should include at
least three columns (X, Y, CellType), and the row names should match the
cell IDs (column names) in the expression matrix.</li>
<li>
<code>outdir</code> Directory where the results will be saved.
Defaults to the current directory with a subdirectory named
“SpatialEcoTyper_results_” followed by the current date.</li>
<li>
<code>normalization.method</code> Method for normalizing the
expression data. Options include “None” (default), “SCT”, or other
methods compatible with Seurat’s <code>NormalizeData</code>
function.</li>
<li>
<code>nmf_ranks</code> Integer or vector specifying the number of
clusters (10 by default). If a vector is provided, the function tests
all ranks and selects the optimal rank for NMF, though this may increase
computation time.</li>
<li>
<code>radius</code> Numeric specifying the radius (in the same units
as spatial coordinates) for defining spatial neighborhoods around each
cell. Default is 50.</li>
<li>
<code>min.cts.per.region</code> Integer specifying the minimum
number of cell types (default: 2) required for a microregion.</li>
<li>
<code>Region</code> Column name in metadata data frames containing
pathologist region annotation (default: NULL).</li>
<li>
<code>ncores</code> Integer specifying the number of cores for
parallel processing. Default is 1.</li>
</ul>
<p>You can type <code><a href="../reference/MultiSpatialEcoTyper.html">?MultiSpatialEcoTyper</a></code> to visualize the full
manual.</p>
</details><div class="section level3">
<h3 id="preparing-data">Preparing data<a class="anchor" aria-label="anchor" href="#preparing-data"></a>
</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>SKCM <span class="op">=</span> <span class="va">normdata1</span>, CRC <span class="op">=</span> <span class="va">normdata2</span><span class="op">)</span></span>
<span><span class="va">metadata_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>SKCM <span class="op">=</span> <span class="va">scmeta1</span>, CRC <span class="op">=</span> <span class="va">scmeta2</span><span class="op">)</span></span></code></pre></div>
<p><strong>Important note</strong>: In the examples below, we will use
10 clusters and 5 runs per rank for demonstration purposes. To select
the optimal number of clusters and obtain robust results, you can use
the <code>nmf_ranks</code> argument with a vector of integers to test
multiple ranks for NMF analysis and set <code>nrun.per.rank</code> to 30
or higher. You can also use the <code>nmfClustering</code> function for
a detailed rank selection process, as outlined in the next section
(<code>NMF clustering for SpatialEcoTyper</code>).</p>
</div>
<div class="section level3">
<h3 id="pathologist-region-annotation-unavailable">Pathologist region annotation unavailable<a class="anchor" aria-label="anchor" href="#pathologist-region-annotation-unavailable"></a>
</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MultiSpatialEcoTyper.html">MultiSpatialEcoTyper</a></span><span class="op">(</span><span class="va">data_list</span>, <span class="va">metadata_list</span>, </span>
<span>                     normalization.method <span class="op">=</span> <span class="st">"None"</span>,</span>
<span>                     nmf_ranks <span class="op">=</span> <span class="fl">10</span>,  <span class="co"># Number of clusters</span></span>
<span>                     nrun.per.rank <span class="op">=</span> <span class="fl">5</span>,  <span class="co"># Recommend 30 or higher for robust results</span></span>
<span>                     ncores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co"># This step takes ~5 minutes on a macOS with an Apple M1 Pro chip and 16 GB memory.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pathologist-region-annotation-available">Pathologist region annotation available<a class="anchor" aria-label="anchor" href="#pathologist-region-annotation-available"></a>
</h3>
<p>If region annotations are available, we recommend to include the
region annotations in all metadata and specify the <code>Region</code>
in the <code>MultiSpatialEcoTyper</code> function, which could improve
integration accuracy.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/MultiSpatialEcoTyper.html">MultiSpatialEcoTyper</a></span><span class="op">(</span><span class="va">data_list</span>, <span class="va">metadata_list</span>, </span>
<span>                     normalization.method <span class="op">=</span> <span class="st">"None"</span>,</span>
<span>                     nmf_ranks <span class="op">=</span> <span class="fl">10</span>,  <span class="co"># Number of clusters</span></span>
<span>                     nrun.per.rank <span class="op">=</span> <span class="fl">5</span>,  <span class="co"># Recommend 30 or higher for robust results</span></span>
<span>                     Region <span class="op">=</span> <span class="st">"Region"</span>, <span class="co"># Use pathologist annotations if available</span></span>
<span>                     ncores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## agg_png </span></span>
<span><span class="co">##       2</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## This step takes ~5 minutes to complete on macOS with an Apple M1 Pro chip and 16 GB memory.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="using-pre-existing-spatialecotyper-results">Using pre-existing SpatialEcoTyper results<a class="anchor" aria-label="anchor" href="#using-pre-existing-spatialecotyper-results"></a>
</h3>
<p>If you’ve already run <code>SpatialEcoTyper</code> for each sample,
you can integrate the results directly using
<code>IntegrateSpatialEcoTyper</code>.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1nlecuQ90md15oYMKGVES68TaSoKHwZIg"</span><span class="op">)</span>, <span class="st">"CRC_SpatialEcoTyper_results.rds"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1nXwimQCgSqcq5NJR4kOtF4xoJoNGlT5t"</span><span class="op">)</span>, <span class="st">"SKCM_SpatialEcoTyper_results.rds"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>SKCM <span class="op">=</span> <span class="va">normdata1</span>, CRC <span class="op">=</span> <span class="va">normdata2</span><span class="op">)</span></span>
<span><span class="va">SpatialEcoTyper_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>SKCM <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"SKCM_SpatialEcoTyper_results.rds"</span><span class="op">)</span>,</span>
<span>                             CRC <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"CRC_SpatialEcoTyper_results.rds"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/IntegrateSpatialEcoTyper.html">IntegrateSpatialEcoTyper</a></span><span class="op">(</span><span class="va">SpatialEcoTyper_list</span>, <span class="va">data_list</span>, </span>
<span>                         outdir <span class="op">=</span> <span class="st">"SpatialEcoTyper_results/"</span>,</span>
<span>                         normalization.method <span class="op">=</span> <span class="st">"None"</span>, </span>
<span>                         nmf_ranks <span class="op">=</span> <span class="fl">10</span>, <span class="co"># Number of clusters</span></span>
<span>                         nrun.per.rank <span class="op">=</span> <span class="fl">5</span>,  <span class="co"># recommend 30 or higher for robust results</span></span>
<span>                         ncores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co"># This step takes ~5 minutes on a macOS with an Apple M1 Pro chip and 16 GB memory.</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="output-results">Output results<a class="anchor" aria-label="anchor" href="#output-results"></a>
</h2>
<p>All results will be saved in the specified directory
(<code>outdir</code>) with a subdirectory named
<code>SpatialEcoTyper_results_</code> followed by the current date. The
output files include:</p>
<ul>
<li>
<strong><code>(SampleName)_SpatialEcoTyper_results.rds</code></strong>
<code>SpatialEcoTyper</code> result for each sample.</li>
<li>
<strong>MultiSE_integrated_final.rds</strong> A matrix representing
the fused similarity matrix of spatial clusters across all samples.</li>
<li>
<strong>MultiSE_integrated_final_hmap.pdf</strong> A heatmap
visualizing the fused similarity matrix of spatial clusters, grouped by
samples and SEs.</li>
<li>
<strong>MultiSE_metadata_final.rds</strong> Single-cell metadata
with an <code>SE</code> column annotating the discovered SEs, saved as
an RDS file.</li>
<li>
<strong>MultiSE_metadata_final.tsv</strong> The same as
<code>MultiSE_metadata_final.rds</code>, but saved in TSV format.</li>
<li>
<strong>MultiSE_NMF_Cophenetic_Dynamics.pdf</strong> A figure
showing the cophenetic coefficients for different numbers of clusters,
available when multiple ranks were provided for NMF analysis.</li>
<li>
<strong>MultiSE_NMF_results.rds</strong> The NMF result derived from
the <code>nmfClustering</code> function. If a single rank was provided,
it contains an <code>NMFfitX1</code> object . If multiple ranks were
provided, it is a <code>list</code> containing the optimal number of
communities (<code>bestK</code>), a list of <code>NMFfitX1</code>
objects (<code>NMFfits</code>), and a <code>ggplot</code> object
(<code>p</code>) displaying the cophenetic coefficient for different
cluster numbers.</li>
</ul>
<p>The results for the demo data can be downloaded from <a href="https://drive.google.com/open?id=1H4S71HENMoig03O3FdeLwnwk2aja-Wtv&amp;usp=drive_fs" class="external-link"><code>MultiSpatialEcoTyper_results</code></a></p>
</div>
<div class="section level2">
<h2 id="nmf-clustering-for-spatialecotyper">NMF clustering for SpatialEcoTyper<a class="anchor" aria-label="anchor" href="#nmf-clustering-for-spatialecotyper"></a>
</h2>
<p>SpatialEcoTyper uses NMF to group spatial clusters from multiple
samples into conserved SEs. To determine the optimal number of clusters,
or rank, we computed cophenetic coefficient, which quantifies the
classification stability for a given rank (i.e., the number of clusters)
and ranges from 0 to 1, with 1 indicating maximally stability.
Typically, the rank at which the cophenetic coefficient begins to
decrease is chosen. However, this method can be challenging when the
cophenetic coefficient exhibits a multi-modal shape across different
ranks. In such cases, for SpatialEcoTyper analysis, we recommend
selecting the number of SEs after which the coefficient drops below 0.95
by default.</p>
<div class="section level3">
<h3 id="selecting-the-optimal-rank">Selecting the optimal rank<a class="anchor" aria-label="anchor" href="#selecting-the-optimal-rank"></a>
</h3>
<p>To determine the optimal rank, you can use the
<code>nmfClustering</code> function, which tests multiple ranks and
computes the cophenetic coefficient for each. This analysis requires the
fused similarity network matrix
(<code>MultiSE_integrated_final.rds</code>) generated from previous
step.</p>
<p>The integrated similarity matrix and metadata for the demo can be
downloaded via:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1noLABiQEyfwlVDqZ_uYZ4haooxaI7CM6"</span><span class="op">)</span>, <span class="st">"MultiSE_integrated_final.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1nrXWdvBJPkZRqg0IdP288_oGu-1qPl7A"</span><span class="op">)</span>, <span class="st">"MultiSE_metadata_final.rds"</span><span class="op">)</span></span></code></pre></div>
<p>Load the files into R:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## A matrix representing the fused similarity matrix of spatial clusters across all samples</span></span>
<span><span class="va">integrated</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"MultiSE_integrated_final.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">integrated</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 376 376</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Single-cell metadata with an `SE` column annotating the discovered SEs</span></span>
<span><span class="va">finalmeta</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"MultiSE_metadata_final.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">finalmeta</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Sample"</span>, <span class="st">"X"</span>, <span class="st">"Y"</span>, <span class="st">"CellType"</span>, <span class="st">"Region"</span>, <span class="st">"InitSE"</span>, <span class="st">"SE"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                  Sample        X         Y   CellType Region</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657   SKCM 1942.480 -6369.602 Fibroblast Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658   SKCM 1963.007 -6374.026 Fibroblast Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660   SKCM 1981.600 -6372.266 Fibroblast Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661   SKCM 1742.939 -6374.851 Fibroblast Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3664   SKCM 1706.253 -6383.428 Fibroblast Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3665   SKCM 1968.926 -6385.541 Fibroblast Stroma</span></span>
<span><span class="co">##                                           InitSE  SE</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657 SKCM..InitSE173 SE6</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658 SKCM..InitSE173 SE6</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660 SKCM..InitSE173 SE6</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661 SKCM..InitSE145 SE5</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3664  SKCM..InitSE19 SE5</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3665 SKCM..InitSE173 SE6</span></span></code></pre>
<p>Next, test multiple ranks to find the optimal number of SEs. The
running time for this step depends on the data size, number of ranks to
test, the number of runs per rank, and the number of cores for parallel
analysis. For demonstration, here we test ranks from 4 to 15 with 5 runs
per rank using four cores.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nmf_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/nmfClustering.html">nmfClustering</a></span><span class="op">(</span><span class="va">integrated</span>, ranks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">15</span>,<span class="fl">2</span><span class="op">)</span>,</span>
<span>                         nrun.per.rank <span class="op">=</span> <span class="fl">5</span>, <span class="co"># Recommend 30 or higher</span></span>
<span>                         min.coph <span class="op">=</span> <span class="fl">0.95</span>, <span class="co"># minimum cophenetic coefficient threshold</span></span>
<span>                         ncores <span class="op">=</span> <span class="fl">4</span>,</span>
<span>                         seed <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">## This process takes ~3 minutes on a macOS with an Apple M1 Pro chip and 16 GB of memory</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"The selected rank is "</span>, <span class="va">nmf_res</span><span class="op">$</span><span class="va">bestK</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "The selected rank is 8"</span></span></code></pre>
<p>To better understand how the cophenetic coefficient changes with
different ranks, you can visualize the results.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">nmf_res</span><span class="op">$</span><span class="va">p</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_files/figure-html/coph-1.png" width="700"></p>
<p>Once the optimal rank is determined, you can use it to generate the
final SE annotations.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ses</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">nmf_res</span><span class="op">$</span><span class="va">NMFfits</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"K."</span>, <span class="va">nmf_res</span><span class="op">$</span><span class="va">bestK</span><span class="op">)</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ses</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   SKCM..InitSE6 SKCM..InitSE111  SKCM..InitSE42  SKCM..InitSE55  SKCM..InitSE13 </span></span>
<span><span class="co">##               1               1               1               1               1 </span></span>
<span><span class="co">##  SKCM..InitSE50 </span></span>
<span><span class="co">##               5 </span></span>
<span><span class="co">## Levels: 1 2 3 4 5 6 7 8</span></span></code></pre>
<p>You can add the newly defined SEs into the meta data</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finalmeta</span><span class="op">$</span><span class="va">SE</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"SE"</span>, <span class="va">ses</span><span class="op">[</span><span class="va">finalmeta</span><span class="op">$</span><span class="va">InitSE</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">finalmeta</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                            CellID        X</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657 HumanMelanomaPatient1__cell_3657 1942.480</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658 HumanMelanomaPatient1__cell_3658 1963.007</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660 HumanMelanomaPatient1__cell_3660 1981.600</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661 HumanMelanomaPatient1__cell_3661 1742.939</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3664 HumanMelanomaPatient1__cell_3664 1706.253</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3665 HumanMelanomaPatient1__cell_3665 1968.926</span></span>
<span><span class="co">##                                          Y   CellType CellTypeName Region</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657 -6369.602 Fibroblast  Fibroblasts Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658 -6374.026 Fibroblast  Fibroblasts Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660 -6372.266 Fibroblast  Fibroblasts Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661 -6374.851 Fibroblast  Fibroblasts Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3664 -6383.428 Fibroblast  Fibroblasts Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3665 -6385.541 Fibroblast  Fibroblasts Stroma</span></span>
<span><span class="co">##                                  Dist2Interface          InitSE Sample  SE</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657      -894.8463 SKCM..InitSE173   SKCM SE4</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658      -904.1115 SKCM..InitSE173   SKCM SE4</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660      -907.8909 SKCM..InitSE173   SKCM SE4</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661      -874.2712 SKCM..InitSE145   SKCM SE8</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3664      -880.4758  SKCM..InitSE19   SKCM SE8</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3665      -916.7790 SKCM..InitSE173   SKCM SE4</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="using-a-different-rank">Using a different rank<a class="anchor" aria-label="anchor" href="#using-a-different-rank"></a>
</h3>
<p>If you simply want to experiment with a different rank, you can use
the <code>nmfClustering</code> function. When a single rank is
specified, it directly returns an NMFfit object for predicting SE
grouping, as shown below:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nmf_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/nmfClustering.html">nmfClustering</a></span><span class="op">(</span><span class="va">integrated</span>, ranks <span class="op">=</span> <span class="fl">6</span>, nrun.per.rank <span class="op">=</span> <span class="fl">30</span>, seed <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">ses</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">nmf_res</span><span class="op">)</span></span>
<span><span class="co">## This process takes ~6 minutes on a macOS with an Apple M1 Pro chip and 16 GB of memory</span></span></code></pre></div>
<p>You can add the newly defined SEs into the meta data</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finalmeta</span><span class="op">$</span><span class="va">SE</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"SE"</span>, <span class="va">ses</span><span class="op">[</span><span class="va">finalmeta</span><span class="op">$</span><span class="va">InitSE</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">finalmeta</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                            CellID        X</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657 HumanMelanomaPatient1__cell_3657 1942.480</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658 HumanMelanomaPatient1__cell_3658 1963.007</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660 HumanMelanomaPatient1__cell_3660 1981.600</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661 HumanMelanomaPatient1__cell_3661 1742.939</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3664 HumanMelanomaPatient1__cell_3664 1706.253</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3665 HumanMelanomaPatient1__cell_3665 1968.926</span></span>
<span><span class="co">##                                          Y   CellType CellTypeName Region</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657 -6369.602 Fibroblast  Fibroblasts Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658 -6374.026 Fibroblast  Fibroblasts Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660 -6372.266 Fibroblast  Fibroblasts Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661 -6374.851 Fibroblast  Fibroblasts Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3664 -6383.428 Fibroblast  Fibroblasts Stroma</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3665 -6385.541 Fibroblast  Fibroblasts Stroma</span></span>
<span><span class="co">##                                  Dist2Interface          InitSE Sample  SE</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657      -894.8463 SKCM..InitSE173   SKCM SE2</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658      -904.1115 SKCM..InitSE173   SKCM SE2</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660      -907.8909 SKCM..InitSE173   SKCM SE2</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661      -874.2712 SKCM..InitSE145   SKCM SE3</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3664      -880.4758  SKCM..InitSE19   SKCM SE3</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3665      -916.7790 SKCM..InitSE173   SKCM SE2</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="visualizing-ses">Visualizing SEs<a class="anchor" aria-label="anchor" href="#visualizing-ses"></a>
</h2>
<div class="section level3">
<h3 id="visualizing-the-integrated-similarity-matrix">Visualizing the integrated similarity matrix<a class="anchor" aria-label="anchor" href="#visualizing-the-integrated-similarity-matrix"></a>
</h3>
<p>After reordering spatial clusters based on the newly identified SEs,
you can re-draw the heatmap to show the fused similarity matrix of
spatial clusters across samples.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ords</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ses</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">ses</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">integrated</span> <span class="op">=</span> <span class="va">integrated</span><span class="op">[</span><span class="va">ords</span>, <span class="va">ords</span><span class="op">]</span> <span class="co"># reorder the cells</span></span>
<span><span class="va">ann</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\.\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">integrated</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  SE <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"SE"</span>, <span class="va">ses</span><span class="op">[</span><span class="va">ords</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                  row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">integrated</span><span class="op">)</span><span class="op">)</span> <span class="co"># cell groups</span></span>
<span><span class="va">SE_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getColors.html">getColors</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">ann</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span><span class="op">)</span>, palette <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="co"># colors for SEs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">SE_cols</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">ann</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span></span>
<span><span class="va">sample_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getColors.html">getColors</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">ann</span><span class="op">$</span><span class="va">Sample</span><span class="op">)</span><span class="op">)</span>, palette <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="co"># colors for samples</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sample_cols</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">ann</span><span class="op">$</span><span class="va">Sample</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## draw heatmap</span></span>
<span><span class="fu"><a href="../reference/HeatmapView.html">HeatmapView</a></span><span class="op">(</span><span class="va">integrated</span>, show_row_names <span class="op">=</span> <span class="cn">FALSE</span>, show_column_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>            top_ann <span class="op">=</span> <span class="va">ann</span>, top_ann_col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="va">sample_cols</span>, SE <span class="op">=</span> <span class="va">SE_cols</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/drawRectangleAnnotation.html">drawRectangleAnnotation</a></span><span class="op">(</span><span class="va">ann</span><span class="op">$</span><span class="va">SE</span>, <span class="va">ann</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_files/figure-html/newsehmap-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="visualizing-ses-in-the-tissue">Visualizing SEs in the tissue<a class="anchor" aria-label="anchor" href="#visualizing-ses-in-the-tissue"></a>
</h3>
<p>To examine the spatial distribution of the identified SEs, you can
use the <code>SpatialView</code> function visualize the SEs in the
tissues.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">finalmeta</span>, by <span class="op">=</span> <span class="st">"SE"</span><span class="op">)</span>  <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Sample</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_files/figure-html/spatialmapse-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="association-between-ses-and-pre-annotated-regions">Association between SEs and pre-annotated regions<a class="anchor" aria-label="anchor" href="#association-between-ses-and-pre-annotated-regions"></a>
</h3>
<p>To examine the enrichment of SEs in pre-annotated regions, you can
visualize the cell fractions from each region, for each SE.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">finalmeta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">Region</span>, <span class="va">Sample</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Sample</span>, <span class="va">SE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Frac <span class="op">=</span> <span class="va">n</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co">## cell type fractions within each sample</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">Region</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>Frac <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">Frac</span><span class="op">)</span><span class="op">)</span> <span class="co">## average cell type fractions across all samples</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">gg</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">Frac</span>, fill <span class="op">=</span> <span class="va">Region</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span>, palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"Fraction"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_files/figure-html/bar_region1-1.png" width="700"></p>
<details><summary><strong>Association between SEs and pre-annotated regions within each
sample</strong>
</summary><div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">finalmeta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">Region</span>, <span class="va">Sample</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">gg</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">n</span>, fill <span class="op">=</span> <span class="va">Region</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span>, palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Sample</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"Fraction of regions"</span><span class="op">)</span></span></code></pre></div>
<img src="Integration_files/figure-html/bar_region2-1.png" width="700"></details>
</div>
<div class="section level3">
<h3 id="distance-of-ses-to-tumorstroma-interface">Distance of SEs to tumor/stroma interface<a class="anchor" aria-label="anchor" href="#distance-of-ses-to-tumorstroma-interface"></a>
</h3>
<p>The distance of SEs to tumor/stroma interface can be visualized in a
box plot.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">finalmeta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Sample</span>, <span class="va">SE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>Dist2Interface <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">Dist2Interface</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Dist2Interface</span><span class="op">)</span></span>
<span><span class="va">gg</span><span class="op">$</span><span class="va">SE</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">gg</span><span class="op">$</span><span class="va">SE</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">gg</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">gg</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">Dist2Interface</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>color <span class="op">=</span> <span class="va">Sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"Distance to tumor/stroma interface (μm)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_files/figure-html/box_dist-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="cell-type-composition-of-ses">Cell type composition of SEs<a class="anchor" aria-label="anchor" href="#cell-type-composition-of-ses"></a>
</h3>
<p>To understand the cellular makeup of each SE, plot the average cell
type composition.</p>
<p><strong>Average cell type composition of SEs across all
samples</strong></p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">finalmeta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">CellType</span>, <span class="va">Sample</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Sample</span>, <span class="va">SE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Frac <span class="op">=</span> <span class="va">n</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="co">## cell type fractions within each sample</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">CellType</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>Frac <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">Frac</span><span class="op">)</span><span class="op">)</span> <span class="co">## average cell type fractions across all samples</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">gg</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">Frac</span>, fill <span class="op">=</span> <span class="va">CellType</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/discrete.html" class="external-link">kelly</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"Cell type abundance"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_files/figure-html/bar_ct2-1.png" width="700"></p>
<details><summary><strong>Cell type composition of SEs within each sample</strong>
</summary><div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">finalmeta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">CellType</span>, <span class="va">Sample</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">gg</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">n</span>, fill <span class="op">=</span> <span class="va">CellType</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/discrete.html" class="external-link">kelly</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Sample</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"Cell type abundance"</span><span class="op">)</span></span></code></pre></div>
<img src="Integration_files/figure-html/bar_ct1-1.png" width="700"></details>
</div>
</div>
<div class="section level2">
<h2 id="identification-of-cell-type-specific-se-markers">Identification of cell-type-specific SE markers<a class="anchor" aria-label="anchor" href="#identification-of-cell-type-specific-se-markers"></a>
</h2>
<p>You can identify cell-type-specific SE markers by differential
expression analysis using the <a href="https://github.com/immunogenomics/presto" class="external-link"><code>presto</code>
package</a>.</p>
<div class="section level3">
<h3 id="de-analysis-within-each-sample">DE analysis within each sample<a class="anchor" aria-label="anchor" href="#de-analysis-within-each-sample"></a>
</h3>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"presto"</span><span class="op">)</span></span>
<span><span class="co">## DE analysis within the first sample</span></span>
<span><span class="va">tmpmeta1</span> <span class="op">=</span> <span class="va">finalmeta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">CellType</span><span class="op">==</span><span class="st">"CD8T"</span> <span class="op">&amp;</span> <span class="va">Sample</span><span class="op">==</span><span class="st">"SKCM"</span> <span class="op">&amp;</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmpdata1</span> <span class="op">=</span> <span class="va">normdata1</span><span class="op">[</span>, <span class="va">tmpmeta1</span><span class="op">$</span><span class="va">CellID</span><span class="op">]</span></span>
<span><span class="va">degs1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/presto/man/wilcoxauc.html" class="external-link">wilcoxauc</a></span><span class="op">(</span><span class="va">tmpdata1</span>, <span class="va">tmpmeta1</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span></span>
<span><span class="co">## DE analysis within the second sample</span></span>
<span><span class="va">tmpmeta2</span> <span class="op">=</span> <span class="va">finalmeta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">CellType</span><span class="op">==</span><span class="st">"CD8T"</span> <span class="op">&amp;</span> <span class="va">Sample</span><span class="op">==</span><span class="st">"CRC"</span> <span class="op">&amp;</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmpdata2</span> <span class="op">=</span> <span class="va">normdata2</span><span class="op">[</span>, <span class="va">tmpmeta2</span><span class="op">$</span><span class="va">CellID</span><span class="op">]</span></span>
<span><span class="va">degs2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/presto/man/wilcoxauc.html" class="external-link">wilcoxauc</a></span><span class="op">(</span><span class="va">tmpdata2</span>, <span class="va">tmpmeta2</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span> <span class="co"># DE analysis</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">degs1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    feature group    avgExpr        logFC statistic       auc      pval padj</span></span>
<span><span class="co">## 1     PDK4   SE1 0.00000000  0.000000000  162228.0 0.5000000 1.0000000    1</span></span>
<span><span class="co">## 2 TNFRSF17   SE1 0.00000000  0.000000000  162228.0 0.5000000 1.0000000    1</span></span>
<span><span class="co">## 3    ICAM3   SE1 0.51370888  0.032194821  166290.0 0.5125194 0.6006048    1</span></span>
<span><span class="co">## 4      FAP   SE1 0.00000000  0.000000000  162228.0 0.5000000 1.0000000    1</span></span>
<span><span class="co">## 5     GZMB   SE1 0.19865862 -0.029811340  159553.5 0.4917570 0.6655314    1</span></span>
<span><span class="co">## 6     TSC2   SE1 0.01575335 -0.004433358  161227.5 0.4969164 0.6788433    1</span></span>
<span><span class="co">##      pct_in   pct_out</span></span>
<span><span class="co">## 1  0.000000  0.000000</span></span>
<span><span class="co">## 2  0.000000  0.000000</span></span>
<span><span class="co">## 3 50.757576 51.586656</span></span>
<span><span class="co">## 4  0.000000  0.000000</span></span>
<span><span class="co">## 5 21.969697 23.393002</span></span>
<span><span class="co">## 6  2.272727  2.888527</span></span></code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">degs2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    feature group   avgExpr       logFC statistic       auc       pval      padj</span></span>
<span><span class="co">## 1     PDK4   SE1 0.0000000  0.00000000  646892.5 0.5000000 1.00000000 1.0000000</span></span>
<span><span class="co">## 2   CX3CL1   SE1 0.0000000  0.00000000  646892.5 0.5000000 1.00000000 1.0000000</span></span>
<span><span class="co">## 3      CD4   SE1 0.2757405 -0.01459332  638083.5 0.4931913 0.53952676 1.0000000</span></span>
<span><span class="co">## 4    SNAI2   SE1 0.0000000  0.00000000  646892.5 0.5000000 1.00000000 1.0000000</span></span>
<span><span class="co">## 5 TNFRSF17   SE1 0.0000000  0.00000000  646892.5 0.5000000 1.00000000 1.0000000</span></span>
<span><span class="co">## 6    ICAM3   SE1 0.5182874  0.05711767  683122.0 0.5280027 0.02831882 0.1944559</span></span>
<span><span class="co">##     pct_in  pct_out</span></span>
<span><span class="co">## 1  0.00000  0.00000</span></span>
<span><span class="co">## 2  0.00000  0.00000</span></span>
<span><span class="co">## 3 28.27324 29.36864</span></span>
<span><span class="co">## 4  0.00000  0.00000</span></span>
<span><span class="co">## 5  0.00000  0.00000</span></span>
<span><span class="co">## 6 53.32068 48.10591</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="identifying-conserved-markers-across-samples">Identifying conserved markers across samples<a class="anchor" aria-label="anchor" href="#identifying-conserved-markers-across-samples"></a>
</h3>
<p>To identify markers conserved across different samples, we conduct a
meta-analysis by averaging the log2 fold changes (log2FC) from the DE
analyses.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="va">degs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">degs1</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">4</span><span class="op">)</span><span class="op">]</span>, <span class="va">degs2</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">4</span><span class="op">)</span><span class="op">]</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"feature"</span>, <span class="st">"group"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">degs</span><span class="op">$</span><span class="va">AvgLogFC</span> <span class="op">=</span> <span class="op">(</span><span class="va">degs</span><span class="op">$</span><span class="va">logFC.x</span> <span class="op">+</span> <span class="va">degs</span><span class="op">$</span><span class="va">logFC.y</span><span class="op">)</span> <span class="op">/</span> <span class="fl">2</span></span>
<span><span class="va">lfcs</span> <span class="op">=</span> <span class="va">degs</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>id_cols <span class="op">=</span> <span class="va">feature</span>, names_from <span class="op">=</span> <span class="va">group</span>, </span>
<span>                            values_from <span class="op">=</span> <span class="va">AvgLogFC</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">as.data.frame</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">lfcs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">lfcs</span><span class="op">$</span><span class="va">feature</span></span>
<span><span class="va">lfcs</span> <span class="op">&lt;-</span> <span class="va">lfcs</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lfcs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                  SE1         SE3          SE4         SE5          SE6</span></span>
<span><span class="co">## ACKR3    0.000000000  0.00000000  0.000000000  0.00000000  0.000000000</span></span>
<span><span class="co">## ACTA2    0.000000000  0.00000000  0.000000000  0.00000000  0.000000000</span></span>
<span><span class="co">## ADAMTS4  0.000000000  0.00000000  0.000000000  0.00000000  0.000000000</span></span>
<span><span class="co">## AKT1    -0.092686120 -0.10035625 -0.190867965  0.03809412  0.203712806</span></span>
<span><span class="co">## AKT2    -0.017319258  0.01809832  0.007465479 -0.01284803 -0.007846364</span></span>
<span><span class="co">## AKT3    -0.008900001  0.03862885  0.012651430 -0.02977051 -0.007425763</span></span></code></pre>
<p>To identify markers that are specific to each SE-associated cell
state, we calculate the difference between the maximum log2FC for each
gene and the second-highest log2FC:</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">secondmax</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">lfcs</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">-</span><span class="va">x</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">}</span><span class="op">)</span></span>
<span><span class="va">delta</span> <span class="op">=</span> <span class="va">lfcs</span> <span class="op">-</span> <span class="va">secondmax</span></span>
<span><span class="co">## Markers are considered specific if they have a positive delta and a log2FC greater than 0.05:</span></span>
<span><span class="va">idx</span> <span class="op">=</span> <span class="va">delta</span><span class="op">&gt;</span><span class="fl">0</span> <span class="op">&amp;</span> <span class="va">lfcs</span><span class="op">&gt;</span><span class="fl">0.05</span></span>
<span><span class="va">markers</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">lfcs</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">gs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">idx</span><span class="op">)</span><span class="op">[</span><span class="va">idx</span><span class="op">[</span>, <span class="va">se</span><span class="op">]</span><span class="op">]</span></span>
<span>  <span class="va">gs</span> <span class="op">=</span> <span class="va">gs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="va">lfcs</span><span class="op">[</span><span class="va">gs</span>, <span class="va">se</span><span class="op">]</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="va">gs</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">markers</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">lfcs</span><span class="op">)</span></span>
<span><span class="va">markers</span> <span class="op">=</span> <span class="va">markers</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lengths.html" class="external-link">lengths</a></span><span class="op">(</span><span class="va">markers</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">]</span></span>
<span><span class="co">## Select the top five markers</span></span>
<span><span class="va">top5</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">markers</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">}</span><span class="op">)</span></span>
<span><span class="va">top5</span></span></code></pre></div>
<pre><code><span><span class="co">## $SE1</span></span>
<span><span class="co">## [1] "TGFBR1" "FN1"    "IFNGR1" "FOS"    "COL1A1"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $SE3</span></span>
<span><span class="co">## [1] "EGR1"   "CCL5"   "ETS1"   "TGFBR2" "PTPRC" </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $SE4</span></span>
<span><span class="co">## [1] "CXCR4" "GZMK"  "CCR7"  "ZAP70" "KLRG1"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $SE5</span></span>
<span><span class="co">## [1] "BST2"   "STAT1"  "TAP2"   "NFKBIA" "IL2RB" </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $SE6</span></span>
<span><span class="co">## [1] "PKM"     "CDK2"    "GZMB"    "HAVCR2"  "TNFRSF9"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="visualizing-top-se-cell-state-markers">Visualizing top SE cell state markers<a class="anchor" aria-label="anchor" href="#visualizing-top-se-cell-state-markers"></a>
</h3>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/scale.html" class="external-link">scale</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">lfcs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">top5</span><span class="op">)</span>, <span class="op">]</span><span class="op">)</span>, center <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/HeatmapView.html">HeatmapView</a></span><span class="op">(</span><span class="va">gg</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">1.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="Integration_files/figure-html/hmaptop-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>The session info can be invaluable for users who encounter issues, as
it allows them to replicate the exact environment and identify potential
discrepancies in package versions or configurations that might be
causing problems.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur/Monterey 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] grid      parallel  stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [8] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] tidyr_1.3.0           presto_1.0.0          Rcpp_1.0.10          </span></span>
<span><span class="co">##  [4] pals_1.7              SpatialEcoTyper_0.0.4 RANN_2.6.1           </span></span>
<span><span class="co">##  [7] Matrix_1.5-3          ComplexHeatmap_2.12.1 NMF_0.27             </span></span>
<span><span class="co">## [10] Biobase_2.56.0        BiocGenerics_0.42.0   cluster_2.1.4        </span></span>
<span><span class="co">## [13] rngtools_1.5.2        registry_0.5-1        googledrive_2.1.1    </span></span>
<span><span class="co">## [16] data.table_1.14.6     SeuratObject_4.1.3    Seurat_4.3.0         </span></span>
<span><span class="co">## [19] ggplot2_3.4.0         dplyr_1.1.0          </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] circlize_0.4.15        systemfonts_1.0.4      plyr_1.8.8            </span></span>
<span><span class="co">##   [4] igraph_1.3.5           lazyeval_0.2.2         sp_1.6-0              </span></span>
<span><span class="co">##   [7] splines_4.2.0          listenv_0.9.0          scattermore_0.8       </span></span>
<span><span class="co">##  [10] gridBase_0.4-7         digest_0.6.31          foreach_1.5.2         </span></span>
<span><span class="co">##  [13] htmltools_0.5.8.1      magick_2.7.3           fansi_1.0.4           </span></span>
<span><span class="co">##  [16] magrittr_2.0.3         memoise_2.0.1          tensor_1.5            </span></span>
<span><span class="co">##  [19] doParallel_1.0.17      ROCR_1.0-11            globals_0.16.2        </span></span>
<span><span class="co">##  [22] matrixStats_0.63.0     R.utils_2.12.2         pkgdown_2.0.9         </span></span>
<span><span class="co">##  [25] spatstat.sparse_3.0-0  colorspace_2.1-0       ggrepel_0.9.2         </span></span>
<span><span class="co">##  [28] textshaping_0.3.7      xfun_0.37              crayon_1.5.2          </span></span>
<span><span class="co">##  [31] jsonlite_1.8.4         progressr_0.13.0       spatstat.data_3.0-0   </span></span>
<span><span class="co">##  [34] survival_3.5-0         zoo_1.8-11             iterators_1.0.14      </span></span>
<span><span class="co">##  [37] glue_1.6.2             polyclip_1.10-4        gtable_0.3.1          </span></span>
<span><span class="co">##  [40] gargle_1.5.1           leiden_0.4.3           GetoptLong_1.0.5      </span></span>
<span><span class="co">##  [43] shape_1.4.6            future.apply_1.10.0    maps_3.4.1            </span></span>
<span><span class="co">##  [46] abind_1.4-5            scales_1.2.1           DBI_1.1.3             </span></span>
<span><span class="co">##  [49] spatstat.random_3.1-3  miniUI_0.1.1.1         viridisLite_0.4.1     </span></span>
<span><span class="co">##  [52] xtable_1.8-4           clue_0.3-64            reticulate_1.28       </span></span>
<span><span class="co">##  [55] mapproj_1.2.11         stats4_4.2.0           htmlwidgets_1.6.1     </span></span>
<span><span class="co">##  [58] httr_1.4.7             RColorBrewer_1.1-3     ellipsis_0.3.2        </span></span>
<span><span class="co">##  [61] ica_1.0-3              farver_2.1.1           R.methodsS3_1.8.2     </span></span>
<span><span class="co">##  [64] pkgconfig_2.0.3        sass_0.4.9             uwot_0.1.14           </span></span>
<span><span class="co">##  [67] deldir_1.0-6           utf8_1.2.3             labeling_0.4.2        </span></span>
<span><span class="co">##  [70] tidyselect_1.2.0       rlang_1.1.1            reshape2_1.4.4        </span></span>
<span><span class="co">##  [73] later_1.3.0            munsell_0.5.0          tools_4.2.0           </span></span>
<span><span class="co">##  [76] cachem_1.0.6           cli_3.6.2              generics_0.1.3        </span></span>
<span><span class="co">##  [79] ggridges_0.5.4         evaluate_0.20          stringr_1.5.0         </span></span>
<span><span class="co">##  [82] fastmap_1.1.1          yaml_2.3.7             ragg_1.3.0            </span></span>
<span><span class="co">##  [85] goftest_1.2-3          knitr_1.42             fs_1.6.0              </span></span>
<span><span class="co">##  [88] fitdistrplus_1.1-8     purrr_1.0.1            pbapply_1.7-0         </span></span>
<span><span class="co">##  [91] future_1.30.0          nlme_3.1-157           mime_0.12             </span></span>
<span><span class="co">##  [94] R.oo_1.25.0            compiler_4.2.0         rstudioapi_0.14       </span></span>
<span><span class="co">##  [97] plotly_4.10.1          curl_5.2.1             png_0.1-8             </span></span>
<span><span class="co">## [100] spatstat.utils_3.0-1   tibble_3.1.8           bslib_0.7.0           </span></span>
<span><span class="co">## [103] stringi_1.7.12         highr_0.10             desc_1.4.2            </span></span>
<span><span class="co">## [106] lattice_0.20-45        vctrs_0.6.5            pillar_1.9.0          </span></span>
<span><span class="co">## [109] lifecycle_1.0.3        GlobalOptions_0.1.2    spatstat.geom_3.0-6   </span></span>
<span><span class="co">## [112] lmtest_0.9-40          jquerylib_0.1.4        RcppAnnoy_0.0.20      </span></span>
<span><span class="co">## [115] cowplot_1.1.1          irlba_2.3.5.1          httpuv_1.6.8          </span></span>
<span><span class="co">## [118] patchwork_1.1.2        R6_2.5.1               promises_1.2.0.1      </span></span>
<span><span class="co">## [121] KernSmooth_2.23-20     gridExtra_2.3          IRanges_2.30.1        </span></span>
<span><span class="co">## [124] parallelly_1.34.0      codetools_0.2-18       dichromat_2.0-0.1     </span></span>
<span><span class="co">## [127] MASS_7.3-58.2          rjson_0.2.21           rprojroot_2.0.3       </span></span>
<span><span class="co">## [130] withr_2.5.0            sctransform_0.3.5      S4Vectors_0.34.0      </span></span>
<span><span class="co">## [133] rmarkdown_2.20         Rtsne_0.16             spatstat.explore_3.0-6</span></span>
<span><span class="co">## [136] shiny_1.7.4</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://profiles.stanford.edu/wubing-zhang" class="external-link">Wubing Zhang</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
