<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="SpatialEcoTyper">
<title>Recovery of Spatial Ecotypes from Single-Cell Gene Expression Data • SpatialEcoTyper</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Recovery of Spatial Ecotypes from Single-Cell Gene Expression Data">
<meta property="og:description" content="SpatialEcoTyper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-default navbar-expand-lg bg-light" data-bs-theme="default"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpatialEcoTyper</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.0.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href="../articles/SingleSample.html">Discovery of Spatial Ecotypes from a Single Sample</a>
    <a class="dropdown-item" href="../articles/Integration.html">Discovery of Spatial Ecotypes from Multiple Samples</a>
    <a class="dropdown-item" href="../articles/TrainRecoveryModels.html">Development of NMF Models for Spatial Ecotype Recovery</a>
    <a class="dropdown-item" href="../articles/Recovery_Spatial.html">Recovery of Spatial Ecotypes from Spatial Transcriptomics Data</a>
    <a class="dropdown-item" href="../articles/Recovery_scRNA.html">Recovery of Spatial Ecotypes from Single-Cell Gene Expression Data</a>
    <a class="dropdown-item" href="../articles/Recovery_Bulk.html">Recovery of Spatial Ecotypes from Bulk Gene Expression Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">R functions</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Recovery of Spatial Ecotypes from Single-Cell Gene Expression Data</h1>
            
      
      <small class="dont-index">Source: <a href="https://digitalcytometry.github/SpatialEcoTypervignettes/Recovery_scRNA.Rmd" class="external-link"><code>vignettes/Recovery_scRNA.Rmd</code></a></small>
      <div class="d-none name"><code>Recovery_scRNA.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>In this tutorial, we will illustrate how to recover spatial ecotypes
(SEs) from single-cell RNA-seq data.</p>
<p><strong>First load required packages for this vignette</strong></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">parallel</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://googledrive.tidyverse.org" class="external-link">googledrive</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://digitalcytometry.github/SpatialEcoTyper" class="external-link">SpatialEcoTyper</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<p>We will illustrate the SE recovery using a melanoma sample, with raw
count matrix available in <a href="https://drive.google.com/open?id=15n9zlXed74oeGaO1pythOOM_iWIfuMn2&amp;usp=drive_fs" class="external-link">Melanoma_WU2161_counts.rds</a>
and single-cell meta data available in <a href="https://drive.google.com/open?id=15apmSDhcANjx3i05LIFIm01gelbP7etA&amp;usp=drive_fs" class="external-link">Melanoma_WU2161_scmeta.rds</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_deauth.html" class="external-link">drive_deauth</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Disable Google sign-in requirement</span></span>
<span><span class="co"># Download single-cell gene expression matrix and meta data.</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"15n9zlXed74oeGaO1pythOOM_iWIfuMn2"</span><span class="op">)</span>, <span class="st">"Melanoma_WU2161_counts.rds"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"15apmSDhcANjx3i05LIFIm01gelbP7etA"</span><span class="op">)</span>, <span class="st">"Melanoma_WU2161_scmeta.rds"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load single-cell gene expression matrix.</span></span>
<span><span class="va">scdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"Melanoma_WU2161_counts.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 27425  1337</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load single-cell metadata; cell type annotations are required</span></span>
<span><span class="va">scmeta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"Melanoma_WU2161_scmeta.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                   CID   CellType</span></span>
<span><span class="co">## AAACCTGCACATGACT-1 AAACCTGCACATGACT-1 Macrophage</span></span>
<span><span class="co">## AAACCTGGTGGTCCGT-1 AAACCTGGTGGTCCGT-1          B</span></span>
<span><span class="co">## AAACCTGGTTTGTGTG-1 AAACCTGGTTTGTGTG-1       CD4T</span></span>
<span><span class="co">## AAACCTGTCCGATATG-1 AAACCTGTCCGATATG-1 Macrophage</span></span>
<span><span class="co">## AAACCTGTCTAACGGT-1 AAACCTGTCTAACGGT-1          B</span></span>
<span><span class="co">## AAACGGGCACTTAAGC-1 AAACGGGCACTTAAGC-1       CD8T</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">$</span><span class="va">CellType</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Macrophage" "B"          "CD4T"       "CD8T"       "Neuron"    </span></span>
<span><span class="co">## [6] "Fibroblast" "Epithelial" "Plasma"</span></span></code></pre>
<p>Tab-delimited files can be efficiently loaded into R using the
<code>fread</code> function from the <code>data.table</code> package,
while sparse matrix files in the .mtx format can be imported using the
<code>ReadMtx</code> function from the <code>Seurat</code> package.</p>
</div>
<div class="section level2">
<h2 id="data-normalization">Data normalization<a class="anchor" aria-label="anchor" href="#data-normalization"></a>
</h2>
<p>The data can be normalized using <a href="https://satijalab.org/seurat/articles/seurat5_spatial_vignette_2" class="external-link">SCTransform</a>
or <a href="https://satijalab.org/seurat/reference/normalizedata" class="external-link">NormalizeData</a>.</p>
<p>Here, we are normalizing using SCTransform normalization. We
recommend to install the glmGamPoi package for faster computation.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"glmGamPoi"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"glmGamPoi"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">tmpobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SCTransform.html" class="external-link">SCTransform</a></span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_version</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html" class="external-link">packageVersion</a></span><span class="op">(</span><span class="st">"SeuratObject"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="va">seurat_version</span><span class="op">&lt;</span><span class="fl">5</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">normdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">tmpobj</span>, <span class="st">"data"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="kw">else</span><span class="op">{</span></span>
<span>  <span class="va">normdata</span> <span class="op">&lt;-</span> <span class="va">tmpobj</span><span class="op">[[</span><span class="st">"SCT"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">data</span></span>
<span><span class="op">}</span></span></code></pre></div>
<details><summary><strong>To use <code>NormalizeData</code> for the
normalization:</strong>
</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span></span></code></pre></div>
</details>
</div>
<div class="section level2">
<h2 id="se-recovery">SE recovery<a class="anchor" aria-label="anchor" href="#se-recovery"></a>
</h2>
<p>The <code>RecoverSE</code> function will be used to assign single
cells into SEs. Users can either use the default models to recover
predefined SEs or use custom models to recover newly defined SEs.</p>
<p><strong>Note</strong>: When using <code>RecoverSE</code> with
single-cell RNA-seq data, it is essential to specify the
<code>celltypes</code> parameter. If cell type annotation is not
provided, the function will assume that the input data corresponds to
spot-level spatial transcriptomics, and SE abundances will be inferred
directly from each spot.</p>
<div class="section level3">
<h3 id="using-default-models">Using default models<a class="anchor" aria-label="anchor" href="#using-default-models"></a>
</h3>
<p>The default NMF models were trained on discovery MERSCOPE data,
encompassing five cancer types: melanoma, and four carcinomas. These
models are tailored to nine distinct cell types: B cells, CD4+ T cells,
CD8+ T cells, NK cells, plasma cells, macrophages, dendritic cells,
fibroblasts, and endothelial cells. Each model facilitates the recovery
of SEs from single-cell datasets, allowing for cell-type-specific SE
analysis.</p>
<p>For SE recovery, the cells in the query data should be grouped into
one of “B”, “CD4T”, “CD8T”, “NK”, “Plasma”, “Macrophage”, “DC”,
“Fibroblast”, and “Endothelial”, case sensitive. All the other cell
types will be considered non-SE compartments.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sepreds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RecoverSE.html">RecoverSE</a></span><span class="op">(</span><span class="va">normdata</span>, celltypes <span class="op">=</span> <span class="va">scmeta</span><span class="op">$</span><span class="va">CellType</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sepreds</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## AAACCTGCACATGACT-1 AAACCTGGTGGTCCGT-1 AAACCTGGTTTGTGTG-1 AAACCTGTCCGATATG-1 </span></span>
<span><span class="co">##             "SE10"            "NonSE"             "SE01"             "SE10" </span></span>
<span><span class="co">## AAACCTGTCTAACGGT-1 AAACGGGCACTTAAGC-1 </span></span>
<span><span class="co">##            "NonSE"             "SE03"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="using-custom-models">Using custom models<a class="anchor" aria-label="anchor" href="#using-custom-models"></a>
</h3>
<p>To use custom models, users should first develop recovery models
following the tutorial <a href="TrainRecoveryModels.html">Development of
SE Recovery Models</a>. The resulting models can be used for SE
recovery. An example model is available at <a href="https://drive.google.com/open?id=1DJbiPeAHm7-4hSIouaeMMyPBWJUyusK9&amp;usp=drive_fs" class="external-link">SE_Recovery_W_list.rds</a>.</p>
<p>Download the example models</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1DJbiPeAHm7-4hSIouaeMMyPBWJUyusK9"</span><span class="op">)</span>, <span class="st">"SE_Recovery_W_list.rds"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Load the custom models</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Ws</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"SE_Recovery_W_list.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">Ws</span><span class="op">)</span> <span class="co">## named list of W matrices</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "CD4T"        "CD8T"        "DC"          "Endothelial" "Fibroblast" </span></span>
<span><span class="co">## [6] "Macrophage"  "NK"          "Plasma"</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">Ws</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="co">## feature by SE matrix</span></span></code></pre></div>
<pre><code><span><span class="co">##                  SE01      SE02      SE03      SE04      SE05      SE06</span></span>
<span><span class="co">## CD226__pos  0.4145936 0.2555579 0.3323735 0.3241147 0.2263783 0.3279133</span></span>
<span><span class="co">## CDKN1B__pos 0.5729663 0.4552065 0.5120732 0.4627805 0.4453117 0.5156023</span></span>
<span><span class="co">## CXCR4__pos  0.5865790 0.2771225 0.4093821 0.3607076 0.3589891 0.3937211</span></span>
<span><span class="co">## DUSP1__pos  0.4233067 0.3462390 0.2269798 0.1616814 0.3265539 0.2562118</span></span>
<span><span class="co">## KLF2__pos   0.5889618 0.2314586 0.3867504 0.2295206 0.2998234 0.5173246</span></span>
<span><span class="co">## ICAM2__pos  0.4593230 0.2611751 0.4427930 0.3143084 0.3679498 0.4532637</span></span>
<span><span class="co">##                  SE07      SE08      SE09      SE10      SE11</span></span>
<span><span class="co">## CD226__pos  0.2189382 0.3417897 0.2563481 0.2360192 0.2815938</span></span>
<span><span class="co">## CDKN1B__pos 0.4414750 0.4747166 0.4414508 0.3922732 0.3821411</span></span>
<span><span class="co">## CXCR4__pos  0.3751669 0.4456410 0.3023204 0.2629156 0.2729461</span></span>
<span><span class="co">## DUSP1__pos  0.3682236 0.2211759 0.1713413 0.1699067 0.1121763</span></span>
<span><span class="co">## KLF2__pos   0.3570251 0.4327206 0.1732251 0.2086729 0.2118797</span></span>
<span><span class="co">## ICAM2__pos  0.2660441 0.5494585 0.2878338 0.3711269 0.2575091</span></span></code></pre>
<p>Using custom models for SE recovery by specifying the <code>Ws</code>
argument.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sepreds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RecoverSE.html">RecoverSE</a></span><span class="op">(</span><span class="va">normdata</span>, celltypes <span class="op">=</span> <span class="va">scmeta</span><span class="op">$</span><span class="va">CellType</span>, Ws <span class="op">=</span> <span class="va">Ws</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sepreds</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## AAACCTGCACATGACT-1 AAACCTGGTGGTCCGT-1 AAACCTGGTTTGTGTG-1 AAACCTGTCCGATATG-1 </span></span>
<span><span class="co">##             "SE10"            "NonSE"             "SE01"             "SE10" </span></span>
<span><span class="co">## AAACCTGTCTAACGGT-1 AAACGGGCACTTAAGC-1 </span></span>
<span><span class="co">##            "NonSE"             "SE03"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>The session info allows users to replicate the exact environment and
identify potential discrepancies in package versions or configurations
that might be causing problems.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.3 (2023-03-15)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin13.4.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS/LAPACK: /Users/wzhang39/miniconda3/envs/SpatialEcoTyper/lib/libopenblasp-r0.3.27.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] SpatialEcoTyper_0.0.4 NMF_0.28              Biobase_2.58.0       </span></span>
<span><span class="co">##  [4] BiocGenerics_0.44.0   cluster_2.1.6         rngtools_1.5.2       </span></span>
<span><span class="co">##  [7] registry_0.5-1        RANN_2.6.2            Matrix_1.6-5         </span></span>
<span><span class="co">## [10] googledrive_2.1.1     data.table_1.15.4     SeuratObject_5.0.2   </span></span>
<span><span class="co">## [13] Seurat_4.4.0          ggplot2_3.5.1         dplyr_1.1.4          </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] circlize_0.4.16        spatstat.univar_3.0-0  spam_2.10-0           </span></span>
<span><span class="co">##   [4] systemfonts_1.1.0      plyr_1.8.9             igraph_2.0.3          </span></span>
<span><span class="co">##   [7] lazyeval_0.2.2         sp_2.1-4               splines_4.2.3         </span></span>
<span><span class="co">##  [10] listenv_0.9.1          scattermore_1.2        gridBase_0.4-7        </span></span>
<span><span class="co">##  [13] digest_0.6.36          foreach_1.5.2          htmltools_0.5.8.1     </span></span>
<span><span class="co">##  [16] fansi_1.0.6            magrittr_2.0.3         memoise_2.0.1         </span></span>
<span><span class="co">##  [19] tensor_1.5             doParallel_1.0.17      ROCR_1.0-11           </span></span>
<span><span class="co">##  [22] ComplexHeatmap_2.14.0  globals_0.16.3         matrixStats_1.3.0     </span></span>
<span><span class="co">##  [25] pkgdown_2.0.9          spatstat.sparse_3.1-0  colorspace_2.1-1      </span></span>
<span><span class="co">##  [28] ggrepel_0.9.5          textshaping_0.4.0      xfun_0.45             </span></span>
<span><span class="co">##  [31] crayon_1.5.3           jsonlite_1.8.8         progressr_0.14.0      </span></span>
<span><span class="co">##  [34] spatstat.data_3.1-2    survival_3.7-0         zoo_1.8-12            </span></span>
<span><span class="co">##  [37] iterators_1.0.14       glue_1.7.0             polyclip_1.10-7       </span></span>
<span><span class="co">##  [40] gtable_0.3.5           gargle_1.5.2           leiden_0.4.3.1        </span></span>
<span><span class="co">##  [43] GetoptLong_1.0.5       shape_1.4.6.1          future.apply_1.11.2   </span></span>
<span><span class="co">##  [46] abind_1.4-5            scales_1.3.0           spatstat.random_3.3-1 </span></span>
<span><span class="co">##  [49] miniUI_0.1.1.1         Rcpp_1.0.12            viridisLite_0.4.2     </span></span>
<span><span class="co">##  [52] xtable_1.8-4           clue_0.3-65            reticulate_1.38.0     </span></span>
<span><span class="co">##  [55] dotCall64_1.1-1        stats4_4.2.3           htmlwidgets_1.6.4     </span></span>
<span><span class="co">##  [58] httr_1.4.7             RColorBrewer_1.1-3     ica_1.0-3             </span></span>
<span><span class="co">##  [61] pkgconfig_2.0.3        sass_0.4.9             uwot_0.2.2            </span></span>
<span><span class="co">##  [64] deldir_2.0-4           utf8_1.2.4             tidyselect_1.2.1      </span></span>
<span><span class="co">##  [67] rlang_1.1.4            reshape2_1.4.4         later_1.3.2           </span></span>
<span><span class="co">##  [70] munsell_0.5.1          tools_4.2.3            cachem_1.1.0          </span></span>
<span><span class="co">##  [73] cli_3.6.3              generics_0.1.3         ggridges_0.5.6        </span></span>
<span><span class="co">##  [76] evaluate_0.24.0        stringr_1.5.1          fastmap_1.2.0         </span></span>
<span><span class="co">##  [79] yaml_2.3.8             ragg_1.3.2             goftest_1.2-3         </span></span>
<span><span class="co">##  [82] knitr_1.47             fs_1.6.4               fitdistrplus_1.2-1    </span></span>
<span><span class="co">##  [85] purrr_1.0.2            pbapply_1.7-2          future_1.34.0         </span></span>
<span><span class="co">##  [88] nlme_3.1-165           mime_0.12              compiler_4.2.3        </span></span>
<span><span class="co">##  [91] rstudioapi_0.16.0      plotly_4.10.4          curl_5.1.0            </span></span>
<span><span class="co">##  [94] png_0.1-8              spatstat.utils_3.1-0   tibble_3.2.1          </span></span>
<span><span class="co">##  [97] bslib_0.7.0            stringi_1.8.4          desc_1.4.3            </span></span>
<span><span class="co">## [100] lattice_0.22-6         vctrs_0.6.5            pillar_1.9.0          </span></span>
<span><span class="co">## [103] lifecycle_1.0.4        BiocManager_1.30.24    GlobalOptions_0.1.2   </span></span>
<span><span class="co">## [106] spatstat.geom_3.3-2    lmtest_0.9-40          jquerylib_0.1.4       </span></span>
<span><span class="co">## [109] RcppAnnoy_0.0.22       cowplot_1.1.3          irlba_2.3.5.1         </span></span>
<span><span class="co">## [112] httpuv_1.6.15          patchwork_1.2.0        R6_2.5.1              </span></span>
<span><span class="co">## [115] promises_1.3.0         KernSmooth_2.23-24     gridExtra_2.3         </span></span>
<span><span class="co">## [118] IRanges_2.32.0         parallelly_1.38.0      codetools_0.2-20      </span></span>
<span><span class="co">## [121] MASS_7.3-60.0.1        rjson_0.2.21           withr_3.0.0           </span></span>
<span><span class="co">## [124] sctransform_0.4.1      S4Vectors_0.36.2       grid_4.2.3            </span></span>
<span><span class="co">## [127] tidyr_1.3.1            rmarkdown_2.27         Rtsne_0.17            </span></span>
<span><span class="co">## [130] spatstat.explore_3.3-2 shiny_1.8.1.1</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://profiles.stanford.edu/wubing-zhang" class="external-link">Wubing Zhang</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
