<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="SpatialEcoTyper">
<title>Recovery of Spatial Ecotypes from Spatial Transcriptomics Data • SpatialEcoTyper</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Recovery of Spatial Ecotypes from Spatial Transcriptomics Data">
<meta property="og:description" content="SpatialEcoTyper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-default navbar-expand-lg bg-light" data-bs-theme="default"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpatialEcoTyper</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.0.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href="../articles/SingleSample.html">Discovery of Spatial Ecotypes from a Single Sample</a>
    <a class="dropdown-item" href="../articles/Integration.html">Discovery of Spatial Ecotypes from Multiple Samples</a>
    <a class="dropdown-item" href="../articles/TrainRecoveryModels.html">Development of NMF Models for Spatial Ecotype Recovery</a>
    <a class="dropdown-item" href="../articles/Recovery_Spatial.html">Recovery of Spatial Ecotypes from Spatial Transcriptomics Data</a>
    <a class="dropdown-item" href="../articles/Recovery_scRNA.html">Recovery of Spatial Ecotypes from Single-Cell Gene Expression Data</a>
    <a class="dropdown-item" href="../articles/Recovery_Bulk.html">Recovery of Spatial Ecotypes from Bulk Gene Expression Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">R functions</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Recovery of Spatial Ecotypes from Spatial Transcriptomics Data</h1>
            
      
      <small class="dont-index">Source: <a href="https://digitalcytometry.github/SpatialEcoTypervignettes/Recovery_Spatial.Rmd" class="external-link"><code>vignettes/Recovery_Spatial.Rmd</code></a></small>
      <div class="d-none name"><code>Recovery_Spatial.Rmd</code></div>
    </div>

    
    
<p><strong>First load required packages for this vignette</strong></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://googledrive.tidyverse.org" class="external-link">googledrive</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://digitalcytometry.github/SpatialEcoTyper" class="external-link">SpatialEcoTyper</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="se-recovery-from-single-cell-spatial-data">SE recovery from single-cell spatial data<a class="anchor" aria-label="anchor" href="#se-recovery-from-single-cell-spatial-data"></a>
</h2>
<p>In this tutorial, we will demonstrate how to recover spatial ecotypes
(SEs) from single-cell spatial transcriptomics data profiled by
platforms such as MERSCOPE, CosMx SMI, or Xenium.</p>
<div class="section level3">
<h3 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h3>
<p>We will use a subset of a melanoma MERSCOPE sample to illustrate the
SE recovery process. The expression data and single-cell metadata can be
downloaded from the <a href="https://drive.google.com/open?id=17K5XR7VEvxN9walV4RtaLOR2cBRG-aMh&amp;usp=drive_fs" class="external-link">Google
Drive</a>.</p>
<p>First, download the demo data from Google Drive</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_deauth.html" class="external-link">drive_deauth</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Disable Google sign-in requirement</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"13Rc5Rsu8jbnEYYfUse-xQ7ges51LcI7n"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_counts.tsv.gz"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"12xcZNhpT-xbhcG8kX1QAdTeM9TKeFAUW"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_scmeta.tsv"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Then load the gene expression matrix and meta data into R.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load single-cell gene expression matrix. Rows are genes, columns are cells.</span></span>
<span><span class="va">scdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">"HumanMelanomaPatient1_subset_counts.tsv.gz"</span>, </span>
<span>                sep <span class="op">=</span> <span class="st">"\t"</span>,header <span class="op">=</span> <span class="cn">TRUE</span>, data.table <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">scdata</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">scdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          HumanMelanomaPatient1__cell_3655 HumanMelanomaPatient1__cell_3657</span></span>
<span><span class="co">## PDK4                                    0                                1</span></span>
<span><span class="co">## TNFRSF17                                0                                0</span></span>
<span><span class="co">## ICAM3                                   0                                0</span></span>
<span><span class="co">## FAP                                     1                                0</span></span>
<span><span class="co">## GZMB                                    0                                0</span></span>
<span><span class="co">## TSC2                                    0                                0</span></span>
<span><span class="co">##          HumanMelanomaPatient1__cell_3658 HumanMelanomaPatient1__cell_3660</span></span>
<span><span class="co">## PDK4                                    1                                0</span></span>
<span><span class="co">## TNFRSF17                                0                                0</span></span>
<span><span class="co">## ICAM3                                   0                                0</span></span>
<span><span class="co">## FAP                                     0                                0</span></span>
<span><span class="co">## GZMB                                    0                                0</span></span>
<span><span class="co">## TSC2                                    0                                0</span></span>
<span><span class="co">##          HumanMelanomaPatient1__cell_3661</span></span>
<span><span class="co">## PDK4                                    0</span></span>
<span><span class="co">## TNFRSF17                                0</span></span>
<span><span class="co">## ICAM3                                   0</span></span>
<span><span class="co">## FAP                                     0</span></span>
<span><span class="co">## GZMB                                    0</span></span>
<span><span class="co">## TSC2                                    0</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load single-cell metadata, with at least three columns, including X, Y, and CellType</span></span>
<span><span class="va">scmeta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"HumanMelanomaPatient1_subset_scmeta.tsv"</span>, </span>
<span>                      sep <span class="op">=</span> <span class="st">"\t"</span>,header <span class="op">=</span> <span class="cn">TRUE</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">scdata</span> <span class="op">=</span> <span class="va">scdata</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"X"</span>, <span class="st">"Y"</span>, <span class="st">"CellType"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                         X         Y   CellType</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3655 1894.706 -6367.766 Fibroblast</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657 1942.480 -6369.602 Fibroblast</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658 1963.007 -6374.026 Fibroblast</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660 1981.600 -6372.266 Fibroblast</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661 1742.939 -6374.851 Fibroblast</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3663 1921.683 -6383.309 Fibroblast</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="data-normalization">Data normalization<a class="anchor" aria-label="anchor" href="#data-normalization"></a>
</h3>
<p>The gene expression data should be normalized for the SE recovery.
The data can be normalized using <a href="https://satijalab.org/seurat/articles/seurat5_spatial_vignette_2" class="external-link">SCTransform</a>
or <a href="https://satijalab.org/seurat/reference/normalizedata" class="external-link">NormalizeData</a>.</p>
<p>Here, we are normalizing using SCTransform normalization. We
recommend to install the glmGamPoi package for faster computation.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"glmGamPoi"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"glmGamPoi"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">tmpobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://satijalab.org/seurat/reference/SCTransform.html" class="external-link">SCTransform</a></span><span class="op">(</span>clip.range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">10</span>, <span class="fl">10</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_version</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html" class="external-link">packageVersion</a></span><span class="op">(</span><span class="st">"SeuratObject"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="va">seurat_version</span><span class="op">&lt;</span><span class="fl">5</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">normdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">tmpobj</span>, <span class="st">"data"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="kw">else</span><span class="op">{</span></span>
<span>  <span class="va">normdata</span> <span class="op">&lt;-</span> <span class="va">tmpobj</span><span class="op">[[</span><span class="st">"SCT"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">data</span></span>
<span><span class="op">}</span></span></code></pre></div>
<summary>
Using <code>NormalizeData</code> for the normalization
</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span></span></code></pre></div>
<details><summary><strong>Preview of the sample</strong>
</summary><p>This example sample includes nine major cell types, including B
cells, CD4+ T cells, CD8+ T cells, NK cells, plasma cells, macrophages,
dendritic cells (DC), fibroblasts, and endothelial cells.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize the cell type annotations in the tissue</span></span>
<span><span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">scmeta</span>, by <span class="op">=</span> <span class="st">"CellType"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/discrete.html" class="external-link">kelly</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="Recovery_Spatial_files/figure-html/viewct-1.png" width="700"></p>
<p>In this example sample, we annotated four regions, including tumor,
inner margin, outer margin, and stroma. The tumor and stroma regions
were annotated based on a density of cancer cells. The inner margin and
outer margin were annotated as a certain width (250 μm) inside and
outside the tumor regions.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize the regions in the tissue</span></span>
<span><span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">scmeta</span>, by <span class="op">=</span> <span class="st">"Region"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span>, palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span></span></code></pre></div>
<img src="Recovery_Spatial_files/figure-html/viewregion-1.png" width="700"></details>
</div>
<div class="section level3">
<h3 id="se-recovery">SE recovery<a class="anchor" aria-label="anchor" href="#se-recovery"></a>
</h3>
<p>The <code>RecoverSE</code> function will be used to assign single
cells into SEs. Users can either use the default models to recover
predefined SEs or use custom models to recover newly defined SEs.</p>
<p><strong>Note</strong>: To recover SEs from single-cell data, you must
specify either <code>celltypes</code> or <code>se_results</code> in the
<code>RecoverSE</code> function. If neither is provided, it will assume
that the input data represents spot-level spatial transcriptomics, and
SE abundances will be inferred directly from each spot.</p>
<p>The default NMF models were trained on discovery MERSCOPE data,
encompassing five cancer types: melanoma, and four carcinomas. These
models are tailored to nine distinct cell types: B cells, CD4+ T cells,
CD8+ T cells, NK cells, plasma cells, macrophages, dendritic cells,
fibroblasts, and endothelial cells. Each model facilitates the recovery
of SEs from single-cell datasets, allowing for cell-type-specific SE
analysis. Thus, for SE recovery using default models, the cells in the
query data should be grouped into one of “B”, “CD4T”, “CD8T”, “NK”,
“Plasma”, “Macrophage”, “DC”, “Fibroblast”, and “Endothelial”, case
sensitive. All the other cell types will be considered non-SE
compartments.</p>
<div class="section level4">
<h4 id="using-default-models-1">Using default models (1)<a class="anchor" aria-label="anchor" href="#using-default-models-1"></a>
</h4>
<p>Before SE recovery, we recommend to perform
<code>SpatialEcoTyper</code> analysis, which integrate gene expression
and spatial information to learn a unified embedding of spatial
microregions. The resulting embedding will help refine the SE recovery
results. The detailed tutorial for SpatialEcoTyper analysis is available
at <a href="SingleSample.html">Discovery of Spatial Ecotypes from a
Single-cell Spatial Dataset</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Group cells into spatial clusters with a high resolution </span></span>
<span><span class="co">## This step takes ~2 min to finish</span></span>
<span><span class="va">emb</span> <span class="op">=</span> <span class="fu"><a href="../reference/SpatialEcoTyper.html">SpatialEcoTyper</a></span><span class="op">(</span><span class="va">normdata</span>, <span class="va">scmeta</span>, resolution <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">emb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "list"</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">emb</span><span class="op">$</span><span class="va">obj</span> <span class="co">## A seurat object including the spatial embedding</span></span></code></pre></div>
<pre><code><span><span class="co">## An object of class Seurat </span></span>
<span><span class="co">## 1633 features across 1633 samples within 1 assay </span></span>
<span><span class="co">## Active assay: RNA (1633 features, 1633 variable features)</span></span>
<span><span class="co">##  2 dimensional reductions calculated: pca, umap</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">emb</span><span class="op">$</span><span class="va">metadata</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"X"</span>, <span class="st">"Y"</span>, <span class="st">"CellType"</span>, <span class="st">"SE"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>  <span class="co">## Single-cell meta data with SE annotations</span></span></code></pre></div>
<pre><code><span><span class="co">##                                         X         Y   CellType   SE</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3655 1894.706 -6367.766 Fibroblast &lt;NA&gt;</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657 1942.480 -6369.602 Fibroblast  SE9</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658 1963.007 -6374.026 Fibroblast  SE9</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660 1981.600 -6372.266 Fibroblast  SE9</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661 1742.939 -6374.851 Fibroblast SE46</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3663 1921.683 -6383.309 Fibroblast &lt;NA&gt;</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">emb</span><span class="op">$</span><span class="va">metadata</span><span class="op">$</span><span class="va">CellType</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Fibroblast"  "Endothelial" "DC"          "Macrophage"  "CD8T"       </span></span>
<span><span class="co">## [6] "CD4T"        "Plasma"      "NK"          "B"</span></span></code></pre>
<p>Then specify the <code>se_results</code> in the
<code>RecoverSE</code> function for SE recovery.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sepreds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RecoverSE.html">RecoverSE</a></span><span class="op">(</span><span class="va">normdata</span>, se_results <span class="op">=</span> <span class="va">emb</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sepreds</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## HumanMelanomaPatient1__cell_3655 HumanMelanomaPatient1__cell_3657 </span></span>
<span><span class="co">##                          "NonSE"                           "SE11" </span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658 HumanMelanomaPatient1__cell_3660 </span></span>
<span><span class="co">##                           "SE11"                           "SE11" </span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661 HumanMelanomaPatient1__cell_3663 </span></span>
<span><span class="co">##                           "SE01"                          "NonSE"</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="using-default-models-2">Using default models (2)<a class="anchor" aria-label="anchor" href="#using-default-models-2"></a>
</h4>
<p>You can also recover the SEs without using spatial embedding, which
could be less accurate due to the lack of spatial information. The cell
type annotations are required in this case. Cells should be grouped into
one of “B”, “CD4T”, “CD8T”, “NK”, “Plasma”, “Macrophage”, “DC”,
“Fibroblast”, and “Endothelial”, case sensitive. All the other cell
types will be considered non-SE compartments.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">$</span><span class="va">CellType</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sepreds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RecoverSE.html">RecoverSE</a></span><span class="op">(</span><span class="va">normdata</span>, celltypes <span class="op">=</span> <span class="va">scmeta</span><span class="op">$</span><span class="va">CellType</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sepreds</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="using-custom-models">Using custom models<a class="anchor" aria-label="anchor" href="#using-custom-models"></a>
</h4>
<p>To use custom models, users should first develop recovery models
following the tutorial <a href="TrainRecoveryModels.html">Development of
SE Recovery Models</a>. The resulting models can be used for SE
recovery. An example model is available at <a href="https://drive.google.com/open?id=1DJbiPeAHm7-4hSIouaeMMyPBWJUyusK9&amp;usp=drive_fs" class="external-link">SE_Recovery_W_list.rds</a>.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Download SE recovery model</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1DJbiPeAHm7-4hSIouaeMMyPBWJUyusK9"</span><span class="op">)</span>, <span class="st">"SE_Recovery_W_list.rds"</span><span class="op">)</span></span>
<span><span class="va">Ws</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"SE_Recovery_W_list.rds"</span><span class="op">)</span></span></code></pre></div>
<p>Using custom models by specifying the Ws argument:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sepreds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RecoverSE.html">RecoverSE</a></span><span class="op">(</span><span class="va">normdata</span>, se_results <span class="op">=</span> <span class="va">emb</span>, Ws <span class="op">=</span> <span class="va">Ws</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sepreds</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="visualization-of-ses-in-the-tissue">Visualization of SEs in the tissue<a class="anchor" aria-label="anchor" href="#visualization-of-ses-in-the-tissue"></a>
</h3>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Add the recovery result into the meta data</span></span>
<span><span class="va">scmeta</span><span class="op">$</span><span class="va">RecoveredSE</span> <span class="op">&lt;-</span> <span class="va">sepreds</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">## Visualize the SE recovery results</span></span>
<span><span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">scmeta</span>, by <span class="op">=</span> <span class="st">"RecoveredSE"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Recovery_Spatial_files/figure-html/visualscst-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="recovery-of-spatial-ecotypes-from-visium-spatial-gene-expression-data">Recovery of spatial ecotypes from Visium Spatial Gene Expression
data<a class="anchor" aria-label="anchor" href="#recovery-of-spatial-ecotypes-from-visium-spatial-gene-expression-data"></a>
</h2>
<p>SEs can be recovered from Visium spatial transcriptomics data by
inferring SE abundances from each spot. Each spot can then be assigned
to the SE with the highest inferred abundance, allowing for the spatial
mapping of SEs across the tissue.</p>
<div class="section level3">
<h3 id="loading-data-1">Loading data<a class="anchor" aria-label="anchor" href="#loading-data-1"></a>
</h3>
<p>In this tutorial, we will demonstrate how to recover SEs using a
breast cancer sample. The expression data can be accessed from: <a href="https://drive.google.com/open?id=15D9LgvZmCZUfsL62cD67JMf8Jcq_UyuB&amp;usp=drive_fs" class="external-link">V1_Breast_Cancer_Block_A_Section_1_filtered_feature_bc_matrix.h5</a>,
which was downloaded from <a href="https://www.10xgenomics.com/datasets" class="external-link uri">https://www.10xgenomics.com/datasets</a>.</p>
<p>First, download the data from Google Drive</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"15D9LgvZmCZUfsL62cD67JMf8Jcq_UyuB"</span><span class="op">)</span>, <span class="st">"V1_Breast_Cancer_Block_A_Section_1_filtered_feature_bc_matrix.h5"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"15NTZc1HrW_gLS_pmi1ckYLw30kNarf4w"</span><span class="op">)</span>, <span class="st">"V1_Breast_Cancer_Block_A_Section_1_tissue_positions_list.csv"</span><span class="op">)</span></span>
<span><span class="co">## This download should be done within 1 min.</span></span></code></pre></div>
<p>Load the expression data into R using the <code>Read10X_h5</code>
function from the Seurat package.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"hdf5r"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"hdf5r"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://hhoeflin.github.io/hdf5r/" class="external-link">"hdf5r"</a></span><span class="op">)</span></span>
<span><span class="co"># Load Visium gene expression matrix. Rows are genes, columns are spots.</span></span>
<span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Read10X_h5.html" class="external-link">Read10X_h5</a></span><span class="op">(</span><span class="st">"V1_Breast_Cancer_Block_A_Section_1_filtered_feature_bc_matrix.h5"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"V1_Breast_Cancer_Block_A_Section_1_tissue_positions_list.csv"</span>, </span>
<span>                 header <span class="op">=</span> <span class="cn">FALSE</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tissue"</span>, <span class="st">"row"</span>, <span class="st">"col"</span>, <span class="st">"imagerow"</span>, <span class="st">"imagecol"</span><span class="op">)</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">meta</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    tissue row col imagerow imagecol</span></span>
<span><span class="co">## AAACAAGTATCTCCCA-1      1  50 102    15632    17782</span></span>
<span><span class="co">## AAACACCAATAACTGC-1      1  59  19    17734     6447</span></span>
<span><span class="co">## AAACAGAGCGACTCCT-1      1  14  94     7079    16716</span></span>
<span><span class="co">## AAACAGGGTCTATATT-1      1  47  13    14882     5637</span></span>
<span><span class="co">## AAACAGTGTTCCTGGG-1      1  73  43    21069     9712</span></span>
<span><span class="co">## AAACATTTCCCGGATT-1      1  61  97    18242    17091</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="data-normalization-1">Data normalization<a class="anchor" aria-label="anchor" href="#data-normalization-1"></a>
</h3>
<p>Here, we normalize the Visium data using <a href="https://satijalab.org/seurat/reference/normalizedata" class="external-link">NormalizeData</a>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="se-recovery-1">SE recovery<a class="anchor" aria-label="anchor" href="#se-recovery-1"></a>
</h3>
<p>If neither <code>celltypes</code> nor <code>se_results</code> are
specified, the <code>RecoverSE</code> function will assume the input
data is a gene-by-spot gene expression matrix. In this case, the
function will infer SE abundances across spots. Users have the option to
either use the default models to recover predefined SEs or apply custom
models to recover newly defined SEs.</p>
<div class="section level4">
<h4 id="using-default-models">Using default models<a class="anchor" aria-label="anchor" href="#using-default-models"></a>
</h4>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">preds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RecoverSE.html">RecoverSE</a></span><span class="op">(</span><span class="va">normdata</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">preds</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                            SE01         SE02         SE03         SE04</span></span>
<span><span class="co">## AAACAAGTATCTCCCA-1 2.127701e-04 6.877800e-05 3.979458e-01 8.534547e-02</span></span>
<span><span class="co">## AAACACCAATAACTGC-1 1.321616e-01 6.347672e-02 4.580604e-02 9.893653e-02</span></span>
<span><span class="co">## AAACAGAGCGACTCCT-1 1.658118e-01 2.438595e-15 1.257332e-01 1.021615e-01</span></span>
<span><span class="co">## AAACAGGGTCTATATT-1 6.495088e-02 1.975689e-16 2.208619e-01 6.972421e-10</span></span>
<span><span class="co">## AAACAGTGTTCCTGGG-1 9.104830e-02 2.073793e-01 2.579438e-09 6.489594e-02</span></span>
<span><span class="co">## AAACATTTCCCGGATT-1 6.684999e-13 2.987802e-01 2.908525e-08 7.202254e-07</span></span>
<span><span class="co">##                            SE05         SE06         SE07         SE08</span></span>
<span><span class="co">## AAACAAGTATCTCCCA-1 1.162860e-01 1.055567e-05 1.723712e-16 3.026670e-01</span></span>
<span><span class="co">## AAACACCAATAACTGC-1 4.542600e-07 5.479484e-02 4.511416e-01 3.746409e-16</span></span>
<span><span class="co">## AAACAGAGCGACTCCT-1 7.829883e-03 1.888903e-01 3.928577e-05 3.462729e-01</span></span>
<span><span class="co">## AAACAGGGTCTATATT-1 8.604433e-03 2.053562e-11 1.976922e-16 7.055828e-01</span></span>
<span><span class="co">## AAACAGTGTTCCTGGG-1 1.529680e-06 2.046686e-02 2.947065e-01 2.356171e-15</span></span>
<span><span class="co">## AAACATTTCCCGGATT-1 1.993179e-11 2.597762e-08 1.996192e-01 3.022784e-16</span></span>
<span><span class="co">##                            SE09         SE10         SE11</span></span>
<span><span class="co">## AAACAAGTATCTCCCA-1 5.912467e-02 2.078080e-02 1.755819e-02</span></span>
<span><span class="co">## AAACACCAATAACTGC-1 1.616557e-02 1.333849e-01 4.131729e-03</span></span>
<span><span class="co">## AAACAGAGCGACTCCT-1 4.615261e-02 2.635185e-04 1.684507e-02</span></span>
<span><span class="co">## AAACAGGGTCTATATT-1 9.278320e-15 1.739479e-09 1.975689e-16</span></span>
<span><span class="co">## AAACAGTGTTCCTGGG-1 1.582161e-01 1.632831e-01 2.351639e-06</span></span>
<span><span class="co">## AAACATTTCCCGGATT-1 1.697931e-01 2.576151e-01 7.419167e-02</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## This step would take ~3 minutes to complete</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="using-custom-models-1">Using custom models<a class="anchor" aria-label="anchor" href="#using-custom-models-1"></a>
</h4>
<p>To use custom models, users should first develop recovery models
following the tutorial <a href="TrainRecoveryModels.html">Developing SE
Recovery Models</a>. The resulting models can be used for SE
recovery.</p>
<p>An example model is available at <a href="https://drive.google.com/open?id=1DJbiPeAHm7-4hSIouaeMMyPBWJUyusK9&amp;usp=drive_fs" class="external-link">SE_Recovery_W_list.rds</a>.
By specifying the <code>Ws</code> parameter (a list of
cell-type-specific W matrices) in the <code>RecoverSE</code> function,
the custom models will be used for recovering SEs.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Download SE recovery model</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1DJbiPeAHm7-4hSIouaeMMyPBWJUyusK9"</span><span class="op">)</span>, <span class="st">"SE_Recovery_W_list.rds"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">Ws</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"SE_Recovery_W_list.rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Using custom models by specifying the Ws argument</span></span>
<span><span class="va">preds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RecoverSE.html">RecoverSE</a></span><span class="op">(</span><span class="va">normdata</span>, Ws <span class="op">=</span> <span class="va">Ws</span><span class="op">)</span></span>
<span><span class="co">## This step would take ~3 minutes to complete</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="visualization-of-ses-in-the-tissue-1">Visualization of SEs in the tissue<a class="anchor" aria-label="anchor" href="#visualization-of-ses-in-the-tissue-1"></a>
</h3>
<p>Each spot can be assigned to the SE with the highest inferred
abundance, and the spatial mapping of SEs can be visualized using the
<code>SpatialView</code> function.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta</span><span class="op">$</span><span class="va">RecoveredSE</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">preds</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">preds</span>, <span class="fl">1</span>, <span class="va">which.max</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">meta</span><span class="op">$</span><span class="va">Y</span> <span class="op">=</span> <span class="op">-</span><span class="va">meta</span><span class="op">$</span><span class="va">row</span></span>
<span><span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">meta</span>, by <span class="op">=</span> <span class="st">"RecoveredSE"</span>, X <span class="op">=</span> <span class="st">"col"</span>, Y <span class="op">=</span> <span class="st">"Y"</span>, pt.size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="Recovery_Spatial_files/figure-html/visualvisium-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>The session info can be invaluable for users who encounter issues, as
it allows them to replicate the exact environment and identify potential
discrepancies in package versions or configurations that might be
causing problems.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur/Monterey 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] hdf5r_1.3.8           SpatialEcoTyper_0.0.4 NMF_0.27             </span></span>
<span><span class="co">##  [4] Biobase_2.56.0        BiocGenerics_0.42.0   cluster_2.1.4        </span></span>
<span><span class="co">##  [7] rngtools_1.5.2        registry_0.5-1        RANN_2.6.1           </span></span>
<span><span class="co">## [10] Matrix_1.5-3          googledrive_2.1.1     data.table_1.14.6    </span></span>
<span><span class="co">## [13] SeuratObject_4.1.3    Seurat_4.3.0          ggplot2_3.4.0        </span></span>
<span><span class="co">## [16] dplyr_1.1.0          </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] circlize_0.4.15        systemfonts_1.0.4      plyr_1.8.8            </span></span>
<span><span class="co">##   [4] igraph_1.3.5           lazyeval_0.2.2         sp_1.6-0              </span></span>
<span><span class="co">##   [7] splines_4.2.0          listenv_0.9.0          scattermore_0.8       </span></span>
<span><span class="co">##  [10] gridBase_0.4-7         digest_0.6.31          foreach_1.5.2         </span></span>
<span><span class="co">##  [13] htmltools_0.5.8.1      fansi_1.0.4            magrittr_2.0.3        </span></span>
<span><span class="co">##  [16] memoise_2.0.1          tensor_1.5             doParallel_1.0.17     </span></span>
<span><span class="co">##  [19] ROCR_1.0-11            ComplexHeatmap_2.12.1  globals_0.16.2        </span></span>
<span><span class="co">##  [22] matrixStats_0.63.0     R.utils_2.12.2         pkgdown_2.0.9         </span></span>
<span><span class="co">##  [25] spatstat.sparse_3.0-0  colorspace_2.1-0       ggrepel_0.9.2         </span></span>
<span><span class="co">##  [28] textshaping_0.3.7      xfun_0.37              crayon_1.5.2          </span></span>
<span><span class="co">##  [31] jsonlite_1.8.4         progressr_0.13.0       spatstat.data_3.0-0   </span></span>
<span><span class="co">##  [34] survival_3.5-0         zoo_1.8-11             iterators_1.0.14      </span></span>
<span><span class="co">##  [37] glue_1.6.2             polyclip_1.10-4        pals_1.7              </span></span>
<span><span class="co">##  [40] gtable_0.3.1           gargle_1.5.1           leiden_0.4.3          </span></span>
<span><span class="co">##  [43] GetoptLong_1.0.5       shape_1.4.6            future.apply_1.10.0   </span></span>
<span><span class="co">##  [46] maps_3.4.1             abind_1.4-5            scales_1.2.1          </span></span>
<span><span class="co">##  [49] DBI_1.1.3              spatstat.random_3.1-3  miniUI_0.1.1.1        </span></span>
<span><span class="co">##  [52] Rcpp_1.0.10            viridisLite_0.4.1      xtable_1.8-4          </span></span>
<span><span class="co">##  [55] clue_0.3-64            reticulate_1.28        bit_4.0.5             </span></span>
<span><span class="co">##  [58] mapproj_1.2.11         stats4_4.2.0           htmlwidgets_1.6.1     </span></span>
<span><span class="co">##  [61] httr_1.4.7             RColorBrewer_1.1-3     ellipsis_0.3.2        </span></span>
<span><span class="co">##  [64] ica_1.0-3              farver_2.1.1           R.methodsS3_1.8.2     </span></span>
<span><span class="co">##  [67] pkgconfig_2.0.3        sass_0.4.9             uwot_0.1.14           </span></span>
<span><span class="co">##  [70] deldir_1.0-6           utf8_1.2.3             labeling_0.4.2        </span></span>
<span><span class="co">##  [73] tidyselect_1.2.0       rlang_1.1.1            reshape2_1.4.4        </span></span>
<span><span class="co">##  [76] later_1.3.0            munsell_0.5.0          tools_4.2.0           </span></span>
<span><span class="co">##  [79] cachem_1.0.6           cli_3.6.2              generics_0.1.3        </span></span>
<span><span class="co">##  [82] ggridges_0.5.4         evaluate_0.20          stringr_1.5.0         </span></span>
<span><span class="co">##  [85] fastmap_1.1.1          yaml_2.3.7             ragg_1.3.0            </span></span>
<span><span class="co">##  [88] goftest_1.2-3          bit64_4.0.5            knitr_1.42            </span></span>
<span><span class="co">##  [91] fs_1.6.0               fitdistrplus_1.1-8     purrr_1.0.1           </span></span>
<span><span class="co">##  [94] pbapply_1.7-0          future_1.30.0          nlme_3.1-157          </span></span>
<span><span class="co">##  [97] mime_0.12              R.oo_1.25.0            compiler_4.2.0        </span></span>
<span><span class="co">## [100] rstudioapi_0.14        plotly_4.10.1          curl_5.2.1            </span></span>
<span><span class="co">## [103] png_0.1-8              spatstat.utils_3.0-1   tibble_3.1.8          </span></span>
<span><span class="co">## [106] bslib_0.7.0            stringi_1.7.12         highr_0.10            </span></span>
<span><span class="co">## [109] desc_1.4.2             lattice_0.20-45        vctrs_0.6.5           </span></span>
<span><span class="co">## [112] pillar_1.9.0           lifecycle_1.0.3        GlobalOptions_0.1.2   </span></span>
<span><span class="co">## [115] spatstat.geom_3.0-6    lmtest_0.9-40          jquerylib_0.1.4       </span></span>
<span><span class="co">## [118] RcppAnnoy_0.0.20       cowplot_1.1.1          irlba_2.3.5.1         </span></span>
<span><span class="co">## [121] httpuv_1.6.8           patchwork_1.1.2        R6_2.5.1              </span></span>
<span><span class="co">## [124] promises_1.2.0.1       KernSmooth_2.23-20     gridExtra_2.3         </span></span>
<span><span class="co">## [127] IRanges_2.30.1         parallelly_1.34.0      codetools_0.2-18      </span></span>
<span><span class="co">## [130] dichromat_2.0-0.1      MASS_7.3-58.2          rjson_0.2.21          </span></span>
<span><span class="co">## [133] rprojroot_2.0.3        withr_2.5.0            sctransform_0.3.5     </span></span>
<span><span class="co">## [136] S4Vectors_0.34.0       grid_4.2.0             tidyr_1.3.0           </span></span>
<span><span class="co">## [139] rmarkdown_2.20         Rtsne_0.16             spatstat.explore_3.0-6</span></span>
<span><span class="co">## [142] shiny_1.7.4</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://profiles.stanford.edu/wubing-zhang" class="external-link">Wubing Zhang</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
