<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="SpatialEcoTyper">
<title>Recovery of Spatial Ecotypes from Bulk Gene Expression Data â€¢ SpatialEcoTyper</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Recovery of Spatial Ecotypes from Bulk Gene Expression Data">
<meta property="og:description" content="SpatialEcoTyper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-default navbar-expand-lg bg-light" data-bs-theme="default"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpatialEcoTyper</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.0.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href="../articles/SingleSample.html">Discovery of Spatial Ecotypes from a Single Sample</a>
    <a class="dropdown-item" href="../articles/Integration.html">Discovery of Spatial Ecotypes from Multiple Samples</a>
    <a class="dropdown-item" href="../articles/TrainRecoveryModels.html">Development of NMF Models for Spatial Ecotype Recovery</a>
    <a class="dropdown-item" href="../articles/Recovery_Spatial.html">Recovery of Spatial Ecotypes from Spatial Transcriptomics Data</a>
    <a class="dropdown-item" href="../articles/Recovery_scRNA.html">Recovery of Spatial Ecotypes from Single-Cell Gene Expression Data</a>
    <a class="dropdown-item" href="../articles/Recovery_Bulk.html">Recovery of Spatial Ecotypes from Bulk Gene Expression Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">R functions</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Recovery of Spatial Ecotypes from Bulk Gene Expression Data</h1>
            
      
      <small class="dont-index">Source: <a href="https://digitalcytometry.github/SpatialEcoTypervignettes/Recovery_Bulk.Rmd" class="external-link"><code>vignettes/Recovery_Bulk.Rmd</code></a></small>
      <div class="d-none name"><code>Recovery_Bulk.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>In this tutorial, we will demonstrate how to infer spatial ecotype
(SE) abundances from bulk RNA-seq data. Specifically, we will infer SE
abundances from RNA-seq data of TCGA melanoma samples. The gene
expression matrix is available at <a href="https://drive.google.com/open?id=14QvmgISxaArTzWt_UHvf55aAYN2zm84Q&amp;usp=drive_fs" class="external-link">SKCM_RNASeqV2.geneExp.rds</a>,
which is a Transcripts Per Million (TPM) matrix obtained from the <a href="https://gdc.cancer.gov/about-data/publications/pancanatlas" class="external-link">PanCanAtlas</a>.</p>
<p><strong>First load required packages for this vignette</strong></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://googledrive.tidyverse.org" class="external-link">googledrive</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://digitalcytometry.github/SpatialEcoTyper" class="external-link">SpatialEcoTyper</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-bulk-expression-of-tcga-melanoma-samples">Loading bulk expression of TCGA melanoma samples<a class="anchor" aria-label="anchor" href="#loading-bulk-expression-of-tcga-melanoma-samples"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_deauth.html" class="external-link">drive_deauth</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Disable Google sign-in requirement</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"14QvmgISxaArTzWt_UHvf55aAYN2zm84Q"</span><span class="op">)</span>, <span class="st">"SKCM_RNASeqV2.geneExp.rds"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bulkdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"SKCM_RNASeqV2.geneExp.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">bulkdata</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       TCGA-3N-A9WB-06 TCGA-3N-A9WC-06 TCGA-3N-A9WD-06 TCGA-BF-A1PU-01</span></span>
<span><span class="co">## A1BG         381.0660         195.182        360.8790        176.3990</span></span>
<span><span class="co">## A1CF           0.0000           0.000          0.7092          0.0000</span></span>
<span><span class="co">## A2BP1          0.0000           0.000          6.3830          1.2987</span></span>
<span><span class="co">## A2LD1        250.1980         160.755         97.1986        163.2340</span></span>
<span><span class="co">## A2M         2209.5200      169237.000      18257.9000       6716.4500</span></span>
<span><span class="co">## A2ML1          7.2698           0.000          0.0000          7.7922</span></span>
<span><span class="co">##       TCGA-BF-A1PV-01</span></span>
<span><span class="co">## A1BG         216.8470</span></span>
<span><span class="co">## A1CF           0.0000</span></span>
<span><span class="co">## A2BP1          0.0000</span></span>
<span><span class="co">## A2LD1         60.8727</span></span>
<span><span class="co">## A2M         1740.5800</span></span>
<span><span class="co">## A2ML1          0.5977</span></span></code></pre>
<p>Tab-delimited files can be efficiently loaded into R using the
<code>fread</code> function from the <code>data.table</code>
package.</p>
</div>
<div class="section level2">
<h2 id="se-deconvolution">SE deconvolution<a class="anchor" aria-label="anchor" href="#se-deconvolution"></a>
</h2>
<p>The <code>DeconvoluteSE</code> function will be used to infer SE
abundances in bulk tissues. Users can utilize the default model for
estimating the abundance of predefined SEs or apply a custom model to
infer the abundance of newly defined SEs from bulk gene expression data.
Before prediction, the gene expression data will be log2-transformed (if
the max value is greater than 80) and normalized to have a mean of 0 and
unit variance for each gene.</p>
<p>The default model is trained on pseudobulk gene expression data and
is capable of predicting the abundances of nine predefined SEs, along
with a NonSE group. The NonSE group primarily represents cancer cells
and cells that do not belong to any SEs. The predicted abundances of SEs
and NonSE will sum to 1 within each sample.</p>
<p><strong>Note</strong>: 1) The SE abundances are comparable across
different samples, allowing users to analyze SE distribution patterns in
various conditions. 2) Due to the varying number of cell states within
each SE, direct comparisons of abundance between different SEs and NonSE
may not be meaningful. The SEs represent distinct spatial multicellular
communities, each composed of a unique combination of cell types.</p>
<div class="section level3">
<h3 id="using-default-model">Using default model<a class="anchor" aria-label="anchor" href="#using-default-model"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sefracs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DeconvoluteSE.html">DeconvoluteSE</a></span><span class="op">(</span><span class="va">bulkdata</span>, scale <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sefracs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                      NonSE       SE01       SE02       SE03       SE04</span></span>
<span><span class="co">## TCGA-3N-A9WB-06 0.19380234 0.05372667 0.16735600 0.12581663 0.07216666</span></span>
<span><span class="co">## TCGA-3N-A9WC-06 0.05293689 0.13901340 0.08427617 0.08372047 0.04243859</span></span>
<span><span class="co">## TCGA-3N-A9WD-06 0.27200075 0.17114795 0.06288761 0.05641549 0.10166721</span></span>
<span><span class="co">## TCGA-BF-A1PU-01 0.16899510 0.13618971 0.06501306 0.09269615 0.17313296</span></span>
<span><span class="co">## TCGA-BF-A1PV-01 0.12078283 0.06921834 0.05970319 0.16813008 0.11059648</span></span>
<span><span class="co">## TCGA-BF-A1PX-01 0.21398415 0.04019509 0.08906215 0.04499893 0.08561680</span></span>
<span><span class="co">##                       SE05         SE07         SE08       SE10       SE11</span></span>
<span><span class="co">## TCGA-3N-A9WB-06 0.06884572 6.495692e-02 4.610809e-03 0.16471168 0.08400658</span></span>
<span><span class="co">## TCGA-3N-A9WC-06 0.05623098 6.781589e-03 3.162253e-01 0.15964259 0.05873404</span></span>
<span><span class="co">## TCGA-3N-A9WD-06 0.02720647 8.345475e-02 9.239064e-02 0.07760029 0.05522884</span></span>
<span><span class="co">## TCGA-BF-A1PU-01 0.13404194 3.330034e-16 3.330034e-16 0.02543647 0.20449461</span></span>
<span><span class="co">## TCGA-BF-A1PV-01 0.14669963 1.677658e-01 1.260194e-03 0.06961553 0.08622790</span></span>
<span><span class="co">## TCGA-BF-A1PX-01 0.09930790 7.771710e-02 1.239738e-01 0.12590726 0.09923685</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="using-custom-model">Using custom model<a class="anchor" aria-label="anchor" href="#using-custom-model"></a>
</h3>
<p>After the identification of SEs using <code>SpatialEcoTyper</code> or
<code>MultiSpatialEcoTyper</code>, users can develop an NMF model to
infer SE abundances from bulk tissue samples, following the tutorial
below.</p>
<details><summary><strong>Development of SE deconvolution models</strong>
</summary><p>Users can also develop an NMF model to deconvolve SEs from bulk gene
expression data. The training data can be pseudo-bulk mixtures by
aggregating single-cell transcriptomics. In this tutorial, we will
create pseudo-bulk mixtures from single-cell RNA-seq data of a melanoma
sample, which is also used in the tutorial <a href="Recovery_scRNA.html">Recovery of Spatial Ecotypes from Single-Cell
Gene Expression Data</a>. The raw count data is available at <a href="https://drive.google.com/open?id=15n9zlXed74oeGaO1pythOOM_iWIfuMn2&amp;usp=drive_fs" class="external-link">Melanoma_WU2161_counts.rds</a>
and SE recovery results are available at <a href="https://drive.google.com/open?id=15apmSDhcANjx3i05LIFIm01gelbP7etA&amp;usp=drive_fs" class="external-link">Melanoma_WU2161_RecoveredSEs.rds</a>.</p>
<p><strong>Loading data</strong></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Download single-cell gene expression matrix and SE recovery results.</span></span>
<span><span class="co"># The downloads should be finished within 1min.</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"15n9zlXed74oeGaO1pythOOM_iWIfuMn2"</span><span class="op">)</span>, <span class="st">"Melanoma_WU2161_counts.rds"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1nzhKQM6HtX_rzzn3edSQsbodFBO5ycYG"</span><span class="op">)</span>, <span class="st">"Melanoma_WU2161_RecoveredSEs.rds"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load single-cell gene expression matrix.</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"Melanoma_WU2161_counts.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 27425  1337</span></span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load SE recovery results</span></span>
<span><span class="va">SEs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"Melanoma_WU2161_RecoveredSEs.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">SEs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1337</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">SEs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## SEs</span></span>
<span><span class="co">## NonSE  SE01  SE02  SE03  SE04  SE05  SE06  SE07  SE08  SE09  SE10  SE11 </span></span>
<span><span class="co">##   593    71    72    82    24    62    29   174    75    28    71    56</span></span></code></pre>
<p>Tab-delimited files can be efficiently loaded into R using the
<code>fread</code> function from the <code>data.table</code> package,
while sparse matrix files in the .mtx format can be imported using the
<code>ReadMtx</code> function from the <code>Seurat</code> package.</p>
<p><strong>Note</strong>: For demonstration purposes, we used 1,337
cells grouped into SEs to create pseudo-bulk mixtures. While this
limited cell number offers a basic example, it may not fully capture the
diverse characteristics of SEs, potentially affecting the robustness of
model training and subsequent SE recovery. In practice, we recommend
using a more comprehensive dataset that accurately reflects the
properties of SEs, ensuring that the training process results in a
reliable recovery model.</p>
<p><strong>Creating pseudobulk data</strong></p>
<p>The <code>CreatePseudobulks</code> function will be used to create
pseudo-bulk mixtures from single-cell (spatial) transcriptomics data
with all cells grouped into SEs. It samples cell fractions from a
Gaussian distribution, sets negative fractions to 0 and re-normalizes
fractions to sum to 1 across all SEs. Based on the resulting fractions,
it samples 1,000 cells per pseudo-bulk mixture with replacement,
aggregates their transcriptomes in non-log linear space, and normalize
the resulting mixture to logarithm CPM using Seuratâ€™s
<code>NormalizeData</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">result</span> <span class="op">=</span> <span class="fu"><a href="../reference/CreatePseudobulks.html">CreatePseudobulks</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">counts</span>, groups <span class="op">=</span> <span class="va">SEs</span>, n_mixtures <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">Mixtures</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span> <span class="co">## Gene expression matrix of pseudobulks</span></span></code></pre></div>
<pre><code><span><span class="co">## 6 x 5 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##            Pseudobulk1  Pseudobulk2 Pseudobulk3  Pseudobulk4  Pseudobulk5</span></span>
<span><span class="co">## AL627309.1 0.016972085 0.0272125699 0.019494706 0.0151417476 0.0215163218</span></span>
<span><span class="co">## AL627309.5 0.062939626 0.0585763400 0.072680217 0.0832459792 0.0634250393</span></span>
<span><span class="co">## AP006222.2 0.001737547 0.0009746133 0.001615377 0.0012089628 0.0037481743</span></span>
<span><span class="co">## AC114498.1 0.006997929 .            .           0.0009244557 .           </span></span>
<span><span class="co">## AL669831.2 0.001110700 0.0003703704 .           0.0007413447 0.0007413447</span></span>
<span><span class="co">## LINC01409  0.046240107 0.0301088599 0.025292195 0.0297990066 0.0390591928</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">Fracs</span><span class="op">)</span> <span class="co">## SE fractions in pseudobulks</span></span></code></pre></div>
<pre><code><span><span class="co">##                  NonSE       SE01       SE02       SE03       SE04       SE05</span></span>
<span><span class="co">## Pseudobulk1 0.13578633 0.02103231 0.06798558 0.12710418 0.03534255 0.05412050</span></span>
<span><span class="co">## Pseudobulk2 0.11636180 0.11980590 0.04819365 0.16457608 0.04770089 0.10847818</span></span>
<span><span class="co">## Pseudobulk3 0.04572172 0.12322551 0.08136232 0.08215961 0.16635320 0.06385032</span></span>
<span><span class="co">## Pseudobulk4 0.09239499 0.09412134 0.07994071 0.15223339 0.07900003 0.08241206</span></span>
<span><span class="co">## Pseudobulk5 0.00000000 0.16352951 0.10843578 0.08475981 0.02459614 0.15638880</span></span>
<span><span class="co">## Pseudobulk6 0.09423837 0.08819713 0.09091754 0.08608907 0.00000000 0.09000149</span></span>
<span><span class="co">##                   SE06       SE07       SE08       SE09       SE10       SE11</span></span>
<span><span class="co">## Pseudobulk1 0.11506530 0.06752668 0.14421556 0.05536032 0.09577202 0.08068867</span></span>
<span><span class="co">## Pseudobulk2 0.07058555 0.08052256 0.01036493 0.11241051 0.04454126 0.07645870</span></span>
<span><span class="co">## Pseudobulk3 0.04810001 0.08344317 0.12046230 0.08193738 0.05990561 0.04347886</span></span>
<span><span class="co">## Pseudobulk4 0.06384581 0.06932315 0.02105041 0.09074042 0.10213406 0.07280364</span></span>
<span><span class="co">## Pseudobulk5 0.10306946 0.03836312 0.08145100 0.11321193 0.04679763 0.07939680</span></span>
<span><span class="co">## Pseudobulk6 0.11000033 0.08445375 0.09840190 0.08880690 0.10736759 0.06152594</span></span></code></pre>
<p><strong>Training</strong></p>
<p>The <code>NMFGenerateW</code> function will be used to train an NMF
model for SE deconvolution based on the provided SE fractions and gene
expression matrix. Before applying NMF, each geneâ€™s expression is scaled
to have a mean of 0 and unit variance (recommended). To meet the
non-negativity requirement of NMF, the expression matrix is transformed
using the posneg method. This transformation splits the expression
matrix into two matrices: one containing only positive values and the
other containing the absolute values of the negative values. These two
matrices are then concatenated to form the final training data for the
NMF model.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">W</span> <span class="op">=</span> <span class="fu"><a href="../reference/NMFGenerateW.html">NMFGenerateW</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">Fracs</span><span class="op">)</span>, <span class="va">result</span><span class="op">$</span><span class="va">Mixtures</span><span class="op">)</span></span>
<span><span class="co">## This step should be done within 1 min</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">W</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                      NonSE         SE01         SE02         SE03         SE04</span></span>
<span><span class="co">## IGLV3-1__pos  2.220446e-16 2.220446e-16 2.220446e-16 4.989224e-15 3.000074e-10</span></span>
<span><span class="co">## IGLV3-1__neg  8.001643e-04 9.649978e-01 6.595523e-07 5.525819e-13 5.087300e-15</span></span>
<span><span class="co">## IGLV6-57__neg 2.220446e-16 2.220446e-16 3.692198e-01 2.220446e-16 2.220446e-16</span></span>
<span><span class="co">## IGLV1-51__neg 8.921563e-01 8.692045e-11 8.659573e-11 2.220446e-16 2.811801e-01</span></span>
<span><span class="co">## IGKV3-15__neg 2.220446e-16 2.811050e-01 1.122226e-05 2.220446e-16 4.869723e-08</span></span>
<span><span class="co">## APOE__pos     2.220446e-16 2.220446e-16 2.220446e-16 2.220446e-16 2.220446e-16</span></span>
<span><span class="co">##                       SE05         SE06         SE07         SE08         SE09</span></span>
<span><span class="co">## IGLV3-1__pos  2.220446e-16 2.220446e-16 5.802943e+00 2.220446e-16 2.220446e-16</span></span>
<span><span class="co">## IGLV3-1__neg  8.653475e-01 1.871899e+00 2.220446e-16 1.945312e-03 1.180617e-04</span></span>
<span><span class="co">## IGLV6-57__neg 5.817428e-02 2.295402e+00 5.412925e-01 1.748893e-07 2.220446e-16</span></span>
<span><span class="co">## IGLV1-51__neg 1.317293e+00 1.928096e+00 2.737648e-02 3.558364e-02 9.261255e-01</span></span>
<span><span class="co">## IGKV3-15__neg 7.318096e-11 1.504377e+00 3.096751e-01 9.066236e-10 1.291017e-11</span></span>
<span><span class="co">## APOE__pos     4.972596e+00 2.220446e-16 2.220446e-16 2.220446e-16 2.220446e-16</span></span>
<span><span class="co">##                       SE10         SE11</span></span>
<span><span class="co">## IGLV3-1__pos  2.220446e-16 2.220446e-16</span></span>
<span><span class="co">## IGLV3-1__neg  4.877208e-02 1.938151e+00</span></span>
<span><span class="co">## IGLV6-57__neg 2.507537e-01 2.088166e+00</span></span>
<span><span class="co">## IGLV1-51__neg 9.102171e-10 1.213205e-01</span></span>
<span><span class="co">## IGKV3-15__neg 1.386490e-01 3.260772e+00</span></span>
<span><span class="co">## APOE__pos     2.220446e-16 5.215751e-01</span></span></code></pre>
<p>The resulting W matrix can be used to infer SE sbundances from bulk
gene expression profiles by using the <code>DeconvoluteSE</code>
function.</p>
</details><p>An example model is available at <a href="https://drive.google.com/open?id=1mRyABC3NtLIlZtQ1UqtUIr7MhF62Vs3e&amp;usp=drive_fs" class="external-link">SE_Deconvolution_W.rds</a>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1mRyABC3NtLIlZtQ1UqtUIr7MhF62Vs3e"</span><span class="op">)</span>, <span class="st">"SE_Deconvolution_W.rds"</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">W</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"SE_Deconvolution_W.rds"</span><span class="op">)</span></span></code></pre></div>
<p>Then you can use the new model for SE deconvolution by specifying the
<code>W</code> parameter:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sefracs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DeconvoluteSE.html">DeconvoluteSE</a></span><span class="op">(</span><span class="va">bulkdata</span>, W <span class="op">=</span> <span class="va">W</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sefracs</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualization-of-se-abundances">Visualization of SE abundances<a class="anchor" aria-label="anchor" href="#visualization-of-se-abundances"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">grid</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/HeatmapView.html">HeatmapView</a></span><span class="op">(</span><span class="va">sefracs</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.15</span>, <span class="fl">0.3</span><span class="op">)</span>, </span>
<span>            show_row_names <span class="op">=</span> <span class="cn">FALSE</span>, cluster_rows <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="Recovery_Bulk_files/figure-html/visualse-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>The session info allows users to replicate the exact environment and
identify potential discrepancies in package versions or configurations
that might be causing problems.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.3 (2023-03-15)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin13.4.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS/LAPACK: /Users/wzhang39/miniconda3/envs/SpatialEcoTyper/lib/libopenblasp-r0.3.27.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] grid      parallel  stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [8] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] SpatialEcoTyper_0.0.4 NMF_0.28              Biobase_2.58.0       </span></span>
<span><span class="co">##  [4] BiocGenerics_0.44.0   cluster_2.1.6         rngtools_1.5.2       </span></span>
<span><span class="co">##  [7] registry_0.5-1        dplyr_1.1.4           RANN_2.6.2           </span></span>
<span><span class="co">## [10] Matrix_1.6-5          SeuratObject_5.0.2    Seurat_4.4.0         </span></span>
<span><span class="co">## [13] googledrive_2.1.1    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] circlize_0.4.16        spatstat.univar_3.0-0  spam_2.10-0           </span></span>
<span><span class="co">##   [4] systemfonts_1.1.0      plyr_1.8.9             igraph_2.0.3          </span></span>
<span><span class="co">##   [7] lazyeval_0.2.2         sp_2.1-4               splines_4.2.3         </span></span>
<span><span class="co">##  [10] listenv_0.9.1          scattermore_1.2        ggplot2_3.5.1         </span></span>
<span><span class="co">##  [13] gridBase_0.4-7         digest_0.6.36          foreach_1.5.2         </span></span>
<span><span class="co">##  [16] htmltools_0.5.8.1      fansi_1.0.6            magrittr_2.0.3        </span></span>
<span><span class="co">##  [19] memoise_2.0.1          tensor_1.5             doParallel_1.0.17     </span></span>
<span><span class="co">##  [22] ROCR_1.0-11            ComplexHeatmap_2.14.0  globals_0.16.3        </span></span>
<span><span class="co">##  [25] matrixStats_1.3.0      pkgdown_2.0.9          spatstat.sparse_3.1-0 </span></span>
<span><span class="co">##  [28] colorspace_2.1-1       ggrepel_0.9.5          textshaping_0.4.0     </span></span>
<span><span class="co">##  [31] xfun_0.45              crayon_1.5.3           jsonlite_1.8.8        </span></span>
<span><span class="co">##  [34] progressr_0.14.0       spatstat.data_3.1-2    survival_3.7-0        </span></span>
<span><span class="co">##  [37] zoo_1.8-12             iterators_1.0.14       glue_1.7.0            </span></span>
<span><span class="co">##  [40] polyclip_1.10-7        gtable_0.3.5           gargle_1.5.2          </span></span>
<span><span class="co">##  [43] leiden_0.4.3.1         GetoptLong_1.0.5       shape_1.4.6.1         </span></span>
<span><span class="co">##  [46] future.apply_1.11.2    abind_1.4-5            scales_1.3.0          </span></span>
<span><span class="co">##  [49] spatstat.random_3.3-1  miniUI_0.1.1.1         Rcpp_1.0.12           </span></span>
<span><span class="co">##  [52] viridisLite_0.4.2      xtable_1.8-4           clue_0.3-65           </span></span>
<span><span class="co">##  [55] reticulate_1.38.0      dotCall64_1.1-1        stats4_4.2.3          </span></span>
<span><span class="co">##  [58] htmlwidgets_1.6.4      httr_1.4.7             RColorBrewer_1.1-3    </span></span>
<span><span class="co">##  [61] ica_1.0-3              pkgconfig_2.0.3        sass_0.4.9            </span></span>
<span><span class="co">##  [64] uwot_0.2.2             deldir_2.0-4           utf8_1.2.4            </span></span>
<span><span class="co">##  [67] tidyselect_1.2.1       rlang_1.1.4            reshape2_1.4.4        </span></span>
<span><span class="co">##  [70] later_1.3.2            munsell_0.5.1          tools_4.2.3           </span></span>
<span><span class="co">##  [73] cachem_1.1.0           cli_3.6.3              generics_0.1.3        </span></span>
<span><span class="co">##  [76] ggridges_0.5.6         evaluate_0.24.0        stringr_1.5.1         </span></span>
<span><span class="co">##  [79] fastmap_1.2.0          yaml_2.3.8             ragg_1.3.2            </span></span>
<span><span class="co">##  [82] goftest_1.2-3          knitr_1.47             fs_1.6.4              </span></span>
<span><span class="co">##  [85] fitdistrplus_1.2-1     purrr_1.0.2            pbapply_1.7-2         </span></span>
<span><span class="co">##  [88] future_1.34.0          nlme_3.1-165           mime_0.12             </span></span>
<span><span class="co">##  [91] compiler_4.2.3         rstudioapi_0.16.0      plotly_4.10.4         </span></span>
<span><span class="co">##  [94] curl_5.1.0             png_0.1-8              spatstat.utils_3.1-0  </span></span>
<span><span class="co">##  [97] tibble_3.2.1           bslib_0.7.0            stringi_1.8.4         </span></span>
<span><span class="co">## [100] highr_0.11             desc_1.4.3             lattice_0.22-6        </span></span>
<span><span class="co">## [103] vctrs_0.6.5            pillar_1.9.0           lifecycle_1.0.4       </span></span>
<span><span class="co">## [106] BiocManager_1.30.24    GlobalOptions_0.1.2    spatstat.geom_3.3-2   </span></span>
<span><span class="co">## [109] lmtest_0.9-40          jquerylib_0.1.4        RcppAnnoy_0.0.22      </span></span>
<span><span class="co">## [112] data.table_1.15.4      cowplot_1.1.3          irlba_2.3.5.1         </span></span>
<span><span class="co">## [115] httpuv_1.6.15          patchwork_1.2.0        R6_2.5.1              </span></span>
<span><span class="co">## [118] promises_1.3.0         KernSmooth_2.23-24     gridExtra_2.3         </span></span>
<span><span class="co">## [121] IRanges_2.32.0         parallelly_1.38.0      codetools_0.2-20      </span></span>
<span><span class="co">## [124] MASS_7.3-60.0.1        rjson_0.2.21           sctransform_0.4.1     </span></span>
<span><span class="co">## [127] S4Vectors_0.36.2       tidyr_1.3.1            rmarkdown_2.27        </span></span>
<span><span class="co">## [130] Rtsne_0.17             spatstat.explore_3.3-2 shiny_1.8.1.1</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://profiles.stanford.edu/wubing-zhang" class="external-link">Wubing Zhang</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
