<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="SpatialEcoTyper">
<title>Discovery of Spatial Ecotypes from a Single Sample • SpatialEcoTyper</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Discovery of Spatial Ecotypes from a Single Sample">
<meta property="og:description" content="SpatialEcoTyper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-default navbar-expand-lg bg-light" data-bs-theme="default"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpatialEcoTyper</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.0.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href="../articles/SingleSample.html">Discovery of Spatial Ecotypes from a Single Sample</a>
    <a class="dropdown-item" href="../articles/Integration.html">Discovery of Spatial Ecotypes from Multiple Samples</a>
    <a class="dropdown-item" href="../articles/TrainRecoveryModels.html">Development of NMF Models for Spatial Ecotype Recovery</a>
    <a class="dropdown-item" href="../articles/Recovery_Spatial.html">Recovery of Spatial Ecotypes from Spatial Transcriptomics Data</a>
    <a class="dropdown-item" href="../articles/Recovery_scRNA.html">Recovery of Spatial Ecotypes from Single-Cell Gene Expression Data</a>
    <a class="dropdown-item" href="../articles/Recovery_Bulk.html">Recovery of Spatial Ecotypes from Bulk Gene Expression Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">R functions</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Discovery of Spatial Ecotypes from a Single Sample</h1>
            
      
      <small class="dont-index">Source: <a href="https://digitalcytometry.github/SpatialEcoTypervignettes/SingleSample.Rmd" class="external-link"><code>vignettes/SingleSample.Rmd</code></a></small>
      <div class="d-none name"><code>SingleSample.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>In this tutorial, we will illustrate how to perform de novo discovery
of spatially distinct cellular ecosystems (SEs) from single-cell spatial
transcriptomics dataset using the <code>SpatialEcoTyper</code>
function.</p>
<p><strong>First load required packages for this vignette</strong></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">parallel</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://googledrive.tidyverse.org" class="external-link">googledrive</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://henrikbengtsson.github.io/R.utils/" class="external-link">R.utils</a></span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://digitalcytometry.github/SpatialEcoTyper" class="external-link">SpatialEcoTyper</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<p>We will be analyzing single-cell spatial transcriptomics data from a
melanoma sample (raw data available in <a href="https://vizgen.com/data-release-program/" class="external-link">Vizgen’s MERSCOPE FFPE
Human Immuno-oncology</a>). To quickly demonstrate how SpatialEcoTyper
works, we’ve selected a subset of the sample, which can be downloaded
from <a href="https://drive.google.com/drive/folders/17K5XR7VEvxN9walV4RtaLOR2cBRG-aMh?usp=drive_link" class="external-link"><code>here</code></a>.</p>
<details open><summary><strong>Text files as input</strong>
</summary><p>Download the data from Google Drive</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_deauth.html" class="external-link">drive_deauth</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Disable Google sign-in requirement</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"13Rc5Rsu8jbnEYYfUse-xQ7ges51LcI7n"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_counts.tsv.gz"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"12xcZNhpT-xbhcG8kX1QAdTeM9TKeFAUW"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_scmeta.tsv"</span><span class="op">)</span></span></code></pre></div>
<p>Large text files can be loaded into R using the <code>fread</code>
function from the <code>data.table</code> package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load single-cell gene expression matrix. Rows are genes, columns are cells.</span></span>
<span><span class="va">scdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">"HumanMelanomaPatient1_subset_counts.tsv.gz"</span>, </span>
<span>                sep <span class="op">=</span> <span class="st">"\t"</span>,header <span class="op">=</span> <span class="cn">TRUE</span>, data.table <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">scdata</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>  <span class="co"># set genes as row names</span></span>
<span><span class="va">scdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          HumanMelanomaPatient1__cell_3655 HumanMelanomaPatient1__cell_3657</span></span>
<span><span class="co">## PDK4                                    0                                1</span></span>
<span><span class="co">## TNFRSF17                                0                                0</span></span>
<span><span class="co">## ICAM3                                   0                                0</span></span>
<span><span class="co">## FAP                                     1                                0</span></span>
<span><span class="co">## GZMB                                    0                                0</span></span>
<span><span class="co">## TSC2                                    0                                0</span></span>
<span><span class="co">##          HumanMelanomaPatient1__cell_3658 HumanMelanomaPatient1__cell_3660</span></span>
<span><span class="co">## PDK4                                    1                                0</span></span>
<span><span class="co">## TNFRSF17                                0                                0</span></span>
<span><span class="co">## ICAM3                                   0                                0</span></span>
<span><span class="co">## FAP                                     0                                0</span></span>
<span><span class="co">## GZMB                                    0                                0</span></span>
<span><span class="co">## TSC2                                    0                                0</span></span>
<span><span class="co">##          HumanMelanomaPatient1__cell_3661</span></span>
<span><span class="co">## PDK4                                    0</span></span>
<span><span class="co">## TNFRSF17                                0</span></span>
<span><span class="co">## ICAM3                                   0</span></span>
<span><span class="co">## FAP                                     0</span></span>
<span><span class="co">## GZMB                                    0</span></span>
<span><span class="co">## TSC2                                    0</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load single-cell metadata. Three columns are required, including X, Y, and CellType. Others are optional.</span></span>
<span><span class="va">scmeta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"HumanMelanomaPatient1_subset_scmeta.tsv"</span>, </span>
<span>                      sep <span class="op">=</span> <span class="st">"\t"</span>,header <span class="op">=</span> <span class="cn">TRUE</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">scmeta</span> <span class="op">&lt;-</span> <span class="va">scmeta</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span>, <span class="op">]</span> <span class="co"># match the cell ids in scdata and scmeta</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                         X         Y   CellType CellTypeName</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3655 1894.706 -6367.766 Fibroblast  Fibroblasts</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657 1942.480 -6369.602 Fibroblast  Fibroblasts</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658 1963.007 -6374.026 Fibroblast  Fibroblasts</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660 1981.600 -6372.266 Fibroblast  Fibroblasts</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661 1742.939 -6374.851 Fibroblast  Fibroblasts</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3663 1921.683 -6383.309 Fibroblast  Fibroblasts</span></span>
<span><span class="co">##                                  Region Dist2Interface</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3655 Stroma      -883.1752</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657 Stroma      -894.8463</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658 Stroma      -904.1115</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660 Stroma      -907.8909</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661 Stroma      -874.2712</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3663 Stroma      -903.6559</span></span></code></pre>
</details><details><summary><strong>Sparse matrix as input</strong>
</summary><p>SpatialEcoTyper supports sparse matrix as input. Mtx files can be
loaded into R using the <a href="https://satijalab.org/seurat/reference/readmtx" class="external-link"><code>ReadMtx</code></a>
function from the Seurat package.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"13M3xhRxp0xK9gf5F4DE9idSBFqVQIXDT"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_counts.mtx.gz"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"136feRaFjMtNvduLTm5xqa3WhyyoG4Xzo"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_cells.tsv.gz"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"13QprWzJhzzUy_w3XSrjlt9pjf2n-G7HV"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_genes.tsv.gz"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">scdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ReadMtx.html" class="external-link">ReadMtx</a></span><span class="op">(</span>mtx <span class="op">=</span> <span class="st">"HumanMelanomaPatient1_subset_counts.mtx.gz"</span>, cells <span class="op">=</span> <span class="st">"HumanMelanomaPatient1_subset_cells.tsv.gz"</span>, features <span class="op">=</span> <span class="st">"HumanMelanomaPatient1_subset_genes.tsv.gz"</span>, feature.column <span class="op">=</span> <span class="fl">1</span>, cell.column <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
</details>
</div>
<div class="section level2">
<h2 id="data-normalization">Data normalization<a class="anchor" aria-label="anchor" href="#data-normalization"></a>
</h2>
<p>The gene expression data should be normalized before SpatialEcoTyper
analysis. The data can be normalized using <a href="https://satijalab.org/seurat/reference/normalizedata" class="external-link">NormalizeData</a>
or <a href="https://satijalab.org/seurat/articles/seurat5_spatial_vignette_2" class="external-link">SCTransform</a>.</p>
<p>Here, we are normalizing using SCTransform normalization. We
recommend to install the glmGamPoi package for faster computation.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"glmGamPoi"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"glmGamPoi"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">tmpobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://satijalab.org/seurat/reference/SCTransform.html" class="external-link">SCTransform</a></span><span class="op">(</span>clip.range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">10</span>, <span class="fl">10</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_version</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html" class="external-link">packageVersion</a></span><span class="op">(</span><span class="st">"SeuratObject"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="va">seurat_version</span><span class="op">&lt;</span><span class="fl">5</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">normdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">tmpobj</span>, <span class="st">"data"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="kw">else</span><span class="op">{</span></span>
<span>  <span class="va">normdata</span> <span class="op">&lt;-</span> <span class="va">tmpobj</span><span class="op">[[</span><span class="st">"SCT"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">data</span></span>
<span><span class="op">}</span></span></code></pre></div>
<summary>
To use <code>NormalizeData</code> for the normalization:
</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preview-of-the-sample">Preview of the sample<a class="anchor" aria-label="anchor" href="#preview-of-the-sample"></a>
</h2>
<p>The demo sample includes spatial expression of 500 genes across
27,907 cells, categorized into ten cell types, namely B cells, CD4+ T
cells, CD8+ T cells, NK cells, plasma cells, macrophages, dendritic
cells (DC), fibroblasts, endothelial cells, as well as cancer cell of
origin that are eliminated for the analysis.</p>
<p>Besides, all cells were grouped into four regions, including tumor,
inner margin, outer margin, and stroma. The tumor and stroma regions
were defined based on the density of cancer cells. For detailed
information about the tumor and stroma annotations, please refer to the
<a href="https://www.nature.com/articles/s41587-023-01697-9" class="external-link">CytoSPACE
paper</a>. The inner and outer margins were delineated as regions
extending 250 μm inside and outside the tumor boundaries,
respectively.</p>
<p>The <code>SpatialView</code> function can be used to visualize single
cells in spatial space, by cell type or by region.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize the cell type annotations in the tissue</span></span>
<span><span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">scmeta</span>, by <span class="op">=</span> <span class="st">"CellType"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/discrete.html" class="external-link">kelly</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="SingleSample_files/figure-html/viewct-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize the regions in the tissue</span></span>
<span><span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">scmeta</span>, by <span class="op">=</span> <span class="st">"Region"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span>, palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SingleSample_files/figure-html/viewregion-1.png" width="700"></p>
<p>The <code>SpatialView</code> function can also be used to visualize
continuous characteristics, such as the minimum distance of each single
cell to tumor/stroma margin and gene expression levels.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize the distance to tumor margin</span></span>
<span><span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">scmeta</span>, by <span class="op">=</span> <span class="st">"Dist2Interface"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_colour_gradient2</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">"#5e3c99"</span>, high <span class="op">=</span> <span class="st">"#e66101"</span>, </span>
<span>                         mid <span class="op">=</span> <span class="st">"#d9d9d9"</span>, midpoint <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="SingleSample_files/figure-html/viewdist-1.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Visualize gene expression in spatial</span></span>
<span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">scmeta</span></span>
<span><span class="va">gg</span><span class="op">$</span><span class="va">Expression</span> <span class="op">&lt;-</span> <span class="va">normdata</span><span class="op">[</span><span class="st">"STAT1"</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">gg</span>, by <span class="op">=</span> <span class="st">"Expression"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html" class="external-link">scale_color_viridis_c</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="SingleSample_files/figure-html/viewgene-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="se-discovery-using-spatialecotyper">SE discovery using SpatialEcoTyper<a class="anchor" aria-label="anchor" href="#se-discovery-using-spatialecotyper"></a>
</h2>
<p><img src="images/SpatialEcoTyper_Workflow.png" width="100%"></p>
<p>SpatialEcoTyper is designed to identify spatial ecotypes from
single-cell spatial transcriptomics data. The workflow begins by
grouping spatially co-localized single cells into microregions, where
cell-type-specific gene expression profiles (GEPs) are constructed for
each microregion. Following this, similarity networks of these
microregions are constructed based on their cell type-specific GEPs. To
integrate these networks, SpatialEcoTyper utilizes the Similarity
Network Fusion (SNF) method, originally developed for multi-omics data
integration (<a href="https://www.nature.com/articles/nmeth.2810" class="external-link">Wang
et al., 2014</a>). The fused similarity network represents the spatial
architecture of the tissue, enabling the identification of spatial
ecotypes (SEs). Finally, SEs are detected by grouping similar spatial
microregions using the Louvain algorithm.</p>
<details open><summary><strong>Key arguments for SpatialEcoTyper</strong>
</summary><ul>
<li>
<code>normdata</code> A matrix representing normalized gene
expression data, where rows correspond to genes and columns correspond
to cells.</li>
<li>
<code>metadata</code> A data frame containing metadata associated
with each cell. Must include spatial coordinates (e.g., X and Y) as well
as cell type annotations. The row names must match the column names of
the .</li>
<li>
<code>outprefix</code> Character string specifying the prefix for
output file names.</li>
<li>
<code>radius</code> Numeric specifying the radius (default: 50 µm)
for defining spatial microregion.</li>
<li>
<code>resolution</code> Numeric specifying the resolution for
Louvain clustering (default: 0.8).</li>
<li>
<code>nfeatures</code> Integer specifying the number of top variable
features (default: 3000) for PCA.</li>
<li>
<code>min.cts.per.region</code> Integer specifying the minimum
number of cell types (default: 2) required for a microregion.</li>
<li>
<code>ncores</code> Integer specifying the number of CPU cores to
use for parallel processing.</li>
</ul>
<p>You can type <code><a href="../reference/SpatialEcoTyper.html">?SpatialEcoTyper</a></code> to visualize the full
manual. Here, we only show a list of important parameters including:</p>
</details><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## This step takes ~2 minutes to complete on macOS with an Apple M1 Pro chip and 16 GB memory.</span></span>
<span><span class="va">se_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SpatialEcoTyper.html">SpatialEcoTyper</a></span><span class="op">(</span><span class="va">normdata</span>, <span class="va">scmeta</span>,</span>
<span>                              outprefix <span class="op">=</span> <span class="st">"Melanoma1_subset"</span>,</span>
<span>                              radius <span class="op">=</span> <span class="fl">50</span>, ncores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p>When the <code>outprefix</code> is specified, the SpatialEcoTyper
result will be saved as a RDS file named
<code>outprefix</code>_SpatialEcoTyper_results.rds
(<code>Melanoma1_subset_SpatialEcoTyper_results.rds</code> <a href="https://drive.google.com/open?id=1nSPj2zRywFUdbo1fwiz77ds4NuM6bmV2&amp;usp=drive_fs" class="external-link">here</a>).</p>
<details><summary><strong>The results can be loaded into R via:</strong>
</summary><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"Melanoma1_subset_SpatialEcoTyper_results.rds"</span><span class="op">)</span></span></code></pre></div>
</details><p>The SpatialEcoTyper result is a list that includes two key
components: (1) a Seurat object constructed from the integrated
embedding of spatial microregions, allowing for detailed analysis and
visualization of spatial architecture; (2) A data frame of the
single-cell metadata, which now includes annotations for discovered
spatial ecotypes. These annotations can be used to explore the spatial
distribution and characteristics of SEs within the tissue.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract the Seurat object and updated single-cell metadata</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="va">se_results</span><span class="op">$</span><span class="va">obj</span> <span class="co"># A Seurat object</span></span>
<span><span class="va">scmeta</span> <span class="op">&lt;-</span> <span class="va">se_results</span><span class="op">$</span><span class="va">metadata</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span> <span class="co"># Updated single-cell meta data, with SE annotation added</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                         X         Y   CellType CellTypeName</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_4675 4578.019 -6445.993 Fibroblast  Fibroblasts</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_4738 4529.162 -6504.117 Fibroblast  Fibroblasts</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_4774 4533.228 -6529.591 Fibroblast  Fibroblasts</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_4803 4340.813 -6562.381 Fibroblast  Fibroblasts</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_4806 4401.938 -6567.152 Fibroblast  Fibroblasts</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_4808 4388.422 -6570.215 Fibroblast  Fibroblasts</span></span>
<span><span class="co">##                                        Region Dist2Interface  SE</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_4675 Outer margin     -197.09807 SE0</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_4738 Outer margin     -186.13295 SE0</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_4774 Outer margin     -199.69317 SE0</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_4803 Outer margin      -95.54956 SE0</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_4806 Outer margin     -124.01602 SE0</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_4808 Outer margin     -119.07031 SE0</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="embedding-of-spatial-architecture">Embedding of spatial architecture<a class="anchor" aria-label="anchor" href="#embedding-of-spatial-architecture"></a>
</h2>
<p>The spatial microregions’ embedding can be visualized using standard
<a href="https://satijalab.org/seurat/articles/visualization_vignette" class="external-link">Seurat
functions</a> such as <code>DimPlot</code> and <code>FeaturePlot</code>.
These visualizations help to explore the spatial organization and
heterogeneity within the tissue.</p>
<div class="section level3">
<h3 id="visualizing-tumorstroma-regions-in-the-embedding">Visualizing tumor/stroma regions in the embedding<a class="anchor" aria-label="anchor" href="#visualizing-tumorstroma-regions-in-the-embedding"></a>
</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">obj</span>, group.by <span class="op">=</span> <span class="st">"Region"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span>, palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SingleSample_files/figure-html/umap_region-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="visualizing-the-distance-of-microregions-to-tumorstroma-interface">Visualizing the distance of microregions to tumor/stroma
interface<a class="anchor" aria-label="anchor" href="#visualizing-the-distance-of-microregions-to-tumorstroma-interface"></a>
</h3>
<p>This plot highlights the distance of each spatial microregion to the
tumor/stroma interface, revealing the spatial continuum learnt by
SpatialEcoTyper.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">obj</span>, <span class="st">"Dist2Interface"</span>, min.cutoff <span class="op">=</span> <span class="op">-</span><span class="fl">600</span>, max.cutoff <span class="op">=</span> <span class="fl">600</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_colour_gradient2</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">"#5e3c99"</span>, high <span class="op">=</span> <span class="st">"#e66101"</span>, mid <span class="op">=</span> <span class="st">"#d9d9d9"</span>, midpoint <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="SingleSample_files/figure-html/umap_dist-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="visualizing-spatial-ecotypes-in-the-embedding">Visualizing spatial ecotypes in the embedding<a class="anchor" aria-label="anchor" href="#visualizing-spatial-ecotypes-in-the-embedding"></a>
</h3>
<p>This plot visualizes the SEs within the spatial embedding. The SEs
represent distinct spatial microenvironments, each with unique cellular
compositions and spatial characteristics.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">obj</span>, group.by <span class="op">=</span> <span class="st">"SE"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/discrete.html" class="external-link">kelly</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="SingleSample_files/figure-html/umap_se-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="visualizing-the-most-abundant-cell-types-in-each-spatial-microregion">Visualizing the most abundant cell types in each spatial
microregion<a class="anchor" aria-label="anchor" href="#visualizing-the-most-abundant-cell-types-in-each-spatial-microregion"></a>
</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">obj</span>, group.by <span class="op">=</span> <span class="st">"CellType"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/discrete.html" class="external-link">kelly</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="SingleSample_files/figure-html/umap_ct-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="se-characteristics">SE characteristics<a class="anchor" aria-label="anchor" href="#se-characteristics"></a>
</h2>
<div class="section level3">
<h3 id="visualizing-ses-in-the-tissue">Visualizing SEs in the tissue<a class="anchor" aria-label="anchor" href="#visualizing-ses-in-the-tissue"></a>
</h3>
<p>The spatial distribution of SEs within the tissue can be visualized
using the <code>SpatialView</code> function. This visualization provides
insights into the localization and spatial organization of different SEs
across the tissue section.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialView.html">SpatialView</a></span><span class="op">(</span><span class="va">scmeta</span>, by <span class="op">=</span> <span class="st">"SE"</span><span class="op">)</span> </span></code></pre></div>
<p><img src="SingleSample_files/figure-html/spatialmap-1.png" width="700"></p>
<p><strong>Note</strong>: Cells with NA values in the SE annotations
were excluded from this analysis. The exclusions occur due to the
absence of neighboring cells or an insufficient number of detected
genes.</p>
</div>
<div class="section level3">
<h3 id="visualizing-the-cell-type-composition-of-ses">Visualizing the cell type composition of SEs<a class="anchor" aria-label="anchor" href="#visualizing-the-cell-type-composition-of-ses"></a>
</h3>
<p>The bar plot below illustrates the cell type composition within each
SE. This visualization helps to identify the dominant cell types in each
SE and compare the cellular makeup across different SEs.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">scmeta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">CellType</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">gg</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">n</span>, fill <span class="op">=</span> <span class="va">CellType</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu">pals</span><span class="fu">::</span><span class="fu"><a href="https://kwstat.github.io/pals/reference/discrete.html" class="external-link">kelly</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"Cell type abundance"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SingleSample_files/figure-html/bar_ct-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="visualizing-the-association-between-ses-and-pre-annotated-regions">Visualizing the association between SEs and pre-annotated
regions<a class="anchor" aria-label="anchor" href="#visualizing-the-association-between-ses-and-pre-annotated-regions"></a>
</h3>
<p>This bar plot shows the relationship between SEs and pre-annotated
regions, such as tumor and stroma. The proportion of each SE associated
with these regions is displayed, providing insights into how SEs
correspond to different tissue regions.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">scmeta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">Region</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">gg</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">n</span>, fill <span class="op">=</span> <span class="va">Region</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"fill"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"qual"</span>, palette <span class="op">=</span> <span class="st">"Set1"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"Fraction"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SingleSample_files/figure-html/bar_region-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="visualizing-the-distance-of-ses-to-tumorstroma-interface">Visualizing the distance of SEs to tumor/stroma interface<a class="anchor" aria-label="anchor" href="#visualizing-the-distance-of-ses-to-tumorstroma-interface"></a>
</h3>
<p>This box plot visualizes the distribution of distances of SEs to the
tumor/stroma interface. SEs are ordered by their median distance,
highlighting spatial localization of SEs.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">scmeta</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## Order SEs by their distance to tumor/stroma interface</span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span>  <span class="va">gg</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>Mid <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="va">Dist2Interface</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">Mid</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span></span>
<span><span class="va">gg</span><span class="op">$</span><span class="va">SE</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">gg</span><span class="op">$</span><span class="va">SE</span>, levels <span class="op">=</span> <span class="va">tmp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">gg</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">Dist2Interface</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"Distance to tumor/stroma interface (μm)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SingleSample_files/figure-html/box_dist-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="identification-of-cell-type-specific-se-markers">Identification of cell-type-specific SE markers<a class="anchor" aria-label="anchor" href="#identification-of-cell-type-specific-se-markers"></a>
</h2>
<div class="section level3">
<h3 id="differential-expression-analysis">Differential expression analysis<a class="anchor" aria-label="anchor" href="#differential-expression-analysis"></a>
</h3>
<p>To identify cell-type-specific SE markers, we can perform
differential expression analysis using the <a href="https://github.com/immunogenomics/presto" class="external-link"><code>presto</code>
package</a>. Below is an example of how to identify fibroblast-specific
markers for each SE.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"presto"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">{</span> </span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"remotes"</span>, <span class="st">"immunogenomics/presto"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"presto"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Ensure the normalized data is aligned with the single-cell metadata</span></span>
<span><span class="va">normdata</span> <span class="op">=</span> <span class="va">normdata</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Perform differential expression analysis for fibroblasts across SEs</span></span>
<span><span class="va">idx</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">$</span><span class="va">CellType</span><span class="op">==</span><span class="st">"Fibroblast"</span> <span class="op">&amp;</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">degs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/presto/man/wilcoxauc.html" class="external-link">wilcoxauc</a></span><span class="op">(</span><span class="va">normdata</span><span class="op">[</span>, <span class="va">idx</span><span class="op">]</span>, <span class="va">scmeta</span><span class="op">$</span><span class="va">SE</span><span class="op">[</span><span class="va">idx</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter for significant markers with positive log-fold change</span></span>
<span><span class="va">degs</span> <span class="op">=</span> <span class="va">degs</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">logFC</span><span class="op">&gt;</span><span class="fl">0</span> <span class="op">&amp;</span> <span class="va">pval</span><span class="op">&lt;</span><span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">logFC</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">degs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   feature group  avgExpr     logFC statistic       auc          pval</span></span>
<span><span class="co">## 1   ACTA2   SE7 2.163320 1.8197807   3798740 0.7728462 3.967332e-181</span></span>
<span><span class="co">## 2 PLA2G2A   SE2 2.026133 1.4387873   5126624 0.8350462 7.288467e-307</span></span>
<span><span class="co">## 3     PKM   SE8 2.323114 1.3090753   1893378 0.8785406 2.013579e-115</span></span>
<span><span class="co">## 4 PLA2G2A   SE4 1.881473 1.2685947   4812222 0.7964803 5.779646e-237</span></span>
<span><span class="co">## 5     FN1   SE3 2.736915 1.0057420   4577625 0.7694509 1.404543e-156</span></span>
<span><span class="co">## 6  COL1A1   SE0 4.357378 0.9544396   8092777 0.8819916  0.000000e+00</span></span>
<span><span class="co">##            padj    pct_in  pct_out</span></span>
<span><span class="co">## 1 5.448469e-179  70.41499 31.61094</span></span>
<span><span class="co">## 2 3.002848e-304  91.39896 34.67463</span></span>
<span><span class="co">## 3 4.147972e-113  99.02280 76.12536</span></span>
<span><span class="co">## 4 2.381214e-234  86.27244 35.59561</span></span>
<span><span class="co">## 5 5.786716e-154  98.38710 84.30514</span></span>
<span><span class="co">## 6  0.000000e+00 100.00000 99.35360</span></span></code></pre>
<p><strong>Note</strong>: The MERSCOPE data used for SE discovery
includes a limited number of genes. To enhance the analysis, you can
extend the differential expression analysis to the whole transcriptome
by aligning scRNA-seq data to the spatial transcriptomics data via
CytoSPACE.</p>
</div>
<div class="section level3">
<h3 id="visualizing-the-expression-of-cell-state-markers">Visualizing the expression of cell state markers<a class="anchor" aria-label="anchor" href="#visualizing-the-expression-of-cell-state-markers"></a>
</h3>
<p>Once you’ve identified potential markers, you can visualize their
expression across SEs. Below is an example using the STAT1 gene.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Gene expression across all SEs</span></span>
<span><span class="va">gg</span> <span class="op">&lt;-</span> <span class="va">scmeta</span><span class="op">[</span><span class="va">idx</span>, <span class="op">]</span></span>
<span><span class="va">gg</span><span class="op">$</span><span class="va">Expression</span> <span class="op">&lt;-</span> <span class="va">normdata</span><span class="op">[</span><span class="st">"STAT1"</span>, <span class="va">idx</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">gg</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">SE</span>, <span class="va">Expression</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"STAT1 expression"</span><span class="op">)</span></span></code></pre></div>
<p><img src="SingleSample_files/figure-html/visualizedeg-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>The session info can be invaluable for users who encounter issues, as
it allows them to replicate the exact environment and identify potential
discrepancies in package versions or configurations that might be
causing problems.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur/Monterey 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] presto_1.0.0          Rcpp_1.0.10           SpatialEcoTyper_0.0.4</span></span>
<span><span class="co">##  [4] NMF_0.27              Biobase_2.56.0        BiocGenerics_0.42.0  </span></span>
<span><span class="co">##  [7] cluster_2.1.4         rngtools_1.5.2        registry_0.5-1       </span></span>
<span><span class="co">## [10] RANN_2.6.1            Matrix_1.5-3          R.utils_2.12.2       </span></span>
<span><span class="co">## [13] R.oo_1.25.0           R.methodsS3_1.8.2     googledrive_2.1.1    </span></span>
<span><span class="co">## [16] data.table_1.14.6     SeuratObject_4.1.3    Seurat_4.3.0         </span></span>
<span><span class="co">## [19] ggplot2_3.4.0         dplyr_1.1.0          </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] circlize_0.4.15        systemfonts_1.0.4      plyr_1.8.8            </span></span>
<span><span class="co">##   [4] igraph_1.3.5           lazyeval_0.2.2         sp_1.6-0              </span></span>
<span><span class="co">##   [7] splines_4.2.0          listenv_0.9.0          scattermore_0.8       </span></span>
<span><span class="co">##  [10] gridBase_0.4-7         digest_0.6.31          foreach_1.5.2         </span></span>
<span><span class="co">##  [13] htmltools_0.5.8.1      fansi_1.0.4            magrittr_2.0.3        </span></span>
<span><span class="co">##  [16] memoise_2.0.1          tensor_1.5             doParallel_1.0.17     </span></span>
<span><span class="co">##  [19] ROCR_1.0-11            ComplexHeatmap_2.12.1  globals_0.16.2        </span></span>
<span><span class="co">##  [22] matrixStats_0.63.0     pkgdown_2.0.9          spatstat.sparse_3.0-0 </span></span>
<span><span class="co">##  [25] colorspace_2.1-0       ggrepel_0.9.2          textshaping_0.3.7     </span></span>
<span><span class="co">##  [28] xfun_0.37              crayon_1.5.2           jsonlite_1.8.4        </span></span>
<span><span class="co">##  [31] progressr_0.13.0       spatstat.data_3.0-0    survival_3.5-0        </span></span>
<span><span class="co">##  [34] zoo_1.8-11             iterators_1.0.14       glue_1.6.2            </span></span>
<span><span class="co">##  [37] polyclip_1.10-4        pals_1.7               gtable_0.3.1          </span></span>
<span><span class="co">##  [40] gargle_1.5.1           leiden_0.4.3           GetoptLong_1.0.5      </span></span>
<span><span class="co">##  [43] shape_1.4.6            future.apply_1.10.0    maps_3.4.1            </span></span>
<span><span class="co">##  [46] abind_1.4-5            scales_1.2.1           DBI_1.1.3             </span></span>
<span><span class="co">##  [49] spatstat.random_3.1-3  miniUI_0.1.1.1         viridisLite_0.4.1     </span></span>
<span><span class="co">##  [52] xtable_1.8-4           clue_0.3-64            reticulate_1.28       </span></span>
<span><span class="co">##  [55] mapproj_1.2.11         stats4_4.2.0           htmlwidgets_1.6.1     </span></span>
<span><span class="co">##  [58] httr_1.4.7             RColorBrewer_1.1-3     ellipsis_0.3.2        </span></span>
<span><span class="co">##  [61] ica_1.0-3              farver_2.1.1           pkgconfig_2.0.3       </span></span>
<span><span class="co">##  [64] sass_0.4.9             uwot_0.1.14            deldir_1.0-6          </span></span>
<span><span class="co">##  [67] utf8_1.2.3             labeling_0.4.2         tidyselect_1.2.0      </span></span>
<span><span class="co">##  [70] rlang_1.1.1            reshape2_1.4.4         later_1.3.0           </span></span>
<span><span class="co">##  [73] munsell_0.5.0          tools_4.2.0            cachem_1.0.6          </span></span>
<span><span class="co">##  [76] cli_3.6.2              generics_0.1.3         ggridges_0.5.4        </span></span>
<span><span class="co">##  [79] evaluate_0.20          stringr_1.5.0          fastmap_1.1.1         </span></span>
<span><span class="co">##  [82] yaml_2.3.7             ragg_1.3.0             goftest_1.2-3         </span></span>
<span><span class="co">##  [85] knitr_1.42             fs_1.6.0               fitdistrplus_1.1-8    </span></span>
<span><span class="co">##  [88] purrr_1.0.1            pbapply_1.7-0          future_1.30.0         </span></span>
<span><span class="co">##  [91] nlme_3.1-157           mime_0.12              compiler_4.2.0        </span></span>
<span><span class="co">##  [94] rstudioapi_0.14        plotly_4.10.1          curl_5.2.1            </span></span>
<span><span class="co">##  [97] png_0.1-8              spatstat.utils_3.0-1   tibble_3.1.8          </span></span>
<span><span class="co">## [100] bslib_0.7.0            stringi_1.7.12         highr_0.10            </span></span>
<span><span class="co">## [103] desc_1.4.2             lattice_0.20-45        vctrs_0.6.5           </span></span>
<span><span class="co">## [106] pillar_1.9.0           lifecycle_1.0.3        GlobalOptions_0.1.2   </span></span>
<span><span class="co">## [109] spatstat.geom_3.0-6    lmtest_0.9-40          jquerylib_0.1.4       </span></span>
<span><span class="co">## [112] RcppAnnoy_0.0.20       cowplot_1.1.1          irlba_2.3.5.1         </span></span>
<span><span class="co">## [115] httpuv_1.6.8           patchwork_1.1.2        R6_2.5.1              </span></span>
<span><span class="co">## [118] promises_1.2.0.1       KernSmooth_2.23-20     gridExtra_2.3         </span></span>
<span><span class="co">## [121] IRanges_2.30.1         parallelly_1.34.0      codetools_0.2-18      </span></span>
<span><span class="co">## [124] dichromat_2.0-0.1      MASS_7.3-58.2          rjson_0.2.21          </span></span>
<span><span class="co">## [127] rprojroot_2.0.3        withr_2.5.0            sctransform_0.3.5     </span></span>
<span><span class="co">## [130] S4Vectors_0.34.0       grid_4.2.0             tidyr_1.3.0           </span></span>
<span><span class="co">## [133] rmarkdown_2.20         Rtsne_0.16             spatstat.explore_3.0-6</span></span>
<span><span class="co">## [136] shiny_1.7.4</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://profiles.stanford.edu/wubing-zhang" class="external-link">Wubing Zhang</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
