<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="SpatialEcoTyper">
<title>Development of NMF Models for Spatial Ecotype Recovery • SpatialEcoTyper</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Development of NMF Models for Spatial Ecotype Recovery">
<meta property="og:description" content="SpatialEcoTyper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-default navbar-expand-lg bg-light" data-bs-theme="default"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpatialEcoTyper</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.0.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href="../articles/SingleSample.html">Discovery of Spatial Ecotypes from a Single Sample</a>
    <a class="dropdown-item" href="../articles/Integration.html">Discovery of Spatial Ecotypes from Multiple Samples</a>
    <a class="dropdown-item" href="../articles/TrainRecoveryModels.html">Development of NMF Models for Spatial Ecotype Recovery</a>
    <a class="dropdown-item" href="../articles/Recovery_Spatial.html">Recovery of Spatial Ecotypes from Spatial Transcriptomics Data</a>
    <a class="dropdown-item" href="../articles/Recovery_scRNA.html">Recovery of Spatial Ecotypes from Single-Cell Gene Expression Data</a>
    <a class="dropdown-item" href="../articles/Recovery_Bulk.html">Recovery of Spatial Ecotypes from Bulk Gene Expression Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">R functions</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Development of NMF Models for Spatial Ecotype Recovery</h1>
            
      
      <small class="dont-index">Source: <a href="https://digitalcytometry.github/SpatialEcoTypervignettes/TrainRecoveryModels.Rmd" class="external-link"><code>vignettes/TrainRecoveryModels.Rmd</code></a></small>
      <div class="d-none name"><code>TrainRecoveryModels.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="development-of-se-recovery-models">Development of SE recovery models<a class="anchor" aria-label="anchor" href="#development-of-se-recovery-models"></a>
</h2>
<p>After performing Spatial EcoTyper analysis, users can develop machine
learning models to recover spatial ecotypes (SEs) from various types of
gene expression data, including spatial transcriptomics and single-cell
RNA-seq.</p>
<p><strong>First load required packages for this vignette</strong></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://googledrive.tidyverse.org" class="external-link">googledrive</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://renozao.github.io/NMF/" class="external-link">NMF</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://digitalcytometry.github/SpatialEcoTyper" class="external-link">SpatialEcoTyper</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h3>
<p>In this tutorial, we will use the Vizgen MERSCOPE data of a melanoma
sample to demonstrate the process of training SE recovery models. The
training requires a gene expression matrix, cell type annotations, and
SE annotations. The cell type and SE annotations can be obtained from
the SpatialEcoTyper results, which is available in <a href="https://drive.google.com/open?id=1mKz0IHHUGYpS7GpDuNh93YAEPLbd-vTE&amp;usp=drive_fs" class="external-link">Melanoma1_subset_SpatialEcoTyper_results.rds</a>.
The gene expression matrix for the melanoma sample can be downloaded
from <a href="https://drive.google.com/open?id=17K5XR7VEvxN9walV4RtaLOR2cBRG-aMh&amp;usp=drive_fs" class="external-link">here</a>.</p>
<p><strong>Download the data from Google Drive</strong></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_deauth.html" class="external-link">drive_deauth</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Disable Google sign-in requirement</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"13Rc5Rsu8jbnEYYfUse-xQ7ges51LcI7n"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_counts.tsv.gz"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1mKz0IHHUGYpS7GpDuNh93YAEPLbd-vTE"</span><span class="op">)</span>, <span class="st">"Melanoma1_subset_SpatialEcoTyper_results.rds"</span><span class="op">)</span></span></code></pre></div>
<p><strong>Load gene expression matrix</strong></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">"HumanMelanomaPatient1_subset_counts.tsv.gz"</span>, </span>
<span>                sep <span class="op">=</span> <span class="st">"\t"</span>,header <span class="op">=</span> <span class="cn">TRUE</span>, data.table <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">scdata</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">scdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          HumanMelanomaPatient1__cell_3655 HumanMelanomaPatient1__cell_3657</span></span>
<span><span class="co">## PDK4                                    0                                1</span></span>
<span><span class="co">## TNFRSF17                                0                                0</span></span>
<span><span class="co">## ICAM3                                   0                                0</span></span>
<span><span class="co">## FAP                                     1                                0</span></span>
<span><span class="co">## GZMB                                    0                                0</span></span>
<span><span class="co">## TSC2                                    0                                0</span></span>
<span><span class="co">##          HumanMelanomaPatient1__cell_3658 HumanMelanomaPatient1__cell_3660</span></span>
<span><span class="co">## PDK4                                    1                                0</span></span>
<span><span class="co">## TNFRSF17                                0                                0</span></span>
<span><span class="co">## ICAM3                                   0                                0</span></span>
<span><span class="co">## FAP                                     0                                0</span></span>
<span><span class="co">## GZMB                                    0                                0</span></span>
<span><span class="co">## TSC2                                    0                                0</span></span>
<span><span class="co">##          HumanMelanomaPatient1__cell_3661</span></span>
<span><span class="co">## PDK4                                    0</span></span>
<span><span class="co">## TNFRSF17                                0</span></span>
<span><span class="co">## ICAM3                                   0</span></span>
<span><span class="co">## FAP                                     0</span></span>
<span><span class="co">## GZMB                                    0</span></span>
<span><span class="co">## TSC2                                    0</span></span></code></pre>
<p><strong>Load cell type and SE annotations</strong></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"Melanoma1_subset_SpatialEcoTyper_results.rds"</span><span class="op">)</span></span>
<span><span class="va">scmeta</span> <span class="op">=</span> <span class="va">se_results</span><span class="op">$</span><span class="va">metadata</span></span>
<span><span class="va">scmeta</span><span class="op">$</span><span class="va">SE</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">$</span><span class="va">SE</span><span class="op">)</span><span class="op">]</span> <span class="op">=</span> <span class="st">"NonSE"</span> <span class="co">## cells with NA predictions were assigned into the NonSE group</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CellType"</span>, <span class="st">"SE"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                    CellType    SE</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3655 Fibroblast NonSE</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3657 Fibroblast   SE7</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3658 Fibroblast   SE7</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3660 Fibroblast   SE7</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3661 Fibroblast   SE2</span></span>
<span><span class="co">## HumanMelanomaPatient1__cell_3663 Fibroblast NonSE</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="data-normalization">Data normalization<a class="anchor" aria-label="anchor" href="#data-normalization"></a>
</h3>
<p>The data can be normalized using <a href="https://satijalab.org/seurat/articles/seurat5_spatial_vignette_2" class="external-link">SCTransform</a>
or <a href="https://satijalab.org/seurat/reference/normalizedata" class="external-link">NormalizeData</a>.</p>
<p>Here, we are normalizing using SCTransform normalization. We
recommend to install the glmGamPoi package for faster computation.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="st">"glmGamPoi"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"glmGamPoi"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">tmpobj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://satijalab.org/seurat/reference/SCTransform.html" class="external-link">SCTransform</a></span><span class="op">(</span>clip.range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">10</span>, <span class="fl">10</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_version</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\..*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/packageDescription.html" class="external-link">packageVersion</a></span><span class="op">(</span><span class="st">"SeuratObject"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="va">seurat_version</span><span class="op">&lt;</span><span class="fl">5</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">normdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">tmpobj</span>, <span class="st">"data"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="kw">else</span><span class="op">{</span></span>
<span>  <span class="va">normdata</span> <span class="op">&lt;-</span> <span class="va">tmpobj</span><span class="op">[[</span><span class="st">"SCT"</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">data</span></span>
<span><span class="op">}</span></span></code></pre></div>
<details open><summary>
To use <code>NormalizeData</code> for the normalization:
</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span></span></code></pre></div>
</details>
</div>
<div class="section level3">
<h3 id="training">Training<a class="anchor" aria-label="anchor" href="#training"></a>
</h3>
<p>The <code>NMFGenerateWList</code> function will be used to train
cell-type specific NMF models for recovering SEs. Prior to NMF, each
gene is scaled to mean 0 and unit variance within each cell
type-specific expression matrix. To satisfy the non-negativity
requirement of NMF, the cell type-specific expression matrices are
individually processed using posneg transformation, which converts each
expression matrix into two matrices, one containing only positive values
and the other containing only negative values with the sign inverted.
The two matrices are subsequently concatenated to produce the training
data. The <code>NMFGenerateWList</code> function will return a list of
<code>feature × SE</code> matrices, each representing a cell
type-specific gene signature matrix with rows as features and columns as
SEs.</p>
<p><strong>Note</strong>: By default, the <code>NMFGenerateWList</code>
selects the top 2000 variable features (<code>nfeature</code> argument)
and down-samples 2500 cells (<code>downsample</code> argument) within
each cell type for training. After training, it retains 50 genes per SE
(<code>nfeature.per.se</code> argument) in the model. Users can tune the
parameters accordingly.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Before training, ensure that the rows in scmeta matches the columns in normdata.</span></span>
<span><span class="va">normdata</span> <span class="op">=</span> <span class="va">normdata</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">normdata</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">Ws</span> <span class="op">=</span> <span class="fu"><a href="../reference/NMFGenerateWList.html">NMFGenerateWList</a></span><span class="op">(</span><span class="va">normdata</span>, <span class="va">scmeta</span>, </span>
<span>                      CellType <span class="op">=</span> <span class="st">"CellType"</span>, </span>
<span>                      SE <span class="op">=</span> <span class="st">"SE"</span><span class="op">)</span></span>
<span><span class="co">## This step would take ~3 mins for the demo.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">Ws</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "list"</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">Ws</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "B"           "CD4T"        "CD8T"        "DC"          "Endothelial"</span></span>
<span><span class="co">## [6] "Fibroblast"  "Macrophage"  "NK"          "Plasma"</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">Ws</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                   SE1        SE4        SE0       SE6       SE3          SE7</span></span>
<span><span class="co">## ICAM3__pos  0.4602828 0.68105485 0.46715694 0.4675082 0.3917796 6.925485e-01</span></span>
<span><span class="co">## GZMB__pos   0.2488416 0.04870433 0.23949332 0.1045887 0.1098433 2.307047e-01</span></span>
<span><span class="co">## TSC2__pos   0.1187636 0.22960960 0.05278382 0.1243373 0.1470809 2.220446e-16</span></span>
<span><span class="co">## NKG7__pos   0.3449571 0.07637514 0.17873518 0.2234147 0.2530152 2.220446e-16</span></span>
<span><span class="co">## COL1A1__pos 0.3015023 0.17120270 0.62971107 0.2225017 0.3198627 6.758001e-01</span></span>
<span><span class="co">## NFKB1__pos  0.4968451 0.54468453 0.41478627 0.3769514 0.3119630 5.615067e-01</span></span>
<span><span class="co">##                    SE5          SE8          SE2        NonSE</span></span>
<span><span class="co">## ICAM3__pos  0.35733113 3.734384e-01 4.443022e-01 2.695361e-01</span></span>
<span><span class="co">## GZMB__pos   0.09598664 6.517511e-01 2.191695e-01 1.511514e-01</span></span>
<span><span class="co">## TSC2__pos   0.07541921 2.220446e-16 2.220446e-16 2.220446e-16</span></span>
<span><span class="co">## NKG7__pos   0.36767473 4.190680e-01 3.224955e-01 2.220446e-16</span></span>
<span><span class="co">## COL1A1__pos 0.26242749 8.284001e-02 3.852061e-01 4.427656e-01</span></span>
<span><span class="co">## NFKB1__pos  0.55042603 4.694376e-01 1.649369e-01 1.801613e-01</span></span></code></pre>
<p>The resulting W matrices could be used to recover SEs from spatial
transcriptomics and single-cell RNA-seq, by using the
<code>RecoverSE</code> function. The detailed documentation for SE
recovery is available at <a href="Recovery_scRNA.html">Recovery of
Spatial Ecotypes from Single-Cell Gene Expression Data</a> and <a href="Recovery_Spatial.html">Recovery of Spatial Ecotypes from Spatial
Transcriptomics Data</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="development-of-se-deconvolution-model">Development of SE deconvolution model<a class="anchor" aria-label="anchor" href="#development-of-se-deconvolution-model"></a>
</h2>
<p>Users can also develop an NMF model to deconvolve SEs from bulk gene
expression data. The training data can be pseudo-bulk mixtures by
aggregating single-cell transcriptomics. In this tutorial, we will
create pseudo-bulk mixtures from single-cell RNA-seq data of a melanoma
sample, which is also used in the tutorial <a href="Recovery_scRNA.html">Recovery of Spatial Ecotypes from Single-Cell
Gene Expression Data</a>. The raw count data is available at <a href="https://drive.google.com/open?id=15n9zlXed74oeGaO1pythOOM_iWIfuMn2&amp;usp=drive_fs" class="external-link">Melanoma_WU2161_counts.rds</a>
and SE recovery results are available at <a href="https://drive.google.com/open?id=15apmSDhcANjx3i05LIFIm01gelbP7etA&amp;usp=drive_fs" class="external-link">Melanoma_WU2161_RecoveredSEs.rds</a>.</p>
<div class="section level3">
<h3 id="loading-data-1">Loading data<a class="anchor" aria-label="anchor" href="#loading-data-1"></a>
</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Download single-cell gene expression matrix and SE recovery results.</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"15n9zlXed74oeGaO1pythOOM_iWIfuMn2"</span><span class="op">)</span>, <span class="st">"Melanoma_WU2161_counts.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"1nzhKQM6HtX_rzzn3edSQsbodFBO5ycYG"</span><span class="op">)</span>, <span class="st">"Melanoma_WU2161_RecoveredSEs.rds"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load single-cell gene expression matrix.</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"Melanoma_WU2161_counts.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">counts</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 6 x 10 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##                               </span></span>
<span><span class="co">## AL627309.1 . . . . . . . . . .</span></span>
<span><span class="co">## AL627309.5 . . . 1 . . . . . .</span></span>
<span><span class="co">## AP006222.2 . . . . . . . . . .</span></span>
<span><span class="co">## AC114498.1 . . . . . . . . . .</span></span>
<span><span class="co">## AL669831.2 . . . . . . . . . .</span></span>
<span><span class="co">## LINC01409  . . . . . . . . . .</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load SE recovery results</span></span>
<span><span class="va">SEs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"Melanoma_WU2161_RecoveredSEs.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">SEs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## AAACCTGCACATGACT-1 AAACCTGGTGGTCCGT-1 AAACCTGGTTTGTGTG-1 AAACCTGTCCGATATG-1 </span></span>
<span><span class="co">##             "SE10"            "NonSE"             "SE01"             "SE10" </span></span>
<span><span class="co">## AAACCTGTCTAACGGT-1 AAACGGGCACTTAAGC-1 </span></span>
<span><span class="co">##            "NonSE"             "SE03"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="creating-pseudobulk-data">Creating pseudobulk data<a class="anchor" aria-label="anchor" href="#creating-pseudobulk-data"></a>
</h3>
<p>The <code>CreatePseudobulks</code> function will be used to create
pseudo-bulk mixtures from single-cell (spatial) transcriptomics data
with all cells grouped into SEs. It samples cell fractions from a
Gaussian distribution, sets negative fractions to 0 and re-normalizes
fractions to sum to 1 across all SEs. Based on the resulting fractions,
it samples 1,000 cells per pseudo-bulk mixture with replacement,
aggregates their transcriptomes in non-log linear space, and normalize
the resulting mixture to logarithm CPM using Seurat’s
<code>NormalizeData</code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">result</span> <span class="op">=</span> <span class="fu"><a href="../reference/CreatePseudobulks.html">CreatePseudobulks</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">counts</span>, groups <span class="op">=</span> <span class="va">SEs</span>, n_mixtures <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">Mixtures</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span> <span class="co">## Gene expression matrix of pseudobulks</span></span></code></pre></div>
<pre><code><span><span class="co">## 6 x 5 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##             Pseudobulk1 Pseudobulk2  Pseudobulk3 Pseudobulk4  Pseudobulk5</span></span>
<span><span class="co">## AL627309.1 0.0343937521 0.021640535 0.0170462425 0.014167639 0.0153182452</span></span>
<span><span class="co">## AL627309.5 0.0639772941 0.052209771 0.0527358043 0.043242637 0.0495989846</span></span>
<span><span class="co">## AP006222.2 0.0005117182 0.003035767 0.0004621582 0.001253268 0.0004626203</span></span>
<span><span class="co">## AC114498.1 .            .           .            0.006997929 0.0009235317</span></span>
<span><span class="co">## AL669831.2 0.0003700004 0.001110700 0.0003700004 0.001110700 0.0011106999</span></span>
<span><span class="co">## LINC01409  0.0292987853 0.034893356 0.0359603217 0.020077520 0.0287993391</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">Fracs</span><span class="op">)</span> <span class="co">## SE fractions in pseudobulks</span></span></code></pre></div>
<pre><code><span><span class="co">##                  NonSE       SE01       SE02       SE03       SE04       SE05</span></span>
<span><span class="co">## Pseudobulk1 0.11235130 0.13806322 0.09011557 0.07781085 0.13683850 0.04875623</span></span>
<span><span class="co">## Pseudobulk2 0.07805355 0.05541045 0.12062976 0.06458164 0.09817475 0.08205461</span></span>
<span><span class="co">## Pseudobulk3 0.07979042 0.11866467 0.09893247 0.10360987 0.08585473 0.08719972</span></span>
<span><span class="co">## Pseudobulk4 0.13133824 0.11739958 0.03449139 0.04325339 0.09051472 0.08992931</span></span>
<span><span class="co">## Pseudobulk5 0.06343129 0.09475496 0.11817109 0.00643102 0.10974664 0.06458885</span></span>
<span><span class="co">## Pseudobulk6 0.07562488 0.09858894 0.17418506 0.04682772 0.06299713 0.05126319</span></span>
<span><span class="co">##                   SE06       SE07       SE08       SE09       SE10       SE11</span></span>
<span><span class="co">## Pseudobulk1 0.03860297 0.04623464 0.05792983 0.09854653 0.06519633 0.08955402</span></span>
<span><span class="co">## Pseudobulk2 0.08688914 0.10248104 0.10326609 0.04826384 0.07551141 0.08468373</span></span>
<span><span class="co">## Pseudobulk3 0.11771347 0.03439627 0.11011146 0.03980684 0.03433973 0.08958035</span></span>
<span><span class="co">## Pseudobulk4 0.07696805 0.08457932 0.10827692 0.09354075 0.09662320 0.03308514</span></span>
<span><span class="co">## Pseudobulk5 0.06665770 0.13418141 0.11607981 0.04818743 0.10910353 0.06866627</span></span>
<span><span class="co">## Pseudobulk6 0.09638301 0.08546799 0.08951046 0.13391742 0.04531018 0.03992402</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="training-1">Training<a class="anchor" aria-label="anchor" href="#training-1"></a>
</h3>
<p>The <code>NMFGenerateW</code> function will be used to train an NMF
model for SE deconvolution based on the provided SE fractions and gene
expression matrix. Before applying NMF, each gene’s expression is scaled
to have a mean of 0 and unit variance (recommended). To meet the
non-negativity requirement of NMF, the expression matrix is transformed
using the posneg method. This transformation splits the expression
matrix into two matrices: one containing only positive values and the
other containing the absolute values of the negative values. These two
matrices are then concatenated to form the final training data for the
NMF model.</p>
<p><strong>Note</strong>: By default, the <code>NMFGenerateW</code>
selects the top 2000 variable features (<code>nfeature</code> argument)
for training. After training, it retains 50 genes per SE
(<code>nfeature.per.se</code> argument) in the model. Users can tune the
parameters accordingly.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">W</span> <span class="op">=</span> <span class="fu"><a href="../reference/NMFGenerateW.html">NMFGenerateW</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">Fracs</span><span class="op">)</span>, <span class="va">result</span><span class="op">$</span><span class="va">Mixtures</span><span class="op">)</span></span>
<span><span class="co">## This step would take ~1 min for the demo.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">W</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                      NonSE         SE01         SE02         SE03         SE04</span></span>
<span><span class="co">## IGKV3-11__pos 2.220446e-16 2.220446e-16 1.297543e-11 3.680766e-01 4.577781e-02</span></span>
<span><span class="co">## IGLV3-19__pos 2.220446e-16 2.220446e-16 2.220446e-16 1.166077e-14 2.220446e-16</span></span>
<span><span class="co">## IGLV6-57__pos 1.933388e-05 9.010422e-01 4.752353e-07 3.778070e+00 5.905614e-09</span></span>
<span><span class="co">## IGLV1-51__pos 3.961831e-02 1.174834e-01 2.220446e-16 5.177512e+00 3.443881e-01</span></span>
<span><span class="co">## IGKV1-6__pos  1.018897e-13 1.321957e-03 2.220446e-16 1.981312e-06 2.069945e-03</span></span>
<span><span class="co">## IGHV3-49__pos 1.178233e-11 5.771047e-01 2.220446e-16 1.256758e-06 6.744953e-02</span></span>
<span><span class="co">##                       SE05         SE06         SE07         SE08         SE09</span></span>
<span><span class="co">## IGKV3-11__pos 2.220446e-16 2.220446e-16 2.220446e-16 2.220446e-16 2.220446e-16</span></span>
<span><span class="co">## IGLV3-19__pos 2.220446e-16 2.220446e-16 2.220446e-16 2.220446e-16 2.220446e-16</span></span>
<span><span class="co">## IGLV6-57__pos 7.026100e-09 1.298919e-11 3.597849e-11 1.211816e-01 4.107317e-11</span></span>
<span><span class="co">## IGLV1-51__pos 2.220446e-16 3.529110e-07 3.978048e-16 2.220446e-16 2.220446e-16</span></span>
<span><span class="co">## IGKV1-6__pos  2.220446e-16 2.220446e-16 2.220446e-16 2.220446e-16 2.220446e-16</span></span>
<span><span class="co">## IGHV3-49__pos 2.220446e-16 2.220446e-16 2.220446e-16 5.343617e+00 2.220446e-16</span></span>
<span><span class="co">##                       SE10         SE11</span></span>
<span><span class="co">## IGKV3-11__pos 1.034755e-08 4.696292e+00</span></span>
<span><span class="co">## IGLV3-19__pos 2.220446e-16 5.177722e+00</span></span>
<span><span class="co">## IGLV6-57__pos 6.337725e-01 1.826789e-09</span></span>
<span><span class="co">## IGLV1-51__pos 5.065311e-05 7.663140e-09</span></span>
<span><span class="co">## IGKV1-6__pos  2.749733e-05 4.986856e+00</span></span>
<span><span class="co">## IGHV3-49__pos 2.220446e-16 6.721149e-08</span></span></code></pre>
<p>The resulting W matrix can be used to infer SE sbundances from bulk
gene expression profiles, by using the <code>DeconvoluteSE</code>
function. The detailed documentation for SE deconvolution is available
at <a href="Recovery_Bulk.html">Recovery of Spatial Ecotypes from Bulk
Gene Expression Data</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>The session info can be invaluable for users who encounter issues, as
it allows them to replicate the exact environment and identify potential
discrepancies in package versions or configurations that might be
causing problems.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Big Sur/Monterey 10.16</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] SpatialEcoTyper_0.0.4 RANN_2.6.1            Matrix_1.5-3         </span></span>
<span><span class="co">##  [4] dplyr_1.1.0           data.table_1.14.6     NMF_0.27             </span></span>
<span><span class="co">##  [7] Biobase_2.56.0        BiocGenerics_0.42.0   cluster_2.1.4        </span></span>
<span><span class="co">## [10] rngtools_1.5.2        registry_0.5-1        SeuratObject_4.1.3   </span></span>
<span><span class="co">## [13] Seurat_4.3.0          googledrive_2.1.1    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] circlize_0.4.15        systemfonts_1.0.4      plyr_1.8.8            </span></span>
<span><span class="co">##   [4] igraph_1.3.5           lazyeval_0.2.2         sp_1.6-0              </span></span>
<span><span class="co">##   [7] splines_4.2.0          listenv_0.9.0          scattermore_0.8       </span></span>
<span><span class="co">##  [10] ggplot2_3.4.0          gridBase_0.4-7         digest_0.6.31         </span></span>
<span><span class="co">##  [13] foreach_1.5.2          htmltools_0.5.8.1      fansi_1.0.4           </span></span>
<span><span class="co">##  [16] magrittr_2.0.3         memoise_2.0.1          tensor_1.5            </span></span>
<span><span class="co">##  [19] doParallel_1.0.17      ROCR_1.0-11            ComplexHeatmap_2.12.1 </span></span>
<span><span class="co">##  [22] globals_0.16.2         matrixStats_0.63.0     R.utils_2.12.2        </span></span>
<span><span class="co">##  [25] pkgdown_2.0.9          spatstat.sparse_3.0-0  colorspace_2.1-0      </span></span>
<span><span class="co">##  [28] ggrepel_0.9.2          textshaping_0.3.7      xfun_0.37             </span></span>
<span><span class="co">##  [31] crayon_1.5.2           jsonlite_1.8.4         progressr_0.13.0      </span></span>
<span><span class="co">##  [34] spatstat.data_3.0-0    survival_3.5-0         zoo_1.8-11            </span></span>
<span><span class="co">##  [37] iterators_1.0.14       glue_1.6.2             polyclip_1.10-4       </span></span>
<span><span class="co">##  [40] gtable_0.3.1           gargle_1.5.1           leiden_0.4.3          </span></span>
<span><span class="co">##  [43] GetoptLong_1.0.5       shape_1.4.6            future.apply_1.10.0   </span></span>
<span><span class="co">##  [46] abind_1.4-5            scales_1.2.1           DBI_1.1.3             </span></span>
<span><span class="co">##  [49] spatstat.random_3.1-3  miniUI_0.1.1.1         Rcpp_1.0.10           </span></span>
<span><span class="co">##  [52] viridisLite_0.4.1      xtable_1.8-4           clue_0.3-64           </span></span>
<span><span class="co">##  [55] reticulate_1.28        stats4_4.2.0           htmlwidgets_1.6.1     </span></span>
<span><span class="co">##  [58] httr_1.4.7             RColorBrewer_1.1-3     ellipsis_0.3.2        </span></span>
<span><span class="co">##  [61] ica_1.0-3              R.methodsS3_1.8.2      pkgconfig_2.0.3       </span></span>
<span><span class="co">##  [64] sass_0.4.9             uwot_0.1.14            deldir_1.0-6          </span></span>
<span><span class="co">##  [67] utf8_1.2.3             tidyselect_1.2.0       rlang_1.1.1           </span></span>
<span><span class="co">##  [70] reshape2_1.4.4         later_1.3.0            munsell_0.5.0         </span></span>
<span><span class="co">##  [73] tools_4.2.0            cachem_1.0.6           cli_3.6.2             </span></span>
<span><span class="co">##  [76] generics_0.1.3         ggridges_0.5.4         evaluate_0.20         </span></span>
<span><span class="co">##  [79] stringr_1.5.0          fastmap_1.1.1          yaml_2.3.7            </span></span>
<span><span class="co">##  [82] ragg_1.3.0             goftest_1.2-3          knitr_1.42            </span></span>
<span><span class="co">##  [85] fs_1.6.0               fitdistrplus_1.1-8     purrr_1.0.1           </span></span>
<span><span class="co">##  [88] pbapply_1.7-0          future_1.30.0          nlme_3.1-157          </span></span>
<span><span class="co">##  [91] mime_0.12              R.oo_1.25.0            compiler_4.2.0        </span></span>
<span><span class="co">##  [94] rstudioapi_0.14        plotly_4.10.1          curl_5.2.1            </span></span>
<span><span class="co">##  [97] png_0.1-8              spatstat.utils_3.0-1   tibble_3.1.8          </span></span>
<span><span class="co">## [100] bslib_0.7.0            stringi_1.7.12         desc_1.4.2            </span></span>
<span><span class="co">## [103] lattice_0.20-45        vctrs_0.6.5            pillar_1.9.0          </span></span>
<span><span class="co">## [106] lifecycle_1.0.3        GlobalOptions_0.1.2    spatstat.geom_3.0-6   </span></span>
<span><span class="co">## [109] lmtest_0.9-40          jquerylib_0.1.4        RcppAnnoy_0.0.20      </span></span>
<span><span class="co">## [112] cowplot_1.1.1          irlba_2.3.5.1          httpuv_1.6.8          </span></span>
<span><span class="co">## [115] patchwork_1.1.2        R6_2.5.1               promises_1.2.0.1      </span></span>
<span><span class="co">## [118] KernSmooth_2.23-20     gridExtra_2.3          IRanges_2.30.1        </span></span>
<span><span class="co">## [121] parallelly_1.34.0      codetools_0.2-18       MASS_7.3-58.2         </span></span>
<span><span class="co">## [124] rjson_0.2.21           rprojroot_2.0.3        withr_2.5.0           </span></span>
<span><span class="co">## [127] sctransform_0.3.5      S4Vectors_0.34.0       grid_4.2.0            </span></span>
<span><span class="co">## [130] tidyr_1.3.0            rmarkdown_2.20         Rtsne_0.16            </span></span>
<span><span class="co">## [133] spatstat.explore_3.0-6 shiny_1.7.4</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://profiles.stanford.edu/wubing-zhang" class="external-link">Wubing Zhang</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
