<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function performs SpatialEcoTyper analysis on multiple spatial transcriptomics datasets.
It normalizes the input data, performs SpatialEcoTyper analysis on each dataset,
and integrates the results across samples."><title>Integrate Multiple Spatial Transcriptomics Datasets to Identify Conserved Spatial Ecotypes — MultiSpatialEcoTyper • SpatialEcoTyper</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Integrate Multiple Spatial Transcriptomics Datasets to Identify Conserved Spatial Ecotypes — MultiSpatialEcoTyper"><meta property="og:description" content="This function performs SpatialEcoTyper analysis on multiple spatial transcriptomics datasets.
It normalizes the input data, performs SpatialEcoTyper analysis on each dataset,
and integrates the results across samples."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-default navbar-expand-lg bg-light" data-bs-theme="default"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpatialEcoTyper</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.0.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href="../articles/SingleSample.html">Discovery of Spatial Ecotypes from a Single Sample</a>
    <a class="dropdown-item" href="../articles/Integration.html">Discovery of Spatial Ecotypes from Multiple Samples</a>
    <a class="dropdown-item" href="../articles/TrainRecoveryModels.html">Development of NMF Models for Spatial Ecotype Recovery</a>
    <a class="dropdown-item" href="../articles/Recovery_Spatial.html">Recovery of Spatial Ecotypes from Spatial Transcriptomics Data</a>
    <a class="dropdown-item" href="../articles/Recovery_scRNA.html">Recovery of Spatial Ecotypes from Single-Cell Gene Expression Data</a>
    <a class="dropdown-item" href="../articles/Recovery_Bulk.html">Recovery of Spatial Ecotypes from Bulk Gene Expression Data</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">R functions</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Integrate Multiple Spatial Transcriptomics Datasets to Identify Conserved Spatial Ecotypes</h1>
      <small class="dont-index">Source: <a href="https://digitalcytometry.github/SpatialEcoTyperR/MultiSpatialEcoTyper.R" class="external-link"><code>R/MultiSpatialEcoTyper.R</code></a></small>
      <div class="d-none name"><code>MultiSpatialEcoTyper.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function performs SpatialEcoTyper analysis on multiple spatial transcriptomics datasets.
It normalizes the input data, performs SpatialEcoTyper analysis on each dataset,
and integrates the results across samples.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">MultiSpatialEcoTyper</span><span class="op">(</span></span>
<span>  <span class="va">data_list</span>,</span>
<span>  <span class="va">metadata_list</span>,</span>
<span>  outdir <span class="op">=</span> <span class="st">"./"</span>,</span>
<span>  normalization.method <span class="op">=</span> <span class="st">"None"</span>,</span>
<span>  nmf_ranks <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  nrun.per.rank <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  radius <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  min.cts.per.region <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  nfeatures <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  min.features <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  Region <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  subresolution <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>data_list</dt>
<dd><p>A named list of expression matrices where each matrix represents
gene expression data for a sample. The columns of each matrix correspond to cells,
and the rows correspond to genes. Sample names should be used as list names.
Otherwise, the samples will be named as 'Sample1' through 'SampleN'.</p></dd>


<dt>metadata_list</dt>
<dd><p>A named list of metadata data frames where each data frame
contains metadata corresponding to the cells in the expression matrices. Each row
should correspond to a column (cell) in the expression matrices. Each metadata
should include at least three columns, including X, Y and CellType.</p></dd>


<dt>outdir</dt>
<dd><p>Directory where the results will be saved. Defaults to the current
directory with a subdirectory named "SpatialEcoTyper_results_" followed by the current date.</p></dd>


<dt>normalization.method</dt>
<dd><p>Method for normalizing the expression data. Options
include "None" (default), "SCT", or other methods compatible with Seurat's `NormalizeData` function.</p></dd>


<dt>nmf_ranks</dt>
<dd><p>Integer or a vector specifying the number of clusters (10 by default).
When an integer vector is supplied, the function will test all supplied numbers and
select the optimal number, which takes time.</p></dd>


<dt>nrun.per.rank</dt>
<dd><p>An integer specifying the the number of runs per rank for NMF (default: 30).</p></dd>


<dt>radius</dt>
<dd><p>Numeric specifying the radius (in the same units as spatial coordinates)
for defining spatial neighborhoods around each cell. Default is 50.</p></dd>


<dt>min.cts.per.region</dt>
<dd><p>Integer specifying the minimum number of cell types (default: 2) required for a microregion.</p></dd>


<dt>nfeatures</dt>
<dd><p>An integer specifying the maximum number of top variable genes to select for each cell type.</p></dd>


<dt>min.features</dt>
<dd><p>An integer specifying the minimum number of shared features (genes) required across samples.</p></dd>


<dt>Region</dt>
<dd><p>Character string specifying the column name in metadata data frames
containing region annotations (default: NULL). Pathologist annotation is recommended if available.</p></dd>


<dt>subresolution</dt>
<dd><p>Numeric specifying the resolution for clustering within each sample.</p></dd>


<dt>ncores</dt>
<dd><p>Integer specifying the number of cores for parallel processing. Default is 1.</p></dd>


<dt>seed</dt>
<dd><p>An integer used to seed the random number generator for NMF analysis.</p></dd>


<dt>...</dt>
<dd><p>Additional arguments passed to the `SpatialEcoTyper` function.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>The function saves the results in the specified output directory.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function takes a list of gene expression matrices and corresponding metadata,
         normalizes the data if specified, performs SpatialEcoTyper on each sample, and
         integrates the results across multiple samples to identify conserved spatial ecotypes.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># See https://digitalcytometry.github.io/SpatialEcoTyper/docs/articles/Integration.html</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://profiles.stanford.edu/wubing-zhang" class="external-link">Wubing Zhang</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

