<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function identifies spatially distinct cellular ecosystems (SE) from a single sample."><title>Identify Spatial EcoTypes from Single-cell Spatial Data (A Single Sample) — SpatialEcoTyper • SpatialEcoTyper</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Identify Spatial EcoTypes from Single-cell Spatial Data (A Single Sample) — SpatialEcoTyper"><meta property="og:description" content="This function identifies spatially distinct cellular ecosystems (SE) from a single sample."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-default navbar-expand-lg bg-light" data-bs-theme="default"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SpatialEcoTyper</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.0.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-vignettes">Vignettes</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-vignettes">
    <a class="dropdown-item" href="../articles/SingleSample.html">Discovery of Spatial Ecotypes from a Single Sample</a>
    <a class="dropdown-item" href="../articles/Integration.html">Discovery of Spatial Ecotypes from Multiple Samples</a>
    <a class="dropdown-item" href="../articles/TrainRecoveryModels.html">Development of Spatial Ecotype Recovery Models</a>
    <a class="dropdown-item" href="../articles/Recovery_Spatial.html">Recovery of Spatial Ecotypes from Spatial Transcriptomics Data</a>
    <a class="dropdown-item" href="../articles/Recovery_scRNA.html">Recovery of Spatial Ecotypes from Single-Cell Gene Expression Data</a>
    <a class="dropdown-item" href="../articles/Recovery_Bulk.html">Recovery of Spatial Ecotypes from Bulk Gene Expression Data</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">R functions</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Identify Spatial EcoTypes from Single-cell Spatial Data (A Single Sample)</h1>
      <small class="dont-index">Source: <a href="https://digitalcytometry.github/SpatialEcoTyperR/SpatialEcoTyper.R" class="external-link"><code>R/SpatialEcoTyper.R</code></a></small>
      <div class="d-none name"><code>SpatialEcoTyper.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function identifies spatially distinct cellular ecosystems (SE) from a single sample.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">SpatialEcoTyper</span><span class="op">(</span></span>
<span>  <span class="va">normdata</span>,</span>
<span>  <span class="va">metadata</span>,</span>
<span>  outprefix <span class="op">=</span> <span class="st">"SE"</span>,</span>
<span>  radius <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  resolution <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  nfeatures <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  npcs <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  k.sn <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  k <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  min.cells <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  min.features <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  min.cts.per.region <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  min.cells.per.region <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  iterations <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  binsize <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">radius</span> <span class="op">*</span> <span class="fl">1.4</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>normdata</dt>
<dd><p>A matrix representing normalized gene expression data,
where rows correspond to genes and columns correspond to cells.</p></dd>


<dt>metadata</dt>
<dd><p>A data frame containing metadata associated with each cell.
Must include spatial coordinates (e.g., X and Y) as well as cell type annotations.
The row names of the <code>metadata</code> must match the column names of the <code>normdata</code>.</p></dd>


<dt>outprefix</dt>
<dd><p>Character string specifying the prefix for output file names.</p></dd>


<dt>radius</dt>
<dd><p>Numeric specifying the radius (in the same units as spatial coordinates)
for defining spatial neighborhoods around each cell. Default is 50.</p></dd>


<dt>resolution</dt>
<dd><p>Numeric specifying the resolution for Louvain clustering (default: 0.8).</p></dd>


<dt>nfeatures</dt>
<dd><p>Integer specifying the number of top variable features (default: 3000) used for PCA.</p></dd>


<dt>npcs</dt>
<dd><p>Integer specifying the number of principal components (PCs) (default: 20).</p></dd>


<dt>k.sn</dt>
<dd><p>Integer specifying the number of spatial nearest neighbors (default: 50) for constructing similarity network.</p></dd>


<dt>k</dt>
<dd><p>Integer specifying the number of spatial nearest neighbors (default: 20) used to construct spatial meta-cells.</p></dd>


<dt>min.cells</dt>
<dd><p>Minimum number of cells / spatial-meta-cells (default: 5)  expressing a feature/gene.</p></dd>


<dt>min.features</dt>
<dd><p>Minimum number of features (default: 10) detected in a cell / spatial-meta-cell.</p></dd>


<dt>min.cts.per.region</dt>
<dd><p>Integer specifying the minimum number of cell types (default: 2) required for a microregion.</p></dd>


<dt>min.cells.per.region</dt>
<dd><p>Integer specifying the minimum number of cells (default: 2) in each spatial microregion.</p></dd>


<dt>iterations</dt>
<dd><p>Integer specifying the number of iterations (default: 5) for SNF analysis.</p></dd>


<dt>ncores</dt>
<dd><p>Integer specifying the number of CPU cores to use for parallel processing.</p></dd>


<dt>binsize</dt>
<dd><p>Numeric specifying the bin size for spatial discretization of coordinates. By default,
this size is determined based on the specified radius (radius*1.4 µm). Increasing the binsize will
downsample microregions and expedite the analysis, while it might eliminate cells located between bins from the SE discovery analysis.</p>
<p>#' @return A list containing two elements:</p><dl><dt>obj</dt>
<dd><p>A seurat object constructed from fused similarity network of sptial microregions</p></dd>

  <dt>metadata</dt>
<dd><p>Updated <code>metadata</code>, with a new column (`SE`) added</p></dd>


</dl></dd>

</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># See https://digitalcytometry.github.io/SpatialEcoTyper/docs/articles/SingleSample.html</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">parallel</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://googledrive.tidyverse.org" class="external-link">googledrive</a></span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://henrikbengtsson.github.io/R.utils/" class="external-link">R.utils</a></span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://digitalcytometry.github/SpatialEcoTyper" class="external-link">SpatialEcoTyper</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_deauth.html" class="external-link">drive_deauth</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Disable Google sign-in requirement</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"13Rc5Rsu8jbnEYYfUse-xQ7ges51LcI7n"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_counts.tsv.gz"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in drive_download(as_id("13Rc5Rsu8jbnEYYfUse-xQ7ges51LcI7n"), "HumanMelanomaPatient1_subset_counts.tsv.gz"):</span> Local `path` already exists and overwrite is `FALSE`:</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> <a href="file:///Users/wzhang39/Library/CloudStorage/Box-Box/WZ/EcoSystem/0_Script/SpatialEcoTyper/docs/reference/HumanMelanomaPatient1_subset_counts.tsv.gz"><span style="color: #0000BB;">HumanMelanomaPatient1_subset_counts.tsv.gz</span></a></span>
<span class="r-in"><span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_download.html" class="external-link">drive_download</a></span><span class="op">(</span><span class="fu"><a href="https://googledrive.tidyverse.org/reference/drive_id.html" class="external-link">as_id</a></span><span class="op">(</span><span class="st">"12xcZNhpT-xbhcG8kX1QAdTeM9TKeFAUW"</span><span class="op">)</span>, <span class="st">"HumanMelanomaPatient1_subset_scmeta.tsv"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in drive_download(as_id("12xcZNhpT-xbhcG8kX1QAdTeM9TKeFAUW"), "HumanMelanomaPatient1_subset_scmeta.tsv"):</span> Local `path` already exists and overwrite is `FALSE`:</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> <a href="file:///Users/wzhang39/Library/CloudStorage/Box-Box/WZ/EcoSystem/0_Script/SpatialEcoTyper/docs/reference/HumanMelanomaPatient1_subset_scmeta.tsv"><span style="color: #0000BB;">HumanMelanomaPatient1_subset_scmeta.tsv</span></a></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Load single-cell gene expression matrix. Rows are genes, columns are cells.</span></span></span>
<span class="r-in"><span><span class="va">scdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="st">"HumanMelanomaPatient1_subset_counts.tsv.gz"</span>,</span></span>
<span class="r-in"><span>                sep <span class="op">=</span> <span class="st">"\t"</span>,header <span class="op">=</span> <span class="cn">TRUE</span>, data.table <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">scdata</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>  <span class="co"># set genes as row names</span></span></span>
<span class="r-in"><span><span class="va">scdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">normdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">normdata</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 x 5 sparse Matrix of class "dgCMatrix"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          HumanMelanomaPatient1__cell_3655 HumanMelanomaPatient1__cell_3657</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> PDK4                             .                                4.221184</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TNFRSF17                         .                                .       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ICAM3                            .                                .       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FAP                              3.552438                         .       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GZMB                             .                                .       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TSC2                             .                                .       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          HumanMelanomaPatient1__cell_3658 HumanMelanomaPatient1__cell_3660</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> PDK4                             3.208945                                .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TNFRSF17                         .                                       .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ICAM3                            .                                       .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FAP                              .                                       .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GZMB                             .                                       .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TSC2                             .                                       .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          HumanMelanomaPatient1__cell_3661</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> PDK4                                    .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TNFRSF17                                .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ICAM3                                   .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> FAP                                     .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GZMB                                    .</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TSC2                                    .</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Load single-cell metadata. Three columns are required, including X, Y, and CellType. Others are optional.</span></span></span>
<span class="r-in"><span><span class="va">scmeta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"HumanMelanomaPatient1_subset_scmeta.tsv"</span>,</span></span>
<span class="r-in"><span>                     sep <span class="op">=</span> <span class="st">"\t"</span>,header <span class="op">=</span> <span class="cn">TRUE</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">scmeta</span> <span class="op">&lt;-</span> <span class="va">scmeta</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">scdata</span><span class="op">)</span>, <span class="op">]</span> <span class="co"># match the cell ids in scdata and scmeta</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                         X         Y   CellType CellTypeName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_3655 1894.706 -6367.766 Fibroblast  Fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_3657 1942.480 -6369.602 Fibroblast  Fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_3658 1963.007 -6374.026 Fibroblast  Fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_3660 1981.600 -6372.266 Fibroblast  Fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_3661 1742.939 -6374.851 Fibroblast  Fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_3663 1921.683 -6383.309 Fibroblast  Fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                  Region Dist2Interface</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_3655 Stroma      -883.1752</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_3657 Stroma      -894.8463</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_3658 Stroma      -904.1115</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_3660 Stroma      -907.8909</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_3661 Stroma      -874.2712</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_3663 Stroma      -903.6559</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">se_results</span> <span class="op">&lt;-</span> <span class="fu">SpatialEcoTyper</span><span class="op">(</span><span class="va">normdata</span>, <span class="va">scmeta</span>,</span></span>
<span class="r-in"><span>                              outprefix <span class="op">=</span> <span class="st">"Melanoma1_subset"</span>,</span></span>
<span class="r-in"><span>                              radius <span class="op">=</span> <span class="fl">50</span>, ncores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-21 14:37:13 Remove 88 genes expressed in fewer than 5 cells</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-21 14:37:13 Construct spatial meta cells for each cell type</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 		Total spatial microregions: 2315</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 		Total spatial meta cells: 6343</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 		Total single cells: 27907</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-21 14:37:14 PCA for each cell type</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-21 14:37:16 Construct cell-type-specific similarity network</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 		Remove 609 spatial microregions with fewer than 2 cell types </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-21 14:37:16 Similarity network fusion</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-21 14:37:16 Normalize networks ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-21 14:37:17 Calculate the local transition matrix ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2024-08-21 14:37:17 Perform the diffusion ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	2024-08-21 14:37:17 Iteration: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	2024-08-21 14:37:20 Iteration: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	2024-08-21 14:37:30 Iteration: 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	2024-08-21 14:37:40 Iteration: 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	2024-08-21 14:37:50 Iteration: 5</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> This message will be shown once per session</span>
<span class="r-in"><span><span class="co"># Extract the Seurat object and updated single-cell metadata</span></span></span>
<span class="r-in"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="va">se_results</span><span class="op">$</span><span class="va">obj</span> <span class="co"># A Seurat object</span></span></span>
<span class="r-in"><span><span class="va">obj</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An object of class Seurat </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1633 features across 1633 samples within 1 assay </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Active assay: RNA (1633 features, 1633 variable features)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  2 dimensional reductions calculated: pca, umap</span>
<span class="r-in"><span><span class="va">scmeta</span> <span class="op">&lt;-</span> <span class="va">se_results</span><span class="op">$</span><span class="va">metadata</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">SE</span><span class="op">)</span> <span class="co"># Updated single-cell meta data, with SE annotation added</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scmeta</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                         X         Y   CellType CellTypeName</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_4675 4578.019 -6445.993 Fibroblast  Fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_4738 4529.162 -6504.117 Fibroblast  Fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_4774 4533.228 -6529.591 Fibroblast  Fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_4803 4340.813 -6562.381 Fibroblast  Fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_4806 4401.938 -6567.152 Fibroblast  Fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_4808 4388.422 -6570.215 Fibroblast  Fibroblasts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                        Region Dist2Interface  SE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_4675 Outer margin     -197.09807 SE0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_4738 Outer margin     -186.13295 SE0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_4774 Outer margin     -199.69317 SE0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_4803 Outer margin      -95.54956 SE0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_4806 Outer margin     -124.01602 SE0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> HumanMelanomaPatient1__cell_4808 Outer margin     -119.07031 SE0</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://profiles.stanford.edu/wubing-zhang" class="external-link">Wubing Zhang</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

